<!DOCTYPE html><!--[if lt IE 7]>      <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]>         <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]>         <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]><!--> <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js"> <!--<![endif]-->
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
        <meta name="description" content="random musings of a molecular biologist turned code jockey in the era of big data and open science.">
        <meta name="viewport" content="width=device-width">
        <title>Arthropod Genomics Symposium 2013 – Notre Dame &mdash; moving the lamppost</title>
            <link rel="stylesheet" href="../../../_static/normalize.css" type="text/css">
            <link rel="stylesheet" href="../../../_static/sphinx.css" type="text/css">
            <link rel="stylesheet" href="../../../_static/main.css" type="text/css">
            <link rel="stylesheet" href="../../../_static/flat.css" type="text/css">
            <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
            <link rel="stylesheet" href="../../../_static/webfont.css" type="text/css">
        <link rel="shortcut icon" href="../../../_static/tinkerer.ico" /><!-- Load modernizr and JQuery -->
        <script src="../../../_static/vendor/modernizr-2.6.2.min.js"></script>
        <script src="//ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js"></script>
        <script>window.jQuery || document.write('<script src="../../../_static/vendor/jquery-1.8.2.min.js"><\/script>')</script>
        <script src="../../../_static/plugins.js"></script>
        <script src="../../../_static/main.js"></script>
        <link rel="next" title="Fundamentals of Gene Expression: the ‘Central Dogma’ of Molecular Biology" href="../../05/18/fundamentals_of_gene_expression__the__central_dogma__of_molecular_biology.html" /><link rel="prev" title="Speak, for the Cup of Hemlock is not yet on Your Lips" href="../29/speak__for_the_cup_of_hemlock_is_not_yet_on_your_lips.html" /><link rel="alternate" type="application/rss+xml" title="RSS" href="http://feeds.feedburner.com/MovingTheLamppost" /><script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../../',
        VERSION:     '1.2.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script><script type="text/javascript" src="../../../_static/underscore.js"></script><script type="text/javascript" src="../../../_static/doctools.js"></script><script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><script type="text/javascript" src="../../../_static/disqus.js"></script><script type="text/javascript" src="../../../_static/google_analytics.js"></script>

    <script type="text/javascript">
        $(document).ready(function () {
            // Scroll to content if on small screen
            if (screen.width < 480)
            {
                $(document).scrollTop(document.getElementsByTagName("article")[0].offsetTop - 44);
            }
        });
    </script>
    
    <script type="text/javascript">
        $(document).ready(function () {
            // Adjusts document height if sidebar is taller
            var documentwrapper = document.getElementsByTagName("article")[0];
            var sidebar = document.getElementsByClassName("sidebar")[0];

            if (sidebar.offsetHeight > documentwrapper.offsetHeight)
            {
                var mainwrapper = document.getElementsByClassName("main")[0];
                documentwrapper.style.minHeight = mainwrapper.offsetHeight + "px";
            }

            // Scroll to content if on small screen
            if (screen.width < 480)
            {
                $(document).scrollTop(document.getElementsByTagName("article")[0].offsetTop - 44);
            }
        });
    </script>
    
    <link rel="stylesheet" type="text/css" href="http://fonts.googleapis.com/css?family=Droid+Serif:i|Noto+Sans:i|Titillium+Web|Titillium+Web:i"></head>
    <body>
        <!--[if lt IE 7]>
            <p class="chromeframe">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> or <a href="http://www.google.com/chromeframe/?redirect=true">activate Google Chrome Frame</a> to improve your experience.</p>
        <![endif]-->

      <div id="container"><header>
            <hgroup>
              <h1><a href="../../../index.html">moving the lamppost</a></h1><h2>random musings of a molecular biologist turned code jockey in the era of big data and open science.</h2></hgroup>
          </header>
      <nav>
            <ul><li class="main-nav">
                  <a href="../../../index.html">Home</a>
                </li>
              <li class="main-nav">
                  <a href="../../../pages/about.html">About</a>
                </li>
              <li class="main-nav">
                  <a href="../../../pages/glossary.html">Glossary</a>
                </li>
              </ul>
          </nav><div class="main-container"><div class="main wrapper clearfix"><article><div class="timestamp postmeta">
            <span>June 14, 2013</span>
        </div>
    <div class="section" id="arthropod-genomics-symposium-2013-notre-dame">
<h1>Arthropod Genomics Symposium 2013 &#8211; Notre Dame</h1>
<p>My notes on talks from the <a class="reference external" href="http://globalhealth.nd.edu/7th-annual-arthropod-genomics-symposium/">7th Annual Arthropod Genomics Symposium</a>  If you are interested, you can follow live tweets from <a class="reference external" href="https://twitter.com/search?q=%2313arthgen&amp;src=typd">#13ArthGen</a>.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ol class="arabic simple">
<li><strong>COMING SOON:</strong> links to or embedded versions of the recorded talks for each section. These will be added as soon as I can get the location of where they will be hosted.</li>
<li><strong>My Spelling:</strong> Obviously, my spelling is gonna suck on this.  Please don&#8217;t hold it against me.</li>
<li><strong>Linking to Specific Sections:</strong> If you want to link to a specific person&#8217;s notes you can use &#8216;anchor&#8217; notation as illustrated here for <a class="reference internal" href="#jennifer-brisson"><em>Jennifer Brisson</em></a>:</li>
</ol>
<div class="last highlight-none"><div class="highlight"><pre>http://movingthelamppost.com/blog/html/2013/06/14/arthropod_genomics_symposium_2013____notre_dame.html#jennifer-brisson
</pre></div>
</div>
</div>
<div class="section" id="general-session">
<h2>General Session</h2>
<div class="section" id="dan-lawson-ebi">
<span id="dan-lawson"></span><h3>Dan Lawson (EBI)</h3>
<ul class="simple">
<li>Federated curation and hosting of genomes plus standardized best practices and data formats needed for #i5k</li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="steven-fringy-richards-baylor-college-of-medicine">
<span id="steven-richards"></span><h3>Steven &#8220;Fringy&#8221; Richards (Baylor College of Medicine)</h3>
<ul class="simple">
<li>ND50 10kb basic bar to reach for good annotation through Maker</li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="terence-murphy-ncbi">
<span id="terence-murphy"></span><h3>Terence Murphy (NCBI)</h3>
<p><em>How to annotate 5000 genomes: lessons from NCBI genome annotation pipeline</em></p>
<hr class="docutils" />
</div>
<div class="section" id="monica-munoz-torres-lawrence-berkeley-national-laboratory">
<span id="monica-munoz-torres"></span><h3>Monica Munoz-Torres (Lawrence Berkeley National Laboratory)</h3>
<p><em>Web Apollo web based annotation editing platform</em></p>
<ul class="simple">
<li>first realtime genome collaborative annotation editor on the web</li>
<li>MANUAL annotation is NECESSARY not just desirable: without it many downstream experiments will be poisoned</li>
<li>democratization of genome assembly and annotation require a new model of collaborative annotation</li>
</ul>
<div id="more"> </div></div>
</div>
<hr class="docutils" />
<div class="section" id="emerging-genomes">
<h2>Emerging Genomes</h2>
<div class="section" id="amy-toth">
<span id="id1"></span><h3>Amy Toth</h3>
<p><em>Genome of primitively social wasp &#8220;Polistes dominula&#8221;: insights/opportunity for understanding genomic basis of eusociality</em></p>
<ul>
<li><p class="first">low repetitive elements</p>
</li>
<li><p class="first">even lower TE content</p>
</li>
<li><dl class="first docutils">
<dt>genomic suspects for social evolution</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>genetic toolkits &#8211; conserved gene sets</dt>
<dd><ul class="first last">
<li><p class="first">evo-devo approach &#8211;comparative</p>
</li>
<li><dl class="first docutils">
<dt>conservation of Hox genes etc</dt>
<dd><ul class="first last simple">
<li>cis regulatory changes very important as changes in expression vs gene function important in evolution/speciation</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>transcriptomic look at</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>foraging</dt>
<dd><ul class="first last simple">
<li>foraging transcriptome profiles agree with apis mellifera</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>dominance</dt>
<dd><ul class="first last simple">
<li>compared transcriptomes of brains in dominant vs submissive individuals</li>
<li>found correlations of aggressive behavior in fruit flies and even mammals</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>epigenetics</dt>
<dd><ul class="first last simple">
<li>controls wild diff in phenotypic models &#8211; same genome gives VERY diff morphologies</li>
<li>cast related diff in DNA methylation are present</li>
<li>treatment with methyl inhibitor causes more worker like phenotypes</li>
<li>however dna meth not likely to be related to cast diff through diff expression, but maybe through splicing control</li>
<li>Dnmt3 (usually needed for DNA meth) not found yet in this genome</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">novel genes</p>
</li>
</ul>
</dd>
</dl>
</li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="dave-denlinger-ohio-state">
<span id="dave-denlinger"></span><h3>Dave Denlinger (Ohio State)</h3>
<p><em>Sequencing the Antarctic Midge, Belgica antarctica: the smallest insect genome</em></p>
<ul>
<li><dl class="first docutils">
<dt>poly extremophile</dt>
<dd><ul class="first last simple">
<li>cold &#8211; frozen solid 8-9 months a year</li>
<li>70% loss of body water</li>
<li>salinity</li>
<li>pH</li>
<li>UV EXPOSURE</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Heat shock proteins on almost constitutively</p>
</li>
<li><p class="first">Dessication 30% helps them survive low temps</p>
</li>
<li><p class="first">initial genome size of ~ 85 Mb?! &#8211; then confirmed</p>
</li>
<li><p class="first">13,500 protein encoding genes anyway</p>
</li>
<li><dl class="first docutils">
<dt>uniqueness of the genome:</dt>
<dd><ul class="first last">
<li><p class="first">small size &#8211; probly due to extremophile life</p>
</li>
<li><p class="first">consistently small introns</p>
</li>
<li><p class="first">few repetitive elements</p>
</li>
<li><p class="first">few TEs &#8211; usually old and probably inactive</p>
</li>
<li><p class="first">47% GC</p>
</li>
<li><dl class="first docutils">
<dt>function enrich:</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>up</dt>
<dd><ul class="first last simple">
<li>membrane</li>
<li>ATP binding</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>dwn</dt>
<dd><ul class="first last simple">
<li>oderant binding</li>
<li>oderant receptors</li>
<li>sensory perception of smell in general</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>pathway of dessication protection</dt>
<dd><ul class="first last simple">
<li>up regulation of autophagy</li>
<li>inhibit apoptosis</li>
<li>metabolism genes shutdown</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">lead to focus on aquaporins as related to dehydration responses in diff tissues</p>
</li>
<li><dl class="first docutils">
<dt>How does clock work in these extreme light/dark cycles</dt>
<dd><ul class="first last simple">
<li>clock genes not correlated with light cycle at all</li>
<li>locomotor activity shows no diurnal bias</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">genes still there and still expressed but the cyclic nature of transcription is NOT observed</p>
</li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="virpi-ahola">
<span id="id2"></span><h3>Virpi Ahola</h3>
<p><em>melitaea cinxia genome</em></p>
</div>
</div>
<hr class="docutils" />
<div class="section" id="epigenetics">
<h2>Epigenetics</h2>
<div class="section" id="jennifer-brisson-u-of-neb-lincoln">
<span id="jennifer-brisson"></span><h3>Jennifer Brisson (U of Neb - Lincoln)</h3>
<p><em>Linking DNA methylation to phenotype in the pea aphid</em></p>
<ul>
<li><p class="first">meth found in gene bodies primarily in exons</p>
</li>
<li><p class="first">Methylation associated with higher expression (only at global level?)</p>
</li>
<li><dl class="first docutils">
<dt>differences bt morphs:</dt>
<dd><ul class="first last simple">
<li>methyl patterns plainly morph specific by PCA 2D</li>
<li>intron specific CHH methylation flat in wingless/asex but very pronounced in Sexual morphs</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>function of intergenic DNA methylation:</dt>
<dd><ul class="first last">
<li><p class="first">skipped exons should have higher methylations bc CTCF cant bind the exon to slow down the machinerey to allow that exon to be recognized as NOT intron</p>
</li>
<li><p class="first">this IS detected by the speaker</p>
</li>
<li><dl class="first docutils">
<dt>role for DNA methylation in dcerning paralogs?</dt>
<dd><ul class="first last simple">
<li>example used is RNAi genes (Dicer etc)</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="micheal-goodisman-ga-tech">
<span id="micheal-goodisman"></span><h3>Micheal Goodisman (GA Tech)</h3>
<p><em>Function of DNA methylation in insects</em></p>
<ul>
<li><dl class="first docutils">
<dt>DNA methylation lost in some insects</dt>
<dd><ul class="first last simple">
<li>NO METHEYL IN DIPTERA?!  <em>Did I know that?</em></li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">What is the function of methyl in insects?</p>
</li>
<li><p class="first"><strong>Note:</strong> look up CpG_o/e</p>
</li>
<li><dl class="first docutils">
<dt>in Apis and Pea Aphid:</dt>
<dd><ul class="first last simple">
<li>uniform expression between conditions = high methylation</li>
<li>differential expression bt condition = low methylation</li>
<li>says its a strong repeating pattern</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>fire ant diploid vs haploid males:</dt>
<dd><ul class="first last simple">
<li>increased gene expression tends to track with increased DNAmeth</li>
<li>as variation in gene expression increases DNAmeth decreases</li>
<li>most DNAmeth seen in Haploid males... (not sure I got that right)</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Conservation of methylation and other epigenetic data (honey bee vs fire ant and Dmel [no DNAmeth but yes histone modification])</dt>
<dd><ul class="first last">
<li><p class="first">over all: within ants, correlatiuon of genes with DNAmeth are very high (r ~&gt;0.7)</p>
</li>
<li><p class="first">same in bees</p>
</li>
<li><p class="first">between ants/bees: still relatively high correlation (r ~ 0.6)</p>
</li>
<li><dl class="first docutils">
<dt>are DNAmeth in genes in ants/bees correleated with specific histone mods in Dmel:</dt>
<dd><ul class="first last simple">
<li>Answer is YES</li>
<li>no time to document which hist mods he showed</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Summary:</dt>
<dd><ul class="first last simple">
<li>phenoype specific genes are unmethylated</li>
<li>ubiquitously genes are methylated</li>
<li>DNAmeth associated with ploidy</li>
<li>Patterns of DNAmeth conserved within and bewtween species</li>
<li>DNAmeth tends to be associated with specific histone mods</li>
<li>DNAmeth tied to alternative splicing</li>
</ul>
</dd>
</dl>
</li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="greg-hunt-purdue">
<span id="greg-hunt"></span><h3>Greg Hunt (Purdue)</h3>
<p><em>Parent of origin effects in gene expression in honey bees</em></p>
<ul class="simple">
<li>hybrids between african and european bees sting intermediate of the parents (but if the father is African hybrids are more similar to african bees?)  <strong>NOT SURE I GOT THAT RIGHT</strong></li>
<li>Used multiple different mapping methods but didnt say how &#8211; <em>DONT LIKE THAT</em></li>
<li>Truethfully I dunno if I can talk about the rest because I can&#8217;t tell if his biases are real or due to not using the same mapper in all conditions...</li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="susan-weiner-iowa-state-univ">
<span id="susan-weiner"></span><h3>Susan Weiner (Iowa State Univ)</h3>
<p><em>DNA methylation in the primitively social wasp *Polistes dominulus*</em></p>
<ul>
<li><p class="first">primitavely eusocial = not morpholagical difference (queen/works look the same)</p>
</li>
<li><p class="first">independent origin of socialality? (Did i hear that right?)</p>
</li>
<li><p class="first">Polistes has way more DNAmeth than all other insects looked at</p>
</li>
<li><dl class="first docutils">
<dt>is there a bias to sites DNAmeth in caste?</dt>
<dd><ul class="first last simple">
<li>yes.... <strong>but</strong> PCA 2D was not <strong>THAT</strong> impressive</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Zebularine treatment (<em>inhibits DNAmeth</em>) in multiple replications and variations tend to cause individuals make them more worker-like (opposite as in honey bees)</p>
</li>
<li><p class="first">DNMT3 has not been found yet (DNAmethyltransferase i think)</p>
</li>
<li><p class="first">DNMT1 and 2 are there</p>
</li>
<li><dl class="first docutils">
<dt>question asked: was zebularine validated that it is working uniformly</dt>
<dd><ul class="first last simple">
<li>answer: working on that</li>
</ul>
</dd>
</dl>
</li>
</ul>
</div>
</div>
<hr class="docutils" />
<div class="section" id="comparative-genomics">
<h2>Comparative Genomics</h2>
<div class="section" id="rob-waterhouse-univ-geneva-mit">
<span id="rob-waterhouse"></span><h3>Rob Waterhouse (Univ Geneva/MIT)</h3>
<p><em>Orthology-based genome annotation and interpretation</em></p>
<ul>
<li><p class="first">where we have come since Dmel in early 2000s</p>
</li>
<li><p class="first">what is it you want from your genome BEFORE you start to make sure that you have the quality you need</p>
</li>
<li><dl class="first docutils">
<dt>once your annotation is done:</dt>
<dd><ul class="first last simple">
<li>how do you asses the completness of your annotation?</li>
<li>once complete, orthology becomes useful tool</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">orthoDB: most comprehensive source for orthology amoung arthropods (&gt;57 species)</p>
</li>
<li><dl class="first docutils">
<dt>Assesing completion:</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>BUSCOs -&gt; <strong>needs definition later</strong></dt>
<dd><ul class="first last">
<li><p class="first">orthos with single copy ortho in 90% of other species</p>
</li>
<li><dl class="first docutils">
<dt>expectation:</dt>
<dd><ul class="first last simple">
<li>most should be found in your assembly</li>
<li>most should be single copy</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>implementation:</dt>
<dd><ul class="first last">
<li><p class="first">blast</p>
</li>
<li><dl class="first docutils">
<dt>then:</dt>
<dd><ul class="first last simple">
<li>best BLAST regions</li>
<li>next-best BLAST</li>
<li>homology based gene predictions</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>example:</dt>
<dd><ul class="first last">
<li><p class="first">15 Mosquito species</p>
</li>
<li><p class="first">fewer than 10 BUSCOs in most Mosq species</p>
</li>
<li><dl class="first docutils">
<dt>Aedes:</dt>
<dd><ul class="first last simple">
<li>16 missing</li>
<li>244 multi copy</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Culex:</dt>
<dd><ul class="first last simple">
<li>47 missing</li>
<li>126 multi copy</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>also look at the length differentials between orthologs</dt>
<dd><ul class="first last simple">
<li>helps see whether the fragmentation of assembly is affecting your assembly/gene models</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>mapping: new feature for orthoDB</dt>
<dd><ul class="first last simple">
<li>allows new genomes to be mapped to current frozen set of orthologs</li>
<li>private interface if required</li>
<li>also allows you to predict how many of the BUSCOs are missing to decide whether to make current gene set public</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>orthology to infer gene function:</dt>
<dd><ul class="first last">
<li><p class="first">functional traits</p>
</li>
<li><p class="first">evolutionary traits</p>
</li>
<li><dl class="first docutils">
<dt><strong>CAVEAT EMPTOR</strong>:</dt>
<dd><ul class="first last simple">
<li>orthology does not strictly define function</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Added a <strong>BUNCH</strong> of cool extra information including synteny blocks and relative evolutionary rate</p>
</li>
<li><dl class="first docutils">
<dt><strong>if:</strong></dt>
<dd><ul class="first last simple">
<li>single copy</li>
<li>kept in most species</li>
<li>slow evolutionary rate</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt><strong>then:</strong></dt>
<dd><ul class="first last simple">
<li>functional assumptions are warranted</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="chris-winchell-uc-berkley">
<span id="chris-winchell"></span><h3>Chris Winchell (UC Berkley)</h3>
<p><em>Genome and germline of emerging genome of crustacean __Parhyale hawaiensis__</em></p>
<ul>
<li><dl class="first docutils">
<dt>BAC library:</dt>
<dd><ul class="first last simple">
<li>129 024 clones</li>
<li>median insert ~140 kb</li>
<li>aprox coverage 5X</li>
<li>70 have been seq&#8217;d: 53 genes found</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>germline <em>REPLACEMENT?!</em></dt>
<dd><ul class="first last simple">
<li>you can ablate the original developing germline cells and they will MAKE MORE!! (convert somatic cells to germline?)</li>
<li>very elegant transgenic demonstration of where the new germline cells are originating in the blastoderm</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first"><strong>not</strong> a &#8220;normal&#8221; event, but may be a failsafe for rare germline loss of function in development</p>
</li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="michael-brewer-uc-berkley">
<span id="michael-brewer"></span><h3>Michael Brewer (UC Berkley)</h3>
<p><em>Evolutionary transcriptomics associated with developmental color switching in an adaptive radiation of Hawaiian Tetragnatha spiders (Araneae: Tetragnathidae)</em></p>
<ul>
<li><p class="first">orb weavers that don&#8217;t weave orbs anymore and adopt purely active hunting lifestyle</p>
</li>
<li><dl class="first docutils">
<dt>2 ecomorph:</dt>
<dd><ul class="first last simple">
<li>green</li>
<li>maroon</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">green individuals switch to maroon and their diet changes upon maturity</p>
</li>
<li><p class="first">they can control whether/when they shift?</p>
</li>
<li><p class="first">can date the species due to isolation on the islands</p>
</li>
<li><dl class="first docutils">
<dt>METHOD:</dt>
<dd><ul class="first last simple">
<li>RNAseq</li>
<li>trinity assembly</li>
<li>annotate by blast</li>
<li>trinity MM ORF</li>
<li>RecipBestBlast</li>
<li>Something else: slide change...</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">... got caught up in other things ...</p>
</li>
<li><p class="first"><strong>He has many centipede/millipede RNA-seq libraries but they are being neglected in the community: if you are interested in the results from these libraries please contact him</strong></p>
</li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="igor-sharakhov-virgina-tech">
<span id="igor-sharakhov"></span><span id="pirna"></span><h3>Igor Sharakhov (Virgina Tech)</h3>
<p><em>Ornganization and evolution of piRNAs in _Anopheles gambiae_</em></p>
<ul>
<li><p class="first">in aedes piRNAs bind mostly to GENES not transposable elements</p>
</li>
<li><dl class="first docutils">
<dt>tissue subcell localization:</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>Dmel:</dt>
<dd><ul class="first last simple">
<li>diff localization for each piRNA protein</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>composition of piRNA in Ag</dt>
<dd><ul class="first last">
<li><p class="first">aedes piRNA clusters overlap GENES</p>
</li>
<li><p class="first">most piRNAs in Ag seem to have signiture of Ping Pong amplification</p>
</li>
<li><dl class="first docutils">
<dt>Localization in Ag seems mostly localized with transposable elements not genes</dt>
<dd><ul class="first last simple">
<li>but on X many map with genes...</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Genomic chromosomal localization of piRNA in Ag</dt>
<dd><ul class="first last">
<li><p class="first">they have Abs for PIWI1</p>
</li>
<li><p class="first">PIWI1 localization on chromo looks more like Dmel PIWI</p>
</li>
<li><p class="first">PIWI2 (aub) not succeeded</p>
</li>
<li><dl class="first docutils">
<dt>aedes: most piRNA clusters NOT near centromeres like Dmel</dt>
<dd><ul class="first last simple">
<li>mostly in euchromatin</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Ag: most piRNA peaks in centromereic but many outside of centromeric</dt>
<dd><ul class="first last simple">
<li>however still mostly in heterochromatic</li>
<li>M and S forms look pretty similar in localization of clusters</li>
<li>again in X chromosome many more clusters mapping in euchromatin</li>
<li>M and S forms NOT as similar in X chromosome</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">aedes piRNAs clusters as much as 20% of genome</p>
</li>
</ul>
</dd>
</dl>
</li>
</ul>
</div>
</div>
<hr class="docutils" />
<div class="section" id="systems-biology-population-genomics">
<h2>Systems Biology/Population Genomics</h2>
<div class="section" id="andy-clark-cornell">
<span id="andy-clark"></span><h3>Andy Clark (Cornell)</h3>
<p><em>Population genomic and metabolic inference from a global diversity reference panel of _Drosophila melanogaster_</em></p>
<ul>
<li><p class="first">using flies as model for diabetes: insulin signaling and TOR</p>
</li>
<li><dl class="first docutils">
<dt>lines isofemale lines (&gt;90 lines):</dt>
<dd><ul class="first last simple">
<li>Ithica</li>
<li>Netherlands</li>
<li>Beijing</li>
<li>Zimbabwe</li>
<li>Tasmania</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">significant metabolic phenotypic variation between the lines</p>
</li>
<li><dl class="first docutils">
<dt>scored for MANY types of info:</dt>
<dd><ul class="first last simple">
<li>enzyme activity</li>
<li>fat storage</li>
<li>more</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>modeling objective:</dt>
<dd><ul class="first last simple">
<li>fit models predicting endpoints like total fat storage from genome</li>
<li>assess sensitivity of component of models</li>
<li>explain genetic architecture</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>modeling:</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>multiple regression</dt>
<dd><ul class="first last simple">
<li>observed vs expected: r^2 = 0.23</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Flux balance Analysis (FBA)</p>
</li>
<li><dl class="first docutils">
<dt>Structural Equation MOodels (SEM)</dt>
<dd><ul class="first last simple">
<li>obsvd vs exp: r^2 = 0.44</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Bayesian hierarchical model</p>
</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">3 cam 120 frames/sec linked in 3D</p>
</li>
<li><p class="first">perturbations by flying through magnetic field with magnet glued to their back</p>
</li>
<li><p class="first">recovery from perturbation begins in a single wing beat</p>
</li>
<li><p class="first"><strong>VERY COOL VIDEOS</strong></p>
</li>
<li><p class="first">Data suggesting that the Drosophila melanogaster population variance may mirror the human data supporting the idea that we brought Dmel with us as we left Africa and they follow similar variance constrictions as they followed the particular human population as they split</p>
</li>
<li><p class="first">New descriptions of hybrid dysgenesis through cross population mating allowing exploration of other mobile elements and their effects (similar to P-element)</p>
</li>
<li><p class="first">Wolbachia data suggests a single invasion of Dmel</p>
</li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="peter-cherbas-indiana-univ">
<span id="peter-cherbas"></span><h3>Peter Cherbas (Indiana Univ)</h3>
<p><em>modENCODE and the potential of systems biology in _Drosophila_</em></p>
<ul>
<li><p class="first">1938 new transcribed regions (not linked to any other gene model)</p>
</li>
<li><p class="first">almost every gene included new 5&#8217; or 3&#8217; UTR...</p>
</li>
<li><p class="first">promoter switching linked to certain lines where Broad in activated or deactivated by ecdysone</p>
</li>
<li><p class="first">environmental perturbations reveal totally new genes</p>
</li>
<li><p class="first">neural specific expressed isoforms have LONG 3&#8217; extensions</p>
</li>
<li><p class="first">18 loci where there is almost exactly overlapped (sense/antisense) transcription (seems protein coding too but I am not sure)</p>
</li>
<li><p class="first">in general: the <strong>least certain</strong> part of any annotation is the non-coding regions</p>
</li>
<li><p class="first"><strong>RESOURCE:</strong> MANY cell lines used to produced transcriptomes for public use</p>
</li>
<li><dl class="first docutils">
<dt>cell lines unique and represents different &#8220;test tubes&#8221; for biological questions</dt>
<dd><ul class="first last simple">
<li>unique complement of transcription factors etc</li>
<li>sufficiently differentiated to be homologous in behavior and components</li>
</ul>
</dd>
</dl>
</li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="jeremy-lynch-univ-illinois-chicago">
<span id="jeremy-lynch"></span><h3>Jeremy Lynch (Univ Illinois Chicago)</h3>
<p><em>Global analysis of the dorsal-ventral patterning regulatory network in the wasp _Nasonia vitripennis_ using quantitative transcriptomics</em></p>
<ul>
<li><p class="first">opening statement: <strong>DON&#8217;T BE AFRAID OF THE COMMAND LINE!!!!</strong></p>
</li>
<li><dl class="first docutils">
<dt>Nasonia embryo layed out very similar to that of Dmel</dt>
<dd><ul class="first last simple">
<li>morphology and gene expression</li>
<li>most likely convergent evolution</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Nasonia patterning systems more dynamic</p>
</li>
<li><p class="first">BMP provides most patterning information</p>
</li>
<li><p class="first">also once cells start migrating in the embryo there are major divergences from Dmel as well</p>
</li>
<li><dl class="first docutils">
<dt>does Nasonia use the same genes as Dmel?</dt>
<dd><ul class="first last simple">
<li>knock out BMP: lose dorsal gene expression</li>
<li>KO toll: lose ventral gene expression</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">used tuxedo protocol</p>
</li>
<li><dl class="first docutils">
<dt>also used trinity to see what happened de novo followed by DESeq</dt>
<dd><ul class="first last simple">
<li>works mostly ok but has a lot of nonsense as well</li>
</ul>
</dd>
</dl>
</li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="sara-mitchell-harvard-public-health">
<span id="sarah-mitchell"></span><h3>Sara Mitchell (Harvard Public Health)</h3>
<p><em>Genetic pathways induced by mating have a key role in the reproductive biology of _Anopheles gambiae_</em></p>
<ul class="simple">
<li>many discoveries regarding elements that are required for mating plug formation as well as signals effecting the female post mating changes (sperm is not required)</li>
<li>found changes in the midgut after mating</li>
<li>looking at early middle and late matting reponses (3, 12, 24 h)</li>
<li>mostly induction of genes in mated vs virgins</li>
<li>wow.  LOTS of information very fast!  Not keeping up with typing very well!</li>
</ul>
</div>
</div>
<hr class="docutils" />
<div class="section" id="ecological-genomics">
<h2>Ecological Genomics</h2>
<div class="section" id="leslie-vosshall-the-rockefeller-univ">
<span id="leslie-vosshall"></span><h3>Leslie Vosshall (The Rockefeller Univ)</h3>
<p><em>Dissecting mosquito host-seeking behavior through loss-of-function genetics</em></p>
<ul>
<li><dl class="first docutils">
<dt>two clases of insect oder recepters</dt>
<dd><ul class="first last simple">
<li>OR</li>
<li>IR</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">each insect has different numbers of these receptors</p>
</li>
<li><p class="first">&#8220;Aedes has a &#8216;horrifically&#8217; enormous genome&#8221;</p>
</li>
<li><p class="first">KO OR-coeffector effectively removes function of all ORs</p>
</li>
<li><p class="first">used Zinc Fingers to do the KO</p>
</li>
<li><dl class="first docutils">
<dt>Behavior assay:</dt>
<dd><ul class="first last simple">
<li>tube with a human scent at the end, see if the females choose the right tube.</li>
<li>ORCO-/- cant see human scent alone but can see if CO2 is added</li>
<li>IRs must also be involved</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">given choice between human or guinee pig the double mutants are much less good at choosing the human</p>
</li>
<li><p class="first">ORco not needed to choose human over nothing but are important for choosing human vs non-human</p>
</li>
<li><dl class="first docutils">
<dt>Vienette 2:</dt>
<dd><ul class="first last simple">
<li>many cues moisture, heat, smell, C02 etc</li>
<li>moving on to gustatory receptors (GR)</li>
<li>GR3 knocks out all???</li>
<li>GR3-/- blind to CO2</li>
<li>CO2 + Heat causes biting</li>
<li>both needed</li>
<li>GR3- means no CO2 signal</li>
<li>mosqs need more than one signal to trigger feeding</li>
<li>mutants severly impaired in lab but in semi field conditions much LESS impaired</li>
</ul>
</dd>
</dl>
</li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="brian-lazzaro-cornell-univ">
<span id="brian-lazzaro"></span><h3>Brian Lazzaro (Cornell Univ)</h3>
<p><em>Complexity in the function and evolution of insect immunity</em></p>
<ul>
<li><dl class="first docutils">
<dt>how does diet affect immunity?</dt>
<dd><ul class="first last">
<li><p class="first">it does btw...</p>
</li>
<li><dl class="first docutils">
<dt>?? Could it not be that the more sugar you add to the food the more you make it easier for the bacteria? (you feed them more?)</dt>
<dd><ul class="first last simple">
<li>there seems to be some link to genotype</li>
<li>could be that the genetic component is how well the fly converts sugar to less usable substrates</li>
<li>?? could you use a sugar that the flies can use but is not easy for the bacteria to use?</li>
<li>?? a bactaeria that doesnt eat glucose?</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">4th PC out of 5 deminsions in PCA?</p>
</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">how does mating effect immunity?</p>
</li>
</ul>
</div>
</div>
<hr class="docutils" />
<div class="section" id="my-own-personal-remember-this-list">
<h2>My Own Personal &#8220;Remember This&#8221; List</h2>
<ul>
<li><dl class="first docutils">
<dt>GMAP to map de novo assembled transcriptomes to the genome</dt>
<dd><ul class="first last simple">
<li>plan to convert to GTF and merge with cufflinks GTF using cuffmerge to help define UTRs better for CRM/CRE discovery in my Mosquito species</li>
</ul>
</dd>
</dl>
</li>
</ul>
</div>
</div>

    <div class="postmeta">
        <div class="author">
            <span>Posted by W Augustine Dunn</span>
        </div>
        <div class="categories">
            <span>
                Filed under:
                <a href="../../../categories/conferences.html">Conferences</a></span>
        </div>
        <div class="tags">
            <span>
                Tags:
                <a href="../../../tags/arthgen2013.html">ArthGen2013</a>, <a href="../../../tags/genomics.html">genomics</a>, <a href="../../../tags/arthropods.html">arthropods</a>, <a href="../../../tags/transcriptomics.html">transcriptomics</a>, <a href="../../../tags/twitter.html">twitter</a>, <a href="../../../tags/science_twitter.html">science@twitter</a>, <a href="../../../tags/live_blogging.html">live blogging</a></span>
        </div>
        </div><ul class="related clearfix">
            <li class="left"> &laquo; <a href="../29/speak__for_the_cup_of_hemlock_is_not_yet_on_your_lips.html">Speak, for the Cup of Hemlock is not yet on Your Lips</a></li>
            <li class="right"><a href="../../05/18/fundamentals_of_gene_expression__the__central_dogma__of_molecular_biology.html">Fundamentals of Gene Expression: the &#8216;Central Dogma&#8217; of Molecular Biology</a> &raquo; </li>
        </ul><div id="disqus_thread"></div><script type="text/javascript">    var disqus_shortname = "movingthelamppost";    var disqus_identifier = "2013/06/14/arthropod_genomics_symposium_2013____notre_dame";    disqus_thread();</script><noscript>Please enable JavaScript to view the    <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript></article><aside class="sidebar"><section>
<div class="widget" id="sub_by_email">
    <h1>Subscribe</h1>
    <a href='http://www.feedly.com/home#subscription/feed%2Fhttp%3A%2F%2Ffeeds.feedburner.com%2FMovingTheLamppost'  target='blank'><img id='feedlyFollow' src='http://s3.feedly.com/img/follows/feedly-follow-rectangle-volume-medium_2x.png' alt='follow us in feedly' width='71' height='28'></a>
    <br>
    <a href="http://feeds.feedburner.com/MovingTheLamppost">via feedburner</a>
    <br>
    <a href="http://feedburner.google.com/fb/a/mailverify?uri=MovingTheLamppost&amp;loc=en_US">via email</a>
    
</div>

</section><section><div class="widget" id="searchbox">
    <h1>Search</h1>
    <form action="../../../search.html" method="get">
        <input type="text" name="q" />
        <button type="submit"><span class="webfont">L</span></button>
    </form>
</div></section><section><div class="widget">
    <h1>Categories</h1>
    <ul><li><a href="../../../categories/background_science.html">Background Science</a> (1)</li><li><a href="../../../categories/conferences.html">Conferences</a> (1)</li><li><a href="../../../categories/journal_club.html">Journal Club</a> (1)</li><li><a href="../../../categories/journal_club__shorts_.html">Journal Club (shorts)</a> (1)</li><li><a href="../../../categories/my_code.html">My Code</a> (2)</li><li><a href="../../../categories/my_dissertation.html">My Dissertation</a> (3)</li><li><a href="../../../categories/my_presentations.html">My Presentations</a> (1)</li><li><a href="../../../categories/my_research.html">My Research</a> (4)</li></ul>
</div></section><section><div class="widget">
    <h1>Tags Cloud</h1>
      <a href="../../../tags/amino_acids.html" style="font-size: 8pt">amino acids</a>&nbsp;&nbsp;
      <a href="../../../tags/anopheles_stephensi.html" style="font-size: 8pt">Anopheles stephensi</a>&nbsp;&nbsp;
      <a href="../../../tags/arthgen2013.html" style="font-size: 8pt">ArthGen2013</a>&nbsp;&nbsp;
      <a href="../../../tags/arthropods.html" style="font-size: 8pt">arthropods</a>&nbsp;&nbsp;
      <a href="../../../tags/asif.html" style="font-size: 8pt">Asif</a>&nbsp;&nbsp;
      <a href="../../../tags/asif_mohiuddin.html" style="font-size: 8pt">Asif Mohiuddin</a>&nbsp;&nbsp;
      <a href="../../../tags/background.html" style="font-size: 10pt">background</a>&nbsp;&nbsp;
      <a href="../../../tags/bangladesh.html" style="font-size: 8pt">Bangladesh</a>&nbsp;&nbsp;
      <a href="../../../tags/bioinformatics.html" style="font-size: 8pt">bioinformatics</a>&nbsp;&nbsp;
      <a href="../../../tags/blacktie.html" style="font-size: 12pt">blacktie</a>&nbsp;&nbsp;
      <a href="../../../tags/blogging.html" style="font-size: 8pt">blogging</a>&nbsp;&nbsp;
      <a href="../../../tags/blood_feeding.html" style="font-size: 8pt">blood feeding</a>&nbsp;&nbsp;
      <a href="../../../tags/cell.html" style="font-size: 8pt">cell</a>&nbsp;&nbsp;
      <a href="../../../tags/conservation.html" style="font-size: 8pt">conservation</a>&nbsp;&nbsp;
      <a href="../../../tags/cuffdiff.html" style="font-size: 10pt">cuffdiff</a>&nbsp;&nbsp;
      <a href="../../../tags/cufflinks.html" style="font-size: 10pt">cufflinks</a>&nbsp;&nbsp;
      <a href="../../../tags/cuffmerge.html" style="font-size: 10pt">cuffmerge</a>&nbsp;&nbsp;
      <a href="../../../tags/cummerbund.html" style="font-size: 10pt">cummeRbund</a>&nbsp;&nbsp;
      <a href="../../../tags/de_novo_transcriptome_assembly.html" style="font-size: 8pt">de novo transcriptome assembly</a>&nbsp;&nbsp;
      <a href="../../../tags/dissertation__chapter_one__background_.html" style="font-size: 10pt">Dissertation: Chapter One (Background)</a>&nbsp;&nbsp;
      <a href="../../../tags/dna.html" style="font-size: 8pt">DNA</a>&nbsp;&nbsp;
      <a href="../../../tags/evolution.html" style="font-size: 8pt">evolution</a>&nbsp;&nbsp;
      <a href="../../../tags/free_expression.html" style="font-size: 8pt">free expression</a>&nbsp;&nbsp;
      <a href="../../../tags/freedom.html" style="font-size: 8pt">freedom</a>&nbsp;&nbsp;
      <a href="../../../tags/gene_drive.html" style="font-size: 8pt">gene drive</a>&nbsp;&nbsp;
      <a href="../../../tags/gene_expression.html" style="font-size: 10pt">gene expression</a>&nbsp;&nbsp;
      <a href="../../../tags/gene_regulation.html" style="font-size: 8pt">gene regulation</a>&nbsp;&nbsp;
      <a href="../../../tags/genomics.html" style="font-size: 10pt">genomics</a>&nbsp;&nbsp;
      <a href="../../../tags/gmm.html" style="font-size: 10pt">GMM</a>&nbsp;&nbsp;
      <a href="../../../tags/gmo.html" style="font-size: 10pt">GMO</a>&nbsp;&nbsp;
      <a href="../../../tags/government.html" style="font-size: 8pt">government</a>&nbsp;&nbsp;
      <a href="../../../tags/gripes.html" style="font-size: 8pt">gripes</a>&nbsp;&nbsp;
      <a href="../../../tags/human_rights.html" style="font-size: 8pt">human rights</a>&nbsp;&nbsp;
      <a href="../../../tags/injustice.html" style="font-size: 8pt">injustice</a>&nbsp;&nbsp;
      <a href="../../../tags/intolerance.html" style="font-size: 8pt">intolerance</a>&nbsp;&nbsp;
      <a href="../../../tags/ipython.html" style="font-size: 10pt">ipython</a>&nbsp;&nbsp;
      <a href="../../../tags/live_blogging.html" style="font-size: 8pt">live blogging</a>&nbsp;&nbsp;
      <a href="../../../tags/midgut.html" style="font-size: 8pt">midgut</a>&nbsp;&nbsp;
      <a href="../../../tags/mohiuddin.html" style="font-size: 8pt">Mohiuddin</a>&nbsp;&nbsp;
      <a href="../../../tags/mosquitoes.html" style="font-size: 10pt">mosquitoes</a>&nbsp;&nbsp;
      <a href="../../../tags/my_dissertation.html" style="font-size: 8pt">My Dissertation</a>&nbsp;&nbsp;
      <a href="../../../tags/my_research.html" style="font-size: 8pt">My Research</a>&nbsp;&nbsp;
      <a href="../../../tags/my_research.html" style="font-size: 8pt">my research</a>&nbsp;&nbsp;
      <a href="../../../tags/namespace.html" style="font-size: 8pt">namespace</a>&nbsp;&nbsp;
      <a href="../../../tags/new_release.html" style="font-size: 10pt">new release</a>&nbsp;&nbsp;
      <a href="../../../tags/ngs.html" style="font-size: 12pt">NGS</a>&nbsp;&nbsp;
      <a href="../../../tags/non_bloodfed.html" style="font-size: 8pt">non-bloodfed</a>&nbsp;&nbsp;
      <a href="../../../tags/nucleic_acids.html" style="font-size: 8pt">nucleic acids</a>&nbsp;&nbsp;
      <a href="../../../tags/nucleus.html" style="font-size: 8pt">nucleus</a>&nbsp;&nbsp;
      <a href="../../../tags/object_oriented_programming.html" style="font-size: 8pt">object oriented programming</a>&nbsp;&nbsp;
      <a href="../../../tags/open_science.html" style="font-size: 20pt">open science</a>&nbsp;&nbsp;
      <a href="../../../tags/open_source.html" style="font-size: 10pt">open source</a>&nbsp;&nbsp;
      <a href="../../../tags/programming.html" style="font-size: 8pt">programming</a>&nbsp;&nbsp;
      <a href="../../../tags/protein.html" style="font-size: 8pt">protein</a>&nbsp;&nbsp;
      <a href="../../../tags/python.html" style="font-size: 8pt">Python</a>&nbsp;&nbsp;
      <a href="../../../tags/python.html" style="font-size: 12pt">python</a>&nbsp;&nbsp;
      <a href="../../../tags/r.html" style="font-size: 10pt">R</a>&nbsp;&nbsp;
      <a href="../../../tags/research.html" style="font-size: 8pt">research</a>&nbsp;&nbsp;
      <a href="../../../tags/research_blogging.html" style="font-size: 8pt">Research Blogging</a>&nbsp;&nbsp;
      <a href="../../../tags/rna.html" style="font-size: 8pt">RNA</a>&nbsp;&nbsp;
      <a href="../../../tags/rna_seq.html" style="font-size: 17pt">RNA-seq</a>&nbsp;&nbsp;
      <a href="../../../tags/science_twitter.html" style="font-size: 8pt">science@twitter</a>&nbsp;&nbsp;
      <a href="../../../tags/selection.html" style="font-size: 8pt">selection</a>&nbsp;&nbsp;
      <a href="../../../tags/sequence_analysis.html" style="font-size: 8pt">sequence analysis</a>&nbsp;&nbsp;
      <a href="../../../tags/the_central_dogma.html" style="font-size: 8pt">the central dogma</a>&nbsp;&nbsp;
      <a href="../../../tags/tinkerer.html" style="font-size: 8pt">tinkerer</a>&nbsp;&nbsp;
      <a href="../../../tags/tophat.html" style="font-size: 10pt">tophat</a>&nbsp;&nbsp;
      <a href="../../../tags/transcription.html" style="font-size: 8pt">transcription</a>&nbsp;&nbsp;
      <a href="../../../tags/transcriptomics.html" style="font-size: 8pt">transcriptomics</a>&nbsp;&nbsp;
      <a href="../../../tags/transgenic_mosquitoes.html" style="font-size: 10pt">transgenic mosquitoes</a>&nbsp;&nbsp;
      <a href="../../../tags/translation.html" style="font-size: 8pt">translation</a>&nbsp;&nbsp;
      <a href="../../../tags/trinity.html" style="font-size: 8pt">Trinity</a>&nbsp;&nbsp;
      <a href="../../../tags/twitter.html" style="font-size: 8pt">twitter</a>&nbsp;&nbsp;
      <a href="../../../tags/vector_control.html" style="font-size: 10pt">vector control</a>
</div></section></aside></div> <!-- #main --></div> <!-- #main-container -->

        <div class="footer-container"><footer class="wrapper">&copy; Copyright 2013, W Augustine Dunn. Powered by <a href="http://www.tinkerer.me/">Tinkerer</a> and <a href="http://sphinx.pocoo.org/">Sphinx</a>.</footer></div> <!-- footer-container -->

      </div> <!--! end of #container --><!--[if lt IE 7 ]>
          <script src="//ajax.googleapis.com/ajax/libs/chrome-frame/1.0.3/CFInstall.min.js"></script>
          <script>window.attachEvent('onload',function(){CFInstall.check({mode:'overlay'})})</script>
        <![endif]-->
    </body>
</html>