<!DOCTYPE html><!--[if lt IE 7]>      <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]>         <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]>         <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]><!--> <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js"> <!--<![endif]-->
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
        <meta name="description" content="random musings of a molecular biologist turned code jockey in the era of big data and open science.">
        <meta name="viewport" content="width=device-width">
        <title>Genome-Level Analysis of Selective Constraint without Apparent Sequence Conservation &mdash; moving the lamppost</title>
            <link rel="stylesheet" href="../../../_static/normalize.css" type="text/css">
            <link rel="stylesheet" href="../../../_static/main.css" type="text/css">
            <link rel="stylesheet" href="../../../_static/modern5.css" type="text/css">
            <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
            <link rel="stylesheet" href="../../../_static/webfont.css" type="text/css">
        <link rel="shortcut icon" href="../../../_static/tinkerer.ico" /><!-- Load modernizr and JQuery -->
        <script src="../../../_static/vendor/modernizr-2.6.2.min.js"></script>
        <script src="//ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js"></script>
        <script>window.jQuery || document.write('<script src="../../../_static/vendor/jquery-1.8.2.min.js"><\/script>')</script>
        <script src="../../../_static/plugins.js"></script>
        <script src="../../../_static/main.js"></script>
        <link rel="next" title="Blogging my dissertation" href="../15/blogging_my_dissertation.html" /><link rel="alternate" type="application/rss+xml" title="RSS" href="http://feeds.feedburner.com/MovingTheLamppost" /><script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../../',
        VERSION:     '1.1_wad',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script><script type="text/javascript" src="../../../_static/underscore.js"></script><script type="text/javascript" src="../../../_static/doctools.js"></script><script type="text/javascript" src="../../../_static/disqus.js"></script><script type="text/javascript" src="../../../_static/google_analytics.js"></script>

    <script type="text/javascript">
        $(document).ready(function () {
            // Adjusts document height if sidebar is taller
            var documentwrapper = document.getElementsByTagName("article")[0];
            var sidebar = document.getElementsByClassName("sidebar")[0];

            if (sidebar.offsetHeight > documentwrapper.offsetHeight)
            {
                var mainwrapper = document.getElementsByClassName("main")[0];
                documentwrapper.style.minHeight = mainwrapper.offsetHeight + "px";
            }

            // Scroll to content if on small screen
            if (screen.width < 480)
            {
                $(document).scrollTop(document.getElementsByTagName("article")[0].offsetTop - 44);
            }
        });
    </script></head>
    <body>
        <!--[if lt IE 7]>
            <p class="chromeframe">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> or <a href="http://www.google.com/chromeframe/?redirect=true">activate Google Chrome Frame</a> to improve your experience.</p>
        <![endif]-->

      <div id="container"><header>
            <hgroup>
              <h1><a href="../../../index.html">moving the lamppost</a></h1><h2>random musings of a molecular biologist turned code jockey in the era of big data and open science.</h2></hgroup>
          </header>
      <nav>
            <ul><li class="quicklink"><div class="rss">
        <a href="http://feeds.feedburner.com/MovingTheLamppost" title="Subscribe via RSS"><span class="webfont">B</span></a>
    </div></li><li class="main-nav">
                  <a href="../../../index.html">Home</a>
                </li>
              <li class="main-nav">
                  <a href="../../../pages/about.html">About</a>
                </li>
              </ul>
          </nav><div class="main-container"><div class="main wrapper clearfix"><article><ul class="related clearfix">
            <li class="left"></li>
            <li class="right"><a href="../15/blogging_my_dissertation.html">Blogging my dissertation</a> &raquo; </li>
        </ul><div class="timestamp postmeta">
            <span>April 16, 2013</span>
        </div>
    <div class="section" id="genome-level-analysis-of-selective-constraint-without-apparent-sequence-conservation">
<h1>Genome-Level Analysis of Selective Constraint without Apparent Sequence Conservation</h1>
<div class="section" id="full-citation">
<h2>Full Citation</h2>
<p>Vakhrusheva, O. A, Bazykin, G. A, &amp; Kondrashov, A. S. (2013). <em>Genome-level analysis of selective constraint without apparent sequence conservation</em>. <strong>Genome biology and evolution</strong>, 5(3), 532–541. doi:10.1093/gbe/evt023</p>
<p>Links: <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/23418180">PubMed</a> &#8211; <a class="reference external" href="http://gbe.oxfordjournals.org/content/5/3/532.full.pdf+html">Full Text (PDF)</a> &#8211; <a class="reference external" href="http://gbe.oxfordjournals.org/content/5/3/532.full">Full Text (HTML)</a></p>
<p><a class="reference internal" href="#license-information-pertaining-to-the-original-article">License information pertaining to the original article</a></p>
</div>
<div class="section" id="elevator-speech">
<h2>Elevator Speech</h2>
<div class="section" id="academic-significance">
<h3>Academic Significance</h3>
<p>Other studies have demonstrated individual instances of conservation of sequence elements that are no longer <strong>globally alignable</strong> due to extended phylogenetic divergence.  Familiar examples include the demonstration that some transgenic enhancer regions from orthologous genes in distantly related species are able to drive correct gene expression of the host species even though <strong>global alignment</strong> of the enhancer regions was impossible.</p>
<p>However the authors claim that this is the first genome-scale study they know of that supports the phenomena of selective constraint acting on sequence elements absent of identifiable sequence conservation.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">By terms like &#8216;identifiable sequence conservation&#8217;, they really seem to be referring to the ability to generate meaningful <strong>global alignment</strong> since they plainly use <strong>local alignment</strong> to identity the sequences they propose are being selectively retained.</p>
</div>
</div>
<div class="section" id="main-hypotheses">
<h3>Main Hypotheses</h3>
<p><span class="formula"><i>H</i><sub>1</sub></span>
: the presence of a &#8216;conserved&#8217; segment between two less divergent species should <em>overall</em> predict the presence of corresponding conserved segments in a separate pair of similarly divergent species that together have a much larger pair-to-pair divergence distance (see <a class="reference internal" href="#f1"><em>Figure 1</em></a> for graphical representation). <strong>[relevant figures:</strong> <a class="reference internal" href="#f2"><em>Figure 2</em></a> <strong>]</strong></p>
<p><span class="formula"><i>H</i><sub>2</sub></span>
: if <em>empirically validated</em> regulatory elements persist longer than the sequence similarity of the corresponding DNA segments, introns containing a regulatory element in pair 1 should carry a corresponding segment in pair 2 at rates higher than expected by chance. <strong>[relevant figures:</strong> <a class="reference internal" href="#f3"><em>Figure 3</em></a> <strong>]</strong></p>
<p><span class="formula"><i>H</i><sub>3</sub></span>
: presence of a functional element in an orthologous intron in one species pair should correlate with a reduced instance of intron loss when compared to an outgroup serving as a way to infer the presence/absence of the functional element in a distant common ancestor. <strong>[relevant figures:</strong> <a class="reference internal" href="#f4"><em>Figure 4</em></a> <strong>]</strong></p>
<p><span class="formula"><i>H</i><sub>4</sub></span>
: because others have already established that longer introns are more likely to carry a segment of conservation than short introns, a function conserved in all four species should be expected to be associated with larger intron length  when conserved in pair 1 <em>and</em> pair 2, compared with the introns lost in pair 2. <strong>[relevant figures:</strong> <a class="reference internal" href="#f5"><em>Figure 5</em></a> <strong>]</strong></p>
<div id="more"> </div><div class="line-block">
<div class="line"><br /></div>
</div>
<hr class="docutils" />
<div class="line-block">
<div class="line"><br /></div>
</div>
</div>
</div>
<div class="section" id="highlights-by-section">
<h2>Highlights by Section</h2>
<div class="section" id="introduction">
<h3>Introduction</h3>
<ul>
<li><p class="first">Conservation of function is usually perceived to be associated with conservation of sequence (the main discovery process for orthologous genes is centered on sequence similarity even)</p>
</li>
<li><p class="first">As far as I can tell, when they say &#8220;sequence conservation&#8221; they mean successful sequence alignment across genomes</p>
</li>
<li><dl class="first docutils">
<dt>This coupling of function with primary sequence conservation is not strictly required. Examples given:</dt>
<dd><ul class="first last simple">
<li>unalignable proteins can have very similar 3D structures (and even active site locations)</li>
<li>single stranded RNA molecules can form identical structures regardless of alignability</li>
<li>unalignable enhancers from mice can correctly drive gene expression in zebra fish</li>
</ul>
</dd>
</dl>
</li>
</ul>
<div class="line-block">
<div class="line"><br /></div>
</div>
<hr class="docutils" />
<div class="line-block">
<div class="line"><br /></div>
</div>
</div>
<div class="section" id="materials-and-methods">
<h3>Materials and Methods</h3>
<div class="section" id="data">
<h4>Data</h4>
<ul class="simple">
<li>two sets of species quartets were chosen (one dipteran and one vertebrate quartet)</li>
<li>each quartet represents a basic evolutionary structure characterized by two species couplets and an outgroup</li>
<li>each couplet shows less evolutionary divergence between the species than exists between the two couplet groups of each quartet (see <a class="reference internal" href="#f1"><em>Figure 1</em></a>)</li>
<li>orthologs for all species were obtained from the <a class="reference external" href="http://inparanoid.sbc.su.se/cgi-bin/index.cgi">INPARANOID</a> web resource</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>The following was a bit tricky at first reading:</p>
<p class="last">It seems there were two classes of analyses performed. The first with just the four experimental species (what level of &#8216;conservation&#8217; of small local similarity exists between and within the couplets of the two quartets), and one to assess cases of intron loss which required use of the outgroup.</p>
</div>
<ul>
<li><dl class="first docutils">
<dt>ortholog inclusion depended on several criteria including but not limited to:</dt>
<dd><ul class="first last simple">
<li>all four or five species contain a version of the ortholog</li>
<li>no ortholog set was allowed to have any member contain a premature stop codon</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">the resulting ortholog set sizes were:</p>
</li>
</ul>
<table border="1" class="docutils">
<colgroup>
<col width="24%" />
<col width="26%" />
<col width="50%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Quartet</th>
<th class="head">Experimental</th>
<th class="head">Plus Outgroup</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>dipterans</td>
<td>5,183</td>
<td>3,541</td>
</tr>
<tr class="row-odd"><td>vertebrates</td>
<td>8,159</td>
<td>2,518</td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">evolutionary divergence is defined here as the amount of expected synonymous substitutions per site in each species (<span class="formula"><i>Ks</i></span>
)</p>
</div>
<div class="line-block">
<div class="line"><br /></div>
<div class="line"><br /></div>
</div>
<div class="figure align-center" id="f1">
<a class="reference internal image-reference" href="../../../_images/F1.large.jpg"><img alt="../../../_images/F1.large.jpg" src="../../../_images/F1.large.jpg" style="width: 600px;" /></a>
<p class="caption"><strong>Figure 1</strong></p>
<div class="legend">
<em>Two quartets of species used in the analysis, of 1) dipterans and 2) vertebrates, together with the corresponding outgroup species. Evolutionary distances within each pair of species, characterized by the estimated per site number of synonymous substitutions Ks, are presented.</em></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<hr class="docutils" />
<div class="line-block">
<div class="line"><br /></div>
</div>
</div>
<div class="section" id="identification-and-analysis-of-orthologous-introns">
<h4>Identification and Analysis of Orthologous Introns</h4>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">orthologous introns here <strong>very basically</strong> means that the introns share the same flanking exons in all the species</p>
</div>
<ul>
<li><p class="first">a series of filters were applied to all the introns of the orthologous genes from each quartet</p>
</li>
<li><p class="first">the purpose of the filters were to focus on introns that were least likely to have conserved sequence (<strong>by alignability</strong>) while most likely to actually share common ancestry.</p>
</li>
<li><dl class="first docutils">
<dt>the final useful intron counts were:</dt>
<dd><ul class="first last simple">
<li>dipterans: 5,367</li>
<li>vertebrates: 51,844</li>
</ul>
</dd>
</dl>
</li>
</ul>
<div class="line-block">
<div class="line"><br /></div>
</div>
<hr class="docutils" />
<div class="line-block">
<div class="line"><br /></div>
</div>
</div>
<div class="section" id="calculation-of-expected-number-of-introns-carrying-a-segment-of-similarity-within-both-pairs">
<h4>Calculation of Expected Number of Introns Carrying a Segment of Similarity within Both Pairs</h4>
<div class="topic">
<p class="topic-title first">General Procedure</p>
<ol class="arabic simple">
<li>Orthologous introns in each species pair are blasted against each other looking for regions of <strong>local alignment</strong></li>
<li>Definition of successful <strong>local alignment</strong> are set by using progressively more stringent E-value thresholds</li>
<li>For each E-vale threshold: the number of orthologous introns producing successful <strong>local alignments</strong> are tallied</li>
<li>To determine the number of <em>expected</em> successful <strong>local alignments</strong>, the data are randomly shuffled 10,000 times</li>
<li>Four different shuffling methods for generating the <em>expected</em> successful <strong>local alignments</strong> were employed</li>
</ol>
</div>
<table border="1" class="docutils">
<colgroup>
<col width="81%" />
<col width="19%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Shuffling Methods</th>
<th class="head">Figure</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>no accounting for confounding factors</td>
<td>S1</td>
</tr>
<tr class="row-odd"><td>accounting for intron length</td>
<td>2</td>
</tr>
<tr class="row-even"><td>accounting for gene identity</td>
<td>S3</td>
</tr>
<tr class="row-odd"><td>accounting for gene identity <strong>and</strong> intron order</td>
<td>S4</td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br /></div>
</div>
<hr class="docutils" />
<div class="line-block">
<div class="line"><br /></div>
</div>
</div>
<div class="section" id="analysis-of-data-on-chromatin-modifications">
<h4>Analysis of Data on Chromatin Modifications</h4>
<p>Because so far the idea of &#8216;functional&#8217; has been simply inferred on the basis of coincidental inclusion of similar local sequence identity, the authors attempt to harvest experimentally validated functional regions from <em>Homo sapiens</em> and <em>Drosophila melanogaster</em> using the <a class="reference external" href="http://genome.ucsc.edu/ENCODE/">ENCODE</a> and <a class="reference external" href="http://www.modencode.org/">modENCODE</a> projects data.</p>
<p>For humans they focus on relevant strong enhancers (states 4 and 5) or insulators (state 8).  For flies they focus on enhancers (state 3) or &#8216;active introns&#8217; (state 4).</p>
<div class="line-block">
<div class="line"><br /></div>
</div>
<hr class="docutils" />
<div class="line-block">
<div class="line"><br /></div>
</div>
</div>
<div class="section" id="identification-of-intron-losses">
<h4>Identification of Intron Losses</h4>
<p>Direct quote:</p>
<blockquote>
<div>For the third and the fourth tests, to trace the losses of introns on the phylogeny, the tree corresponding to each species quartet was rooted with an outgroup species ([<a class="reference internal" href="#f1"><em>Figure 1</em></a>]). We then selected introns present in both species from pair 1 and in the outgroup species, assuming that such introns were present in the last common ancestor of both considered species pairs. Again, orthologous introns with significant similarity (bl2seq E-value ≤ 0.0001) between sequences from different pairs were excluded from analysis. These introns were then subdivided into 1) those also present in both pair 2 species and 2) those that had been lost in at least one of the pair 2 species. (Introns absent in one of the pair 2 species were usually also absent in the other, implying loss on the branch separating pair 2 from its common ancestor with pair 1; cases of intron loss mapping to external branches were rare.) For groups of introns indicated in 1) and 2), we compared the distributions of E-values within pair 1, and the distributions of intron lengths within pair 1. Reciprocal tests were performed analogously.</div></blockquote>
<div class="line-block">
<div class="line"><br /></div>
</div>
<hr class="docutils" />
<div class="line-block">
<div class="line"><br /></div>
</div>
</div>
</div>
<div class="section" id="results-and-discussion">
<h3>Results and Discussion</h3>
<div class="figure align-center" id="f2">
<a class="reference internal image-reference" href="../../../_images/F2.large.jpg"><img alt="../../../_images/F2.large.jpg" src="../../../_images/F2.large.jpg" style="width: 600px;" /></a>
<p class="caption"><strong>Figure 2</strong></p>
<div class="legend">
<em>Introns that carry a segment of high similarity between species of one pair are more likely to also carry a segment of high similarity between species of the other pair within a quartet. Each blue dot corresponds to a specific BLAST E-value (shown next to the dot), with lower values corresponding to more stringent similarity thresholds. Each E-value was used to detect similar segments within orthologous introns of the two species belonging to pair 1, and of the two species belonging to pair 2. Horizontal axis, fraction of introns that carry similar segments within pair 1, among introns present in all four species. Vertical axis, number (A, B) or observed-to-expected ratio (C, D) for the number of introns that carry similar segments both within pair 1 and 2. Top, dipterans; bottom, vertebrates. Observed-to-expected ratio was defined as the ratio of the observed number of introns with similarity within both pairs to the expected number if the segments of similarity were distributed randomly over all introns, controlling for intron lengths (see text). The red line and the gray area correspond to the mean and 95% confidence intervals for the expected values calculated in 10,000 resampling trials.</em></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<hr class="docutils" />
<div class="line-block">
<div class="line"><br /></div>
</div>
<div class="figure align-center" id="f3">
<a class="reference internal image-reference" href="../../../_images/F3.large.jpg"><img alt="../../../_images/F3.large.jpg" src="../../../_images/F3.large.jpg" style="width: 600px;" /></a>
<p class="caption"><strong>Figure 3</strong></p>
<div class="legend">
<em>Introns that carry a segment of similarity in pair 2 are more likely to overlap regulation-associated elements within pair 1. Each blue dot corresponds to a specific BLAST E-value (shown next to the dot). Each E-value was used to detect similar segments within orthologous introns of the two species belonging to pair 2. Horizontal axis, fraction of introns that carry similar segments within pair 2, among introns present in all four species. Vertical axis, observed-to-expected ratio for the number of introns that carry a regulation-associated element within pair 1, according to modENCODE (Kharchenko et al. 2010; Roy et al. 2010) (A, B) or ENCODE (ENCODE Project Consortium 2004; Ernst et al. 2011) (C–F) data, and also carry a segment of similarity within pair 2. (A, B) Dipterans; (C–F) vertebrates. (A) Active introns, any cell line; (B) enhancers, any cell line; (C) insulators, all adult cell lines; (D) insulators, both embryonic cell lines; (E) strong enhancers, all adult cell lines; (F) strong enhancers, both embryonic cell lines. Observed-to-expected ratio was defined as the ratio of the observed number of introns with a regulation-associated element within pair 1 and similarity within pair 2 to the same number expected if the regulation-associated elements and the segments of similarity were distributed randomly over all introns, controlling for intron lengths. The red line and the gray area correspond to the mean and 95% confidence intervals for the expected values calculated in 10,000 resampling trials.</em></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<hr class="docutils" />
<div class="line-block">
<div class="line"><br /></div>
</div>
<div class="figure align-center" id="f4">
<a class="reference internal image-reference" href="../../../_images/F4.large.jpg"><img alt="../../../_images/F4.large.jpg" src="../../../_images/F4.large.jpg" style="width: 600px;" /></a>
<p class="caption"><strong>Figure 4</strong></p>
<div class="legend">
<em>Introns preserved in both pair 2 species are more likely to carry a conserved segment within pair 1. For each quartet, the distribution of E-values within pair 1 are shown for introns preserved in both pair 2 species (green), and for introns lost in at least one of the pair 2 species (blue). E-values indicated below the horizontal axis correspond to the lower E-value threshold. Insets show the fraction of introns with at least a marginal (E-value ≤ 1) similarity observed, for the same two groups.</em></div>
</div>
<div class="line-block">
<div class="line"><br /></div>
</div>
<hr class="docutils" />
<div class="line-block">
<div class="line"><br /></div>
</div>
<div class="figure align-center" id="f5">
<a class="reference internal image-reference" href="../../../_images/F5.large.jpg"><img alt="../../../_images/F5.large.jpg" src="../../../_images/F5.large.jpg" style="width: 600px;" /></a>
<p class="caption"><strong>Figure 5</strong></p>
<div class="legend">
<em>Introns preserved in both pair 2 species tend to be longer within pair 1 species. For each quartet, the distribution of lengths within pair 1 are shown for introns preserved in both pair 2 species (green), and for introns lost in at least one of the pair 2 species (blue). Lengths indicated below the horizontal axis correspond to the lower length threshold. The length of the shorter of the two orthologous introns in pair 1 was used.</em></div>
</div>
</div>
</div>
<div class="section" id="license-information-pertaining-to-the-original-article">
<h2>License information pertaining to the original article</h2>
<p>Unless otherwise noted, all figures and text in the captions are the work of the original authors.  I claim no authorship to these whatsoever. The research was originally licensed and <a class="reference external" href="http://www.oxfordjournals.org/our_journals/gbe/terms.html">distributed under the terms of the Creative Commons Attribution Non-Commercial License (http://creativecommons.org/licenses/by-nc/3.0/), which permits unrestricted non-commercial use, distribution, and reproduction in any medium, provided the original work is properly cited.</a></p>
</div>
</div>

    <div class="postmeta">
        <div class="author">
            <span>Posted by W Augustine Dunn</span>
        </div>
        <div class="categories">
            <span>
                Filed under:
                <a href="../../../categories/journal_club.html">Journal Club</a>, <a href="../../../categories/journal_club__shorts_.html">Journal Club (shorts)</a></span>
        </div>
        <div class="tags">
            <span>
                Tags:
                <a href="../../../tags/research_blogging.html">Research Blogging</a>, <a href="../../../tags/evolution.html">evolution</a>, <a href="../../../tags/selection.html">selection</a>, <a href="../../../tags/conservation.html">conservation</a>, <a href="../../../tags/genomics.html">genomics</a>, <a href="../../../tags/sequence_analysis.html">sequence analysis</a></span>
        </div>
        </div>
    <div id="disqus_thread"></div><script type="text/javascript">    var disqus_shortname = "movingthelamppost";    var disqus_identifier = "2013/04/16/vakhrusheva_et_al_gbe_2013";    disqus_thread();</script><noscript>Please enable JavaScript to view the    <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript><ul class="related clearfix">
            <li class="left"></li>
            <li class="right"><a href="../15/blogging_my_dissertation.html">Blogging my dissertation</a> &raquo; </li>
        </ul></article><aside class="sidebar"><section>
<div class="widget" id="sub_by_email">
    <h1>Don't use RSS aggregators?</h1>
    <a href="http://feedburner.google.com/fb/a/mailverify?uri=MovingTheLamppost&amp;loc=en_US">Subscribe via email</a>
</div>

</section><section><div class="widget" id="searchbox">
    <h1>Search</h1>
    <form action="../../../search.html" method="get">
        <input type="text" name="q" />
        <button type="submit"><span class="webfont">L</span></button>
    </form>
</div></section><section><div class="widget">
    <h1>Tags Cloud</h1>
      <a href="../../../tags/blogging.html" style="font-size: 8pt">blogging</a>&nbsp;&nbsp;
      <a href="../../../tags/conservation.html" style="font-size: 8pt">conservation</a>&nbsp;&nbsp;
      <a href="../../../tags/evolution.html" style="font-size: 8pt">evolution</a>&nbsp;&nbsp;
      <a href="../../../tags/genomics.html" style="font-size: 8pt">genomics</a>&nbsp;&nbsp;
      <a href="../../../tags/ipython.html" style="font-size: 8pt">ipython</a>&nbsp;&nbsp;
      <a href="../../../tags/my_dissertation.html" style="font-size: 8pt">my dissertation</a>&nbsp;&nbsp;
      <a href="../../../tags/my_research.html" style="font-size: 8pt">my research</a>&nbsp;&nbsp;
      <a href="../../../tags/open_science.html" style="font-size: 20pt">open science</a>&nbsp;&nbsp;
      <a href="../../../tags/research.html" style="font-size: 8pt">research</a>&nbsp;&nbsp;
      <a href="../../../tags/research_blogging.html" style="font-size: 8pt">Research Blogging</a>&nbsp;&nbsp;
      <a href="../../../tags/selection.html" style="font-size: 8pt">selection</a>&nbsp;&nbsp;
      <a href="../../../tags/sequence_analysis.html" style="font-size: 8pt">sequence analysis</a>&nbsp;&nbsp;
      <a href="../../../tags/tinkerer.html" style="font-size: 8pt">tinkerer</a>
</div></section><section><div class="widget">
    <h1>Categories</h1>
    <ul><li><a href="../../../categories/journal_club.html">Journal Club</a> (1)</li><li><a href="../../../categories/journal_club__shorts_.html">Journal Club (shorts)</a> (1)</li><li><a href="../../../categories/my_dissertation.html">my dissertation</a> (1)</li></ul>
</div></section></aside></div> <!-- #main --></div> <!-- #main-container -->

        <div class="footer-container"><footer class="wrapper">&copy; Copyright 2013, W Augustine Dunn. Powered by <a href="http://www.tinkerer.me/">Tinkerer</a> and <a href="http://sphinx.pocoo.org/">Sphinx</a>.</footer></div> <!-- footer-container -->

      </div> <!--! end of #container --><!--[if lt IE 7 ]>
          <script src="//ajax.googleapis.com/ajax/libs/chrome-frame/1.0.3/CFInstall.min.js"></script>
          <script>window.attachEvent('onload',function(){CFInstall.check({mode:'overlay'})})</script>
        <![endif]-->
    </body>
</html>