<!DOCTYPE html><!--[if lt IE 7]>      <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]>         <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]>         <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]><!--> <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js"> <!--<![endif]-->
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
        <meta name="description" content="random musings of a molecular biologist turned code jockey in the era of big data and open science.">
        <meta name="viewport" content="width=device-width">
        <title>Home &mdash; moving the lamppost</title>
            <link rel="stylesheet" href="_static/normalize.css" type="text/css">
            <link rel="stylesheet" href="_static/sphinx.css" type="text/css">
            <link rel="stylesheet" href="_static/main.css" type="text/css">
            <link rel="stylesheet" href="_static/flat_enhanced_dark_pastel.css" type="text/css">
            <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
            <link rel="stylesheet" href="_static/webfont.css" type="text/css">
        <link rel="shortcut icon" href="_static/tinkerer.ico" /><!-- Load modernizr and JQuery -->
        <script src="_static/vendor/modernizr-2.6.2.min.js"></script>
        <script src="//ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js"></script>
        <script>window.jQuery || document.write('<script src="_static/vendor/jquery-1.8.2.min.js"><\/script>')</script>
        <script src="_static/plugins.js"></script>
        <script src="_static/main.js"></script>
        <link rel="next" title="Older" href="page2.html" /><link rel="alternate" type="application/rss+xml" title="RSS" href="http://feeds.feedburner.com/MovingTheLamppost" /><script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '1.2.1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script><script type="text/javascript" src="_static/underscore.js"></script><script type="text/javascript" src="_static/doctools.js"></script><script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><script type="text/javascript" src="_static/disqus.js"></script><script type="text/javascript" src="_static/google_analytics.js"></script>

    <script type="text/javascript">
        $(document).ready(function () {
            // Scroll to content if on small screen
            if (screen.width < 480)
            {
                $(document).scrollTop(document.getElementsByTagName("article")[0].offsetTop - 44);
            }
        });
    </script>
    
    <script type="text/javascript">
        $(document).ready(function () {
            // Adjusts document height if sidebar is taller
            var documentwrapper = document.getElementsByTagName("article")[0];
            var sidebar = document.getElementsByClassName("sidebar")[0];

            if (sidebar.offsetHeight > documentwrapper.offsetHeight)
            {
                var mainwrapper = document.getElementsByClassName("main")[0];
                documentwrapper.style.minHeight = mainwrapper.offsetHeight + "px";
            }

            // Scroll to content if on small screen
            if (screen.width < 480)
            {
                $(document).scrollTop(document.getElementsByTagName("article")[0].offsetTop - 44);
            }
        });
    </script>
    
    <link rel="stylesheet" type="text/css" href="http://fonts.googleapis.com/css?family=Droid+Serif:i|Noto+Sans:i|Titillium+Web|Titillium+Web:i"></head>
    <body>
        <!--[if lt IE 7]>
            <p class="chromeframe">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> or <a href="http://www.google.com/chromeframe/?redirect=true">activate Google Chrome Frame</a> to improve your experience.</p>
        <![endif]-->

      <div id="container"><header>
            <hgroup>
              <h1><a href="#">moving the lamppost</a></h1><h2>random musings of a molecular biologist turned code jockey in the era of big data and open science.</h2></hgroup>
          </header>
      <nav>
            <ul><li class="main-nav">
                  <a href="#">Home</a>
                </li>
              <li class="main-nav">
                  <a href="pages/about.html">About</a>
                </li>
              <li class="main-nav">
                  <a href="pages/glossary.html">Glossary</a>
                </li>
              </ul>
          </nav><div class="main-container"><div class="main wrapper clearfix"><article><div class="timestamp postmeta">
            <span>August 20, 2013</span>
        </div>
        <div class="section" id="lab-meeting-2013-08-21">
<h1><a href="2013/08/20/lab_meeting_2013_08_21.html">Lab meeting 2013-08-21</a></h1>
<div class="section" id="introduction">
<h2>1 Introduction</h2>
<div class="section" id="purpose">
<h3>1.1 Purpose</h3>
<p>Introduce core concepts relating to my project such as:</p>
<ol class="arabic simple">
<li>Importance of Dengue and Malaria</li>
<li>Transmission cycle dependant on Mosquitoes</li>
<li>Limited effectiveness (medical/finacial) of treatment of the diseases in rural poor human communities</li>
<li>Vector Control Strategies</li>
<li>How Transgenesis modifies the vector control landscape</li>
<li>Elements of Successful Transgenic Vector Control</li>
<li>Introduction to control of transcription and importance of CRE/CRMs</li>
</ol>
</div>
</div>
<hr class="docutils"/>
<div class="section" id="review-of-literature-contributions-and-development-of-analytical-approach">
<h2>2 Review of Literature Contributions and Development of Analytical Approach</h2>
<div class="section" id="id1">
<h3>2.1 Purpose</h3>
<p>Review a selection of my contributions to the literature that have informed the approach used in this project throughout my tenure as a member of the lab and explain the methods that I have used based on the lessons learned.</p>
<ol class="arabic">
<li><p class="first"><strong>Sieglaff, D. H., Dunn, W. A., Xie, X. S., Megy, K., Marinotti, O., &amp; James, A. A. (2009). Comparative genomics allows the discovery of cis-regulatory elements in mosquitoes. Proceedings of the National Academy of Sciences of the United States of America, 106(9), 3053–8. doi:10.1073/pnas.0813264106</strong></p>
<blockquote>
<div><ul class="simple">
<li>Use of comparative genomics to discover putative CREs from orthologous promoter regions in mosquitoes that were able to be correlated with bloodmeal associated transcription control</li>
<li>For any specific transcription profile, typically a single motif was found to be significantly enriched in the corresponding promoter regions (modules not obvious)</li>
</ul>
</div></blockquote>
</li>
<li><p class="first"><strong>Bonizzoni, M., Dunn, W. A., Campbell, C. L., Olson, K. E., Dimon, M. T., Marinotti, O., &amp; James, A. A. (2011). RNA-seq analyses of blood-induced changes in gene expression in the mosquito vector species, Aedes aegypti. BMC genomics, 12(1), 82. doi:10.1186/1471-2164-12-82</strong></p>
<blockquote>
<div><ul>
<li><p class="first">Provided first experience in data handling and analysis of RNA-seq big data</p>
</li>
<li><dl class="first docutils">
<dt>Illustrated challenges to using flanking regions as putative promoters in genomes at this level of annotation completion/quality</dt>
<dd><ul class="first last simple">
<li>annotation contains missed first exons and other incorrect feature definitions</li>
<li>upstream genes sometimes more likely to represent exon of downstream gene</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Nevertheless using SCOPE, putative CRMs were able to be predicted</p>
</li>
</ul>
</div></blockquote>
</li>
<li><p class="first"><strong>Bonizzoni, M., Dunn, W. a., Campbell, C. L., Olson, K. E., Marinotti, O., James, a. a., &amp; Kulathinal, R. (2012). Strain Variation in the Transcriptome of the Dengue Fever Vector, Aedes aegypti. G3: Genes|Genomes|Genetics, 2(1), 103–114. doi:10.1534/g3.111.001107</strong></p>
<blockquote>
<div><ul class="simple">
<li>Provided continued lessons on managing and analyzing RNAseq big data</li>
<li>Revealed that the transcriptomes of Ae. aegypti mosquitoes from distinct strains vary significantly in complexity and abundance of specific transcripts. This variation is evident in non- blood-fed mosquitoes and is enhanced after a bloodmeal.</li>
<li>differential use of paralogous genes</li>
<li>Support the need to take precautions when comparing differences in transcript abundance profiles between species that may have last shared a common ancestor as far back as 200 mya.</li>
</ul>
</div></blockquote>
</li>
<li><p class="first"><strong>Bonizzoni, M., Dunn, W. A., Campbell, C. L., Olson, K. E., Marinotti, O., &amp; James, A. A. (2012). Complex Modulation of the Aedes aegypti Transcriptome in Response to Dengue Virus Infection. PloS one, 7(11), e50512. doi:10.1371/journal.pone.0050512</strong></p>
<blockquote>
<div><ul class="simple">
<li>Example of how results (CRE-discovery in this case) are sensitive to the gene-sets use for analysis.</li>
</ul>
</div></blockquote>
</li>
<li><p class="first">Functional Genomics can be thought of as a problem of generating meaningful gene-sets</p>
<blockquote>
<div><ul class="simple">
<li>signal-to-noise is major challenge in generating useful gene-sets</li>
</ul>
</div></blockquote>
</li>
<li><p class="first">This project aims to generate meaningful gene-sets through integration of multiple data-types</p>
<blockquote>
<div><ul class="simple">
<li>comparative genomics (N-way 1-to-1 orthologs)</li>
<li>comparative transcriptomics (identical RNA-seq experiements on divergent Mosquito species with shared trait: bloodfeeding)</li>
<li>phylogenetics (estimated divergence since last common ancestor)</li>
<li>existing knowledge about certain bloodmeal related transcriptional control programs (20E interacting TFBS motifs)</li>
</ul>
</div></blockquote>
</li>
<li><p class="first">The last item presents analysis challenges that I addressed through custom graph-based analysis model (<span class="docutils literal"><span class="pre">gFunc</span></span>)</p>
</li>
</ol>
</div>
</div>
<hr class="docutils"/>
<div class="section" id="software-development-to-support-analytical-approach">
<h2>3 Software Development to Support Analytical Approach</h2>
<div class="section" id="id2">
<h3>3.1 Purpose</h3>
<p>Illustrate how lessons learned in previous chapter (as well as new lessons introduced here) are addressed through custom software solutions.</p>
<ol class="arabic simple">
<li>RNA-seq analysis, reproducibility, updating (<span class="docutils literal"><span class="pre">blacktie</span></span>)</li>
<li>Integration of multiple disparate Omics scale data types (<span class="docutils literal"><span class="pre">gFunc</span></span>)</li>
<li>Documentation of exactly how analyses were carried out (IPython notebooks and automated log keeping built in to <span class="docutils literal"><span class="pre">blacktie</span></span>)</li>
</ol>
</div>
</div>
<hr class="docutils"/>
<div class="section" id="results">
<h2>Results</h2>
<ol class="arabic">
<li><p class="first"><a class="reference external" href="http://nbviewer.ipython.org/urls/raw.github.com/xguse/ipy_notebooks/master/PTCI_testing_rsrd_1.0_1.1_new_data_ecr_team.ipynb">PTCI score finalization</a></p>
</li>
<li><p class="first"><a class="reference external" href="http://nbviewer.ipython.org/urls/raw.github.com/xguse/ipy_notebooks/master/Getting_and_using_filtered_gfunc_genes_v2.ipynb">Filtering of high-scoring N-way 1-to-1 ortholog sets</a></p>
</li>
<li><dl class="first docutils">
<dt>Motif discovery results with filtered gene-sets using STEME and TOMTOM</dt>
<dd><ul class="first last simple">
<li><a class="reference external" href="http://nbcr-222.ucsd.edu/opal-jobs/appTOMTOM_4.9.01377068911103-22439267/tomtom.html">STEME k9 vs JASPAR insect database</a></li>
<li><a class="reference external" href="http://nbcr-222.ucsd.edu/opal-jobs/appTOMTOM_4.9.01377068109211-275022927/tomtom.html">STEME k13 vs JASPAR insect database</a></li>
</ul>
</dd>
</dl>
</li>
</ol>
</div>
</div>
        <div class="postmeta">
        <div class="author">
            <span>Posted by W Augustine Dunn</span>
        </div>
        
        
        <div class="comments">
            <a href="http://movingthelamppost.com/blog/html/2013/08/20/lab_meeting_2013_08_21.html#disqus_thread" data-disqus-identifier="2013/08/20/lab_meeting_2013_08_21">Leave a comment</a>
        </div></div><div class="separator post_separator"></div><div class="timestamp postmeta">
            <span>July 22, 2013</span>
        </div>
        <div class="section" id="python-gotcha-generators-not-always-drop-in-replacements-for-lists">
<h1><a href="2013/07/22/python_gotcha__generators_not_always_drop_in_replacements_for_lists.html">Python Gotcha: generators not always drop-in replacements for lists</a></h1>
<p>I was recently bitten by this one.</p>
<div class="section" id="generators-iterators-and-lists">
<h2>Generators, Iterators and Lists</h2>
<p>In <a class="reference external" href="http://www.python.org">Python</a>, there is a data construct/concept called a <a class="reference external" href="http://wiki.python.org/moin/Generators">generator</a>.
In brief it is a special type of function that returns a type of <a class="reference external" href="http://wiki.python.org/moin/Iterator">iterator</a> that in most use cases behaves just like a Python <a class="reference external" href="http://rgruet.free.fr/PQR27/PQR2.7.html#list">list</a>.
That is, you can do stuff like this to it:</p>
<div class="highlight-ipython"><div class="highlight"><pre><span class="c"># define a simple generator</span>
<span class="gp">In [1]: </span><span class="k">def</span> <span class="nf">my_generator</span><span class="p">(</span><span class="n">count</span><span class="p">):</span>
<span class="gp">   ...:</span>      <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">count</span><span class="p">):</span>
<span class="gp">   ...:</span>          <span class="k">yield</span> <span class="n">i</span>
<span class="gp">   ...:</span> 
</pre></div>
</div>
<div class="highlight-ipython"><div class="highlight"><pre><span class="gp">In [2]: </span><span class="n">my_iterator</span> <span class="o">=</span> <span class="n">my_generator</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>

<span class="gp">In [3]: </span><span class="k">for</span> <span class="n">each</span> <span class="ow">in</span> <span class="n">my_iterator</span><span class="p">:</span>
<span class="gp">   ...:</span>      <span class="k">print</span> <span class="n">each</span>
<span class="gp">   ...:</span> 
<span class="go">0</span>
<span class="go">1</span>
<span class="go">2</span>
</pre></div>
</div>
<p>The benefit of an <span class="docutils literal"><span class="pre">iterator</span></span> is that it mitigates the memory footprint of the output of something like <span class="docutils literal"><span class="pre">my_generator()</span></span>.
So when you have an idea that a <span class="docutils literal"><span class="pre">list</span></span> of your resulting data will be very large, it may be wise to use an <span class="docutils literal"><span class="pre">iterator</span></span> rather than a <span class="docutils literal"><span class="pre">list</span></span>.
They work by generating the data on the fly rather than all at once.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>Though the title mentions generators specifically (this is because that’s what bit me just now) all of this pertains to all iterators.  I did not mean to imply that generators were special in this regard.</p>
<p class="last">Special thanks to reddit user <a class="reference external" href="http://www.reddit.com/r/Python/comments/1iurg2/til_generators_not_always_dropin_replacements_for/cb88c09">Megatron_McLargeHuge</a> for pointing out my sin of omission.</p>
</div>
<p>In our example above, <span class="docutils literal"><span class="pre">my_iterator</span></span> has no length as demonstrated below.</p>
<div class="highlight-ipython"><div class="highlight"><pre><span class="gp">In [4]: </span><span class="n">my_iterator</span> <span class="o">=</span> <span class="n">my_generator</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>

<span class="gp">In [5]: </span><span class="nb">len</span><span class="p">(</span><span class="n">my_iterator</span><span class="p">)</span>
<span class="go">---------------------------------------------------------------------------</span>
<span class="go">TypeError                                 Traceback (most recent call last)</span>
<span class="go">&lt;ipython-input-5-110da8bf0288&gt; in &lt;module&gt;()</span>
<span class="go">----&gt; 1 len(my_iterator)</span>

<span class="go">TypeError: object of type 'generator' has no len()</span>
</pre></div>
</div>
<p>This can bite you if you don’t know (or <strong>remember</strong>) that you are dealing with a dynamically generated <span class="docutils literal"><span class="pre">iterator</span></span>.
In my case, I was using a generator function that returns an <span class="docutils literal"><span class="pre">iterator</span></span> for all unique 2-way combinations of a <span class="docutils literal"><span class="pre">list</span></span> of values.
I wanted these unique combinations to be used inside a subsequent <span class="docutils literal"><span class="pre">for</span></span> loop.</p>
</div>
<div class="section" id="here-was-the-code-that-got-me">
<h2>Here was the code that got me:</h2>
<div class="highlight-python"><div class="highlight"><pre><span class="k">def</span> <span class="nf">calc_mean_ptcis</span><span class="p">(</span><span class="n">graphHandler</span><span class="p">,</span><span class="n">n_way_ortho_table</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    returns list of n-way averaged PTCI values for N-way ortholog subgraphs if and only if</span>
<span class="sd">    all edges successfully generated non-None value PTCI results.</span>
<span class="sd">    &quot;&quot;&quot;</span>

    <span class="c"># dictionary of nodes/edges indexed by gene names</span>
    <span class="n">node_dict</span> <span class="o">=</span> <span class="n">graphHandler</span><span class="o">.</span><span class="n">node_dict</span>
    <span class="n">edge_dict</span> <span class="o">=</span> <span class="n">graphHandler</span><span class="o">.</span><span class="n">edge_dict</span>

    <span class="n">graph</span> <span class="o">=</span> <span class="n">graphHandler</span><span class="o">.</span><span class="n">graph</span>

    <span class="n">mean_ptcis</span> <span class="o">=</span> <span class="p">[]</span>

    <span class="c"># calculate all pairwise combinations of indexes</span>
    <span class="c"># so that each ortho-edge of n-way orthos are obtained</span>

    <span class="c"># !---- look here ----! #</span>
    <span class="n">index_combos</span> <span class="o">=</span> <span class="n">xuniqueCombinations</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">n_way_ortho_table</span><span class="o">.</span><span class="n">columns</span><span class="p">)),</span><span class="mi">2</span><span class="p">)</span>

    <span class="k">for</span> <span class="n">node_list</span> <span class="ow">in</span> <span class="n">n_way_ortho_table</span><span class="o">.</span><span class="n">itertuples</span><span class="p">():</span>

        <span class="n">node_list</span> <span class="o">=</span> <span class="n">node_list</span><span class="p">[</span><span class="mi">1</span><span class="p">:]</span>
        <span class="n">ortho_edges</span> <span class="o">=</span> <span class="p">[]</span>
        <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="n">index_combos</span><span class="p">:</span>
            <span class="n">key</span> <span class="o">=</span> <span class="nb">tuple</span><span class="p">(</span><span class="nb">sorted</span><span class="p">([</span><span class="n">node_list</span><span class="p">[</span><span class="n">i</span><span class="p">[</span><span class="mi">0</span><span class="p">]],</span><span class="n">node_list</span><span class="p">[</span><span class="n">i</span><span class="p">[</span><span class="mi">1</span><span class="p">]]]))</span>

            <span class="k">try</span><span class="p">:</span>
                <span class="n">ortho_edges</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">edge_dict</span><span class="p">[</span><span class="n">key</span><span class="p">])</span>
            <span class="k">except</span> <span class="ne">KeyError</span><span class="p">:</span>
                <span class="k">break</span>

        <span class="n">ptcis</span> <span class="o">=</span> <span class="p">[</span><span class="n">dev</span><span class="o">.</span><span class="n">get_ptci</span><span class="p">(</span><span class="n">edge</span><span class="p">)</span> <span class="k">for</span> <span class="n">edge</span> <span class="ow">in</span> <span class="n">ortho_edges</span><span class="p">]</span>

        <span class="k">try</span><span class="p">:</span>
            <span class="n">mean_ptci</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">mean</span><span class="p">(</span><span class="n">ptcis</span><span class="p">)</span>
            <span class="n">mean_ptcis</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">mean_ptci</span><span class="p">)</span>
        <span class="k">except</span> <span class="ne">TypeError</span><span class="p">:</span>
            <span class="k">pass</span>

    <span class="k">return</span> <span class="n">mean_ptcis</span>
</pre></div>
</div>
</div>
<div class="section" id="iterators-are-consumed-at-the-time-of-access">
<h2>Iterators are consumed at the time of access</h2>
<p>I expected <span class="docutils literal"><span class="pre">index_combos</span></span> to persist and be re-used in each iteration of the loop.
However, <span class="docutils literal"><span class="pre">index_combos</span></span> is an <span class="docutils literal"><span class="pre">iterator</span></span>, so it is actually consumed in that first loop and returns nothing useful in <strong>ANY</strong> of the subsequent loops.
I did not write <span class="docutils literal"><span class="pre">xuniqueCombinations</span></span> which is how I got bitten; I do host a <a class="reference external" href="https://gist.github.com">gist</a> of it <a class="reference external" href="https://gist.github.com/xguse/6058998">here</a>.
It’s actually quite useful, and you should check it out.</p>
<p>The point is that I was expecting a <span class="docutils literal"><span class="pre">list</span></span> and got an <span class="docutils literal"><span class="pre">iterator</span></span>.
The solution was to use a <a class="reference external" href="http://docs.python.org/2/tutorial/datastructures.html#list-comprehensions">list comprehension</a> to consume the <span class="docutils literal"><span class="pre">iterator</span></span> and store the output in a <span class="docutils literal"><span class="pre">list</span></span>.
That new <span class="docutils literal"><span class="pre">list</span></span> is of course persistent and fixed the whole problem.</p>
<div class="highlight-python"><div class="highlight"><pre><span class="n">index_combos</span> <span class="o">=</span> <span class="p">[</span> <span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">xuniqueCombinations</span><span class="p">(</span><span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">n_way_ortho_table</span><span class="o">.</span><span class="n">columns</span><span class="p">)),</span><span class="mi">2</span><span class="p">)</span> <span class="p">]</span>
</pre></div>
</div>
</div>
<div class="section" id="final-thoughts">
<h2>Final thoughts</h2>
<p>I spent a good long time dissecting and re-writing all of the rest of that code before I realized what was happening.
I hope that others might benefit from my pain by reading this.</p>
<p>Generators and iterators are awesome and help make Python the amazing language that it is, but the fact that they look like a <span class="docutils literal"><span class="pre">list</span></span> in many code use-cases can result in some not-so-fun debugging sessions unless you pay close attention!</p>
</div>
</div>
        <div class="postmeta">
        <div class="author">
            <span>Posted by W Augustine Dunn</span>
        </div>
        <div class="categories">
            <span>
                Filed under:
                <a href="categories/python_shorts.html">Python Shorts</a>, <a href="categories/tutorials.html">Tutorials</a></span>
        </div>
        <div class="tags">
            <span>
                Tags:
                <a href="tags/python_gotcha.html">python gotcha</a>, <a href="tags/python.html">python</a>, <a href="tags/generator.html">generator</a>, <a href="tags/iterator.html">iterator</a>, <a href="tags/list.html">list</a>, <a href="tags/xpermutations_py.html">xpermutations.py</a>, <a href="tags/programming.html">programming</a></span>
        </div>
        <div class="comments">
            <a href="http://movingthelamppost.com/blog/html/2013/07/22/python_gotcha__generators_not_always_drop_in_replacements_for_lists.html#disqus_thread" data-disqus-identifier="2013/07/22/python_gotcha__generators_not_always_drop_in_replacements_for_lists">Leave a comment</a>
        </div></div><div class="separator post_separator"></div><div class="timestamp postmeta">
            <span>July 17, 2013</span>
        </div>
        <div class="section" id="blacktie-bugfix-release-0-2-1-2">
<h1><a href="2013/07/17/blacktie_bugfix_release__0_2_1_2.html">Blacktie BUGFIX release: 0.2.1.2</a></h1>
<a class="reference internal image-reference" href="_images/basic_cummerbund_plots.png"><img alt="../../../_images/basic_cummerbund_plots.png" src="_images/basic_cummerbund_plots.png" style="width: 600px;"/></a>
<ul class="simple">
<li>github repo: <a class="reference external" href="https://github.com/xguse/blacktie">https://github.com/xguse/blacktie</a></li>
<li>documentation: <a class="reference external" href="https://blacktie.readthedocs.org/en/latest">https://blacktie.readthedocs.org/en/latest/</a></li>
<li><strong>issue tracking and feature requests:</strong> <a class="reference external" href="https://github.com/xguse/blacktie/issues">https://github.com/xguse/blacktie/issues</a></li>
</ul>
<div class="section" id="the-bug">
<h2>The Bug:</h2>
<p>In short, on a Mac <span class="docutils literal"><span class="pre">pprocess</span></span> complained:</p>
<div class="highlight-python"><div class="highlight"><pre><span class="s">&quot;AttributeError: 'module' object has no attribute 'poll'&quot;</span>
</pre></div>
</div>
<p>when trying to set up a queue.</p>
<p>This caused a fatal unhandled exception that was discovered on a Macintosh.
Version 0.2.1.2 should restore functionality of <span class="docutils literal"><span class="pre">blacktie</span></span> for people experiencing this problem.</p>
<p>For more detail see github issue: <a class="reference external" href="https://github.com/xguse/blacktie/issues/10">https://github.com/xguse/blacktie/issues/10</a></p>
</div>
<div class="section" id="the-fix">
<h2>The Fix:</h2>
<p>This is a quick fix that looks for that exception and continues without using <span class="docutils literal"><span class="pre">pprocess</span></span> if it is encountered.
A more permanent fix will be included once I understand what is going on.</p>
<div class="admonition attention">
<p class="first admonition-title">Attention</p>
<ul class="last simple">
<li>It is recommended that all users update to 0.2.1.2 to avoid this issue.</li>
<li>Anyone with knowledge of this issue is encouraged to comment on the <a class="reference external" href="https://github.com/xguse/blacktie/issues/10">issues thread</a> above or in the Comments section below.</li>
</ul>
</div>
</div>
<div class="section" id="what-is-blacktie">
<h2>What is <span class="docutils literal"><span class="pre">blacktie</span></span>?</h2>
<p><span class="docutils literal"><span class="pre">blacktie</span></span> is a pipeline to streamline the deployment of the popular “Tuxedo Protocol” for RNA-seq analysis.
Please take a look at the <a class="reference external" href="http://xguse.github.io/blacktie/project.html">Project Summary</a> section of <span class="docutils literal"><span class="pre">blacktie</span></span>‘s documentation for more details.</p>
</div>
</div>
        <div class="postmeta">
        <div class="author">
            <span>Posted by W Augustine Dunn</span>
        </div>
        <div class="categories">
            <span>
                Filed under:
                <a href="categories/my_code.html">My Code</a></span>
        </div>
        <div class="tags">
            <span>
                Tags:
                <a href="tags/blacktie.html">blacktie</a>, <a href="tags/tophat.html">tophat</a>, <a href="tags/cufflinks.html">cufflinks</a>, <a href="tags/cummerbund.html">cummeRbund</a>, <a href="tags/python.html">python</a>, <a href="tags/open_science.html">open science</a>, <a href="tags/open_source.html">open source</a>, <a href="tags/rna_seq.html">RNA-seq</a>, <a href="tags/rna.html">RNA</a></span>
        </div>
        <div class="comments">
            <a href="http://movingthelamppost.com/blog/html/2013/07/17/blacktie_bugfix_release__0_2_1_2.html#disqus_thread" data-disqus-identifier="2013/07/17/blacktie_bugfix_release__0_2_1_2">Leave a comment</a>
        </div></div><div class="separator post_separator"></div><div class="timestamp postmeta">
            <span>July 12, 2013</span>
        </div>
        <div class="section" id="installing-pyqt-because-it-s-too-good-for-pip-or-easy-install">
<h1><a href="2013/07/12/installing_pyqt____because_it_s_too_good_for_pip_or_easy_install_.html">Installing PyQt... because it’s too good for pip or easy_install.</a></h1>
<p>This is a tutorial on installing <span class="docutils literal"><span class="pre">PyQt</span></span> and its dependency <span class="docutils literal"><span class="pre">SIP</span></span>.
I am currently working on a Python package to integrate multiple disparate types of Omics data using a graph model.
Some of my code requires <a class="reference external" href="http://ete.cgenomics.org">ETE: a Python Environment for phylogenetic Tree Exploration</a> which requires <span class="docutils literal"><span class="pre">PyQt</span></span>.</p>
<p>Normally, I would simply use</p>
<div class="highlight-bash"><div class="highlight"><pre>pip install -U package_name
</pre></div>
</div>
<p>to install <span class="docutils literal"><span class="pre">package_name</span></span> and ensure that all of its dependencies are installed and updated.
However, <span class="docutils literal"><span class="pre">PyQt</span></span> and <span class="docutils literal"><span class="pre">SIP</span></span> do not use the standard <span class="docutils literal"><span class="pre">setup.py</span></span> file for encoding the installation procedures.
They use <span class="docutils literal"><span class="pre">configure.py</span></span>.
So <span class="docutils literal"><span class="pre">pip</span></span> and other <a class="reference external" href="https://pypi.python.org/pypi">PyPI</a> installer scripts will locate and download the code, but they will fail during the installation phase and complain that they can not be installed with <span class="docutils literal"><span class="pre">setuptools</span></span>.</p>
<p>This was frustrating me since I know that I had these installed on my last system.
I just couldn’t remember how I did it.
I must have done the whole thing manually.
In any case, thanks to a <a class="reference external" href="http://problemssol.blogspot.com/2010/12/compile-and-install-pyqt4-for-python27.html">post</a> I found using the google box at <a class="reference external" href="http://problemssol.blogspot.com">http://problemssol.blogspot.com</a>, I found a slightly easier way to get the job done.
I am posting it here to maybe help others, but also so that I have it documented for myself the next time I have to do this.</p>
<p>I am borrowing <strong>heavily</strong> from the post above but adding some extras that I ran into on my <a class="reference external" href="https://www.archlinux.org">Arch Linux</a> system.</p>
<p><a class="readmore" href="2013/07/12/installing_pyqt____because_it_s_too_good_for_pip_or_easy_install_.html#more">Read more...</a></p></div>
        <div class="postmeta">
        <div class="author">
            <span>Posted by W Augustine Dunn</span>
        </div>
        <div class="categories">
            <span>
                Filed under:
                <a href="categories/tutorials.html">Tutorials</a></span>
        </div>
        <div class="tags">
            <span>
                Tags:
                <a href="tags/python.html">Python</a>, <a href="tags/pyqt.html">PyQt</a>, <a href="tags/pip.html">pip</a>, <a href="tags/setup_py.html">setup.py</a>, <a href="tags/configure_py.html">configure.py</a>, <a href="tags/sip.html">SIP</a></span>
        </div>
        <div class="comments">
            <a href="http://movingthelamppost.com/blog/html/2013/07/12/installing_pyqt____because_it_s_too_good_for_pip_or_easy_install_.html#disqus_thread" data-disqus-identifier="2013/07/12/installing_pyqt____because_it_s_too_good_for_pip_or_easy_install_">Leave a comment</a>
        </div></div><div class="separator post_separator"></div><div class="timestamp postmeta">
            <span>July 04, 2013</span>
        </div>
        <div class="section" id="first-pass-trinity-results-for-anopheles-stephensi-midgut-rna-seq-non-bloodfed">
<h1><a href="2013/07/04/first_pass_trinity_results_for_anopheles_stephensi_midgut_rna_seq__non_bloodfed_.html">First pass Trinity Results for Anopheles stephensi midgut RNA-seq (non-bloodfed)</a></h1>
<div class="section" id="exploration-of-the-first-set-of-results-from-a-trinity-run-with-default-settings-on-one-library-of-anopheles-stephensi-midgut-rna-seq-as-00-lib-one">
<h2>Exploration of the first set of results from a Trinity run with default settings on one library of Anopheles stephensi midgut RNA-seq (As_00 lib one)</h2>
<div class="section" id="analyze-lengths-of-assembled-contigs-transcripts">
<h3>Analyze Lengths Of Assembled Contigs/Transcripts</h3>
<p>Count all transcripts of length <em>X</em>:</p>
<p>In[1]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="kn">from</span> <span class="nn">collections</span> <span class="kn">import</span> <span class="n">defaultdict</span>
<span class="kn">from</span> <span class="nn">rSeq.utils.files</span> <span class="kn">import</span> <span class="n">ParseFastA</span>

<span class="k">def</span> <span class="nf">measure_and_count_transcripts</span><span class="p">(</span><span class="n">fasta_path</span><span class="p">):</span>

    <span class="n">tx_lens</span> <span class="o">=</span> <span class="n">defaultdict</span><span class="p">(</span><span class="nb">list</span><span class="p">)</span>

    <span class="n">p</span> <span class="o">=</span> <span class="n">ParseFastA</span><span class="p">(</span><span class="n">fasta_path</span><span class="p">)</span>
    <span class="n">seqs</span> <span class="o">=</span> <span class="n">p</span><span class="o">.</span><span class="n">to_dict</span><span class="p">()</span>

    <span class="k">for</span> <span class="n">name</span><span class="p">,</span><span class="n">seq</span> <span class="ow">in</span> <span class="n">seqs</span><span class="o">.</span><span class="n">iteritems</span><span class="p">():</span>
        <span class="n">tx_lens</span><span class="p">[</span><span class="nb">len</span><span class="p">(</span><span class="n">seq</span><span class="p">)]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">name</span><span class="p">)</span>

    <span class="k">print</span> <span class="s">&quot;The number of different transcript sizes encountered: </span><span class="si">%s</span><span class="s">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">tx_lens</span><span class="p">))</span>
    <span class="k">return</span> <span class="n">tx_lens</span>
</pre></div>
</div>
<p>Now I will plot the size data in a number of different ways.</p>
<p>In[2]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="c"># Prep to plot a Histogram of the numbers of transcripts in each length bin</span>

<span class="n">trinity_tx_lens</span> <span class="o">=</span> <span class="n">measure_and_count_transcripts</span><span class="p">(</span><span class="s">'/home/gus/Dropbox/common/projects/trinity_As/trinity_As_00_0_test/Trinity.fasta'</span><span class="p">)</span>

<span class="n">trinity_lengths</span> <span class="o">=</span> <span class="p">[]</span>

<span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">trinity_tx_lens</span><span class="o">.</span><span class="n">iteritems</span><span class="p">():</span>
    <span class="n">trinity_lengths</span><span class="o">.</span><span class="n">extend</span><span class="p">([</span><span class="n">item</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">*</span> <span class="nb">len</span><span class="p">(</span><span class="n">item</span><span class="p">[</span><span class="mi">1</span><span class="p">]))</span>

<span class="n">trinity_lengths</span><span class="o">.</span><span class="n">sort</span><span class="p">()</span>
</pre></div>
</div>
<div class="highlight-python"><pre>The number of different transcript sizes encountered: 4632</pre>
</div>
<p>In[3]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">hist</span><span class="p">(</span><span class="n">trinity_lengths</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">histtype</span><span class="o">=</span><span class="s">'stepfilled'</span><span class="p">,</span><span class="n">log</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">'transcript length'</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">'number of transcripts'</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x3ecb2d0&gt;</pre>
</div>
<img alt="../../../_images/_fig_03.png" src="_images/_fig_03.png"/>
<p>In[4]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">plot</span><span class="p">(</span><span class="n">trinity_lengths</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">'Trinity Contigs (sorted by length)'</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">'Length of Contig'</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x2ad3290&gt;</pre>
</div>
<img alt="../../../_images/_fig_06.png" src="_images/_fig_06.png"/>
<p>Now I calculate the N50_length and N50_index for the trinity assembly.</p>
<p><strong>N50_index:</strong> the number of largest contigs(transcripts) whose summed
lengths equal at least 50% of the sum of <strong>ALL</strong> contigs (smaller
numbers are ‘better’)</p>
<p><strong>N50_length:</strong> the length of the last added contig from above (larger
numbers are ‘better’)</p>
<p>In[5]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="k">def</span> <span class="nf">calc_N50s</span><span class="p">(</span><span class="n">lengths</span><span class="p">):</span>
    <span class="n">total_length</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">lengths</span><span class="p">)</span>
    <span class="k">print</span> <span class="s">&quot;total_length:</span><span class="se">\t\t</span><span class="si">%s</span><span class="s">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">total_length</span><span class="p">)</span>

    <span class="n">running_total</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">collected_lengths</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">n50_i</span> <span class="o">=</span> <span class="bp">None</span>

    <span class="k">for</span> <span class="n">length</span> <span class="ow">in</span> <span class="nb">reversed</span><span class="p">(</span><span class="n">lengths</span><span class="p">):</span>
        <span class="n">running_total</span> <span class="o">+=</span> <span class="n">length</span>
        <span class="n">collected_lengths</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">length</span><span class="p">)</span>

        <span class="k">if</span> <span class="n">running_total</span> <span class="o">&gt;=</span> <span class="n">total_length</span> <span class="o">*</span> <span class="mf">0.5</span><span class="p">:</span>
            <span class="n">n50_i</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">collected_lengths</span><span class="p">)</span>
            <span class="n">n50_l</span> <span class="o">=</span> <span class="n">length</span>
            <span class="k">break</span>

    <span class="k">print</span> <span class="s">&quot;running_total:</span><span class="se">\t\t</span><span class="si">%s</span><span class="s">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">running_total</span><span class="p">)</span>
    <span class="k">print</span> <span class="s">&quot;N50_index:</span><span class="se">\t\t</span><span class="si">%s</span><span class="s"> of </span><span class="si">%s</span><span class="s">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">n50_i</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">lengths</span><span class="p">))</span>
    <span class="k">print</span> <span class="s">&quot;N50_length:</span><span class="se">\t\t</span><span class="si">%s</span><span class="s">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">n50_l</span><span class="p">)</span>
    <span class="k">print</span> <span class="s">&quot;median contig length</span><span class="se">\t</span><span class="si">%s</span><span class="s">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">median</span><span class="p">(</span><span class="n">lengths</span><span class="p">))</span>
</pre></div>
</div>
<p>In[6]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">calc_N50s</span><span class="p">(</span><span class="n">trinity_lengths</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><pre>total_length:               31858558
running_total:              15931068
N50_index:                  3927 of 25167
N50_length:                 2469
median contig length        647.0</pre>
</div>
</div></div><p><a class="readmore" href="2013/07/04/first_pass_trinity_results_for_anopheles_stephensi_midgut_rna_seq__non_bloodfed_.html#more">Read more...</a></p></div>
        <div class="postmeta">
        <div class="author">
            <span>Posted by W Augustine Dunn</span>
        </div>
        <div class="categories">
            <span>
                Filed under:
                <a href="categories/my_research.html">My Research</a></span>
        </div>
        <div class="tags">
            <span>
                Tags:
                <a href="tags/rna_seq.html">RNA-seq</a>, <a href="tags/de_novo_transcriptome_assembly.html">de novo transcriptome assembly</a>, <a href="tags/trinity.html">Trinity</a>, <a href="tags/anopheles_stephensi.html">Anopheles stephensi</a>, <a href="tags/midgut.html">midgut</a>, <a href="tags/non_bloodfed.html">non-bloodfed</a>, <a href="tags/blood_feeding.html">blood feeding</a>, <a href="tags/ipython.html">ipython</a>, <a href="tags/python.html">python</a></span>
        </div>
        <div class="comments">
            <a href="http://movingthelamppost.com/blog/html/2013/07/04/first_pass_trinity_results_for_anopheles_stephensi_midgut_rna_seq__non_bloodfed_.html#disqus_thread" data-disqus-identifier="2013/07/04/first_pass_trinity_results_for_anopheles_stephensi_midgut_rna_seq__non_bloodfed_">Leave a comment</a>
        </div></div><div class="archive_link">
        <a href="archive.html"> &mdash; Blog Archive &mdash; </a>
    </div><ul class="related clearfix">
            <li class="left"></li>
            <li class="right"><a href="page2.html">Older</a> &raquo; </li>
        </ul></article><aside class="sidebar"><section>
<div class="widget" id="sub_by_email">
    <h1>Subscribe</h1>
    <a href='http://www.feedly.com/home#subscription/feed%2Fhttp%3A%2F%2Ffeeds.feedburner.com%2FMovingTheLamppost'  target='blank'><img id='feedlyFollow' src='http://s3.feedly.com/img/follows/feedly-follow-rectangle-volume-medium_2x.png' alt='follow us in feedly' width='71' height='28'></a>
    <br>
    <a href="http://feeds.feedburner.com/MovingTheLamppost">via feedburner</a>
    <br>
    <a href="http://feedburner.google.com/fb/a/mailverify?uri=MovingTheLamppost&amp;loc=en_US">via email</a>
    
</div>

</section><section><div class="widget" id="searchbox">
    <h1>Search</h1>
    <form action="search.html" method="get">
        <input type="text" name="q" />
        <button type="submit"><span class="webfont">L</span></button>
    </form>
</div></section><section><div class="widget">
    <h1>Categories</h1>
    <ul><li><a href="categories/background_science.html">Background Science</a> (1)</li><li><a href="categories/conferences.html">Conferences</a> (1)</li><li><a href="categories/journal_club.html">Journal Club</a> (1)</li><li><a href="categories/journal_club__shorts_.html">Journal Club (shorts)</a> (1)</li><li><a href="categories/my_code.html">My Code</a> (3)</li><li><a href="categories/my_dissertation.html">My Dissertation</a> (3)</li><li><a href="categories/my_presentations.html">My Presentations</a> (1)</li><li><a href="categories/my_research.html">My Research</a> (4)</li><li><a href="categories/python_shorts.html">Python Shorts</a> (1)</li><li><a href="categories/tutorials.html">Tutorials</a> (2)</li></ul>
</div></section><section><div class="widget">
    <h1>Tags Cloud</h1>
      <a href="tags/amino_acids.html" style="font-size: 8pt">amino acids</a>&nbsp;&nbsp;
      <a href="tags/anopheles_stephensi.html" style="font-size: 8pt">Anopheles stephensi</a>&nbsp;&nbsp;
      <a href="tags/arthgen2013.html" style="font-size: 8pt">ArthGen2013</a>&nbsp;&nbsp;
      <a href="tags/arthropods.html" style="font-size: 8pt">arthropods</a>&nbsp;&nbsp;
      <a href="tags/asif.html" style="font-size: 8pt">Asif</a>&nbsp;&nbsp;
      <a href="tags/asif_mohiuddin.html" style="font-size: 8pt">Asif Mohiuddin</a>&nbsp;&nbsp;
      <a href="tags/background.html" style="font-size: 10pt">background</a>&nbsp;&nbsp;
      <a href="tags/bangladesh.html" style="font-size: 8pt">Bangladesh</a>&nbsp;&nbsp;
      <a href="tags/bioinformatics.html" style="font-size: 8pt">bioinformatics</a>&nbsp;&nbsp;
      <a href="tags/blacktie.html" style="font-size: 14pt">blacktie</a>&nbsp;&nbsp;
      <a href="tags/blogging.html" style="font-size: 8pt">blogging</a>&nbsp;&nbsp;
      <a href="tags/blood_feeding.html" style="font-size: 8pt">blood feeding</a>&nbsp;&nbsp;
      <a href="tags/cell.html" style="font-size: 8pt">cell</a>&nbsp;&nbsp;
      <a href="tags/configure_py.html" style="font-size: 8pt">configure.py</a>&nbsp;&nbsp;
      <a href="tags/conservation.html" style="font-size: 8pt">conservation</a>&nbsp;&nbsp;
      <a href="tags/cuffdiff.html" style="font-size: 10pt">cuffdiff</a>&nbsp;&nbsp;
      <a href="tags/cufflinks.html" style="font-size: 12pt">cufflinks</a>&nbsp;&nbsp;
      <a href="tags/cuffmerge.html" style="font-size: 10pt">cuffmerge</a>&nbsp;&nbsp;
      <a href="tags/cummerbund.html" style="font-size: 12pt">cummeRbund</a>&nbsp;&nbsp;
      <a href="tags/de_novo_transcriptome_assembly.html" style="font-size: 8pt">de novo transcriptome assembly</a>&nbsp;&nbsp;
      <a href="tags/dissertation__chapter_one__background_.html" style="font-size: 10pt">Dissertation: Chapter One (Background)</a>&nbsp;&nbsp;
      <a href="tags/dna.html" style="font-size: 8pt">DNA</a>&nbsp;&nbsp;
      <a href="tags/evolution.html" style="font-size: 8pt">evolution</a>&nbsp;&nbsp;
      <a href="tags/free_expression.html" style="font-size: 8pt">free expression</a>&nbsp;&nbsp;
      <a href="tags/freedom.html" style="font-size: 8pt">freedom</a>&nbsp;&nbsp;
      <a href="tags/gene_drive.html" style="font-size: 8pt">gene drive</a>&nbsp;&nbsp;
      <a href="tags/gene_expression.html" style="font-size: 10pt">gene expression</a>&nbsp;&nbsp;
      <a href="tags/gene_regulation.html" style="font-size: 8pt">gene regulation</a>&nbsp;&nbsp;
      <a href="tags/generator.html" style="font-size: 8pt">generator</a>&nbsp;&nbsp;
      <a href="tags/genomics.html" style="font-size: 10pt">genomics</a>&nbsp;&nbsp;
      <a href="tags/gmm.html" style="font-size: 10pt">GMM</a>&nbsp;&nbsp;
      <a href="tags/gmo.html" style="font-size: 10pt">GMO</a>&nbsp;&nbsp;
      <a href="tags/government.html" style="font-size: 8pt">government</a>&nbsp;&nbsp;
      <a href="tags/gripes.html" style="font-size: 8pt">gripes</a>&nbsp;&nbsp;
      <a href="tags/human_rights.html" style="font-size: 8pt">human rights</a>&nbsp;&nbsp;
      <a href="tags/injustice.html" style="font-size: 8pt">injustice</a>&nbsp;&nbsp;
      <a href="tags/intolerance.html" style="font-size: 8pt">intolerance</a>&nbsp;&nbsp;
      <a href="tags/ipython.html" style="font-size: 10pt">ipython</a>&nbsp;&nbsp;
      <a href="tags/iterator.html" style="font-size: 8pt">iterator</a>&nbsp;&nbsp;
      <a href="tags/list.html" style="font-size: 8pt">list</a>&nbsp;&nbsp;
      <a href="tags/live_blogging.html" style="font-size: 8pt">live blogging</a>&nbsp;&nbsp;
      <a href="tags/midgut.html" style="font-size: 8pt">midgut</a>&nbsp;&nbsp;
      <a href="tags/mohiuddin.html" style="font-size: 8pt">Mohiuddin</a>&nbsp;&nbsp;
      <a href="tags/mosquitoes.html" style="font-size: 10pt">mosquitoes</a>&nbsp;&nbsp;
      <a href="tags/my_dissertation.html" style="font-size: 8pt">My Dissertation</a>&nbsp;&nbsp;
      <a href="tags/my_research.html" style="font-size: 8pt">My Research</a>&nbsp;&nbsp;
      <a href="tags/my_research.html" style="font-size: 8pt">my research</a>&nbsp;&nbsp;
      <a href="tags/namespace.html" style="font-size: 8pt">namespace</a>&nbsp;&nbsp;
      <a href="tags/new_release.html" style="font-size: 10pt">new release</a>&nbsp;&nbsp;
      <a href="tags/ngs.html" style="font-size: 12pt">NGS</a>&nbsp;&nbsp;
      <a href="tags/non_bloodfed.html" style="font-size: 8pt">non-bloodfed</a>&nbsp;&nbsp;
      <a href="tags/nucleic_acids.html" style="font-size: 8pt">nucleic acids</a>&nbsp;&nbsp;
      <a href="tags/nucleus.html" style="font-size: 8pt">nucleus</a>&nbsp;&nbsp;
      <a href="tags/object_oriented_programming.html" style="font-size: 8pt">object oriented programming</a>&nbsp;&nbsp;
      <a href="tags/open_science.html" style="font-size: 20pt">open science</a>&nbsp;&nbsp;
      <a href="tags/open_source.html" style="font-size: 12pt">open source</a>&nbsp;&nbsp;
      <a href="tags/pip.html" style="font-size: 8pt">pip</a>&nbsp;&nbsp;
      <a href="tags/programming.html" style="font-size: 10pt">programming</a>&nbsp;&nbsp;
      <a href="tags/protein.html" style="font-size: 8pt">protein</a>&nbsp;&nbsp;
      <a href="tags/pyqt.html" style="font-size: 8pt">PyQt</a>&nbsp;&nbsp;
      <a href="tags/python.html" style="font-size: 10pt">Python</a>&nbsp;&nbsp;
      <a href="tags/python.html" style="font-size: 16pt">python</a>&nbsp;&nbsp;
      <a href="tags/python_gotcha.html" style="font-size: 8pt">python gotcha</a>&nbsp;&nbsp;
      <a href="tags/r.html" style="font-size: 10pt">R</a>&nbsp;&nbsp;
      <a href="tags/research.html" style="font-size: 8pt">research</a>&nbsp;&nbsp;
      <a href="tags/research_blogging.html" style="font-size: 8pt">Research Blogging</a>&nbsp;&nbsp;
      <a href="tags/rna.html" style="font-size: 10pt">RNA</a>&nbsp;&nbsp;
      <a href="tags/rna_seq.html" style="font-size: 18pt">RNA-seq</a>&nbsp;&nbsp;
      <a href="tags/science_twitter.html" style="font-size: 8pt">science@twitter</a>&nbsp;&nbsp;
      <a href="tags/selection.html" style="font-size: 8pt">selection</a>&nbsp;&nbsp;
      <a href="tags/sequence_analysis.html" style="font-size: 8pt">sequence analysis</a>&nbsp;&nbsp;
      <a href="tags/setup_py.html" style="font-size: 8pt">setup.py</a>&nbsp;&nbsp;
      <a href="tags/sip.html" style="font-size: 8pt">SIP</a>&nbsp;&nbsp;
      <a href="tags/the_central_dogma.html" style="font-size: 8pt">the central dogma</a>&nbsp;&nbsp;
      <a href="tags/tinkerer.html" style="font-size: 8pt">tinkerer</a>&nbsp;&nbsp;
      <a href="tags/tophat.html" style="font-size: 12pt">tophat</a>&nbsp;&nbsp;
      <a href="tags/transcription.html" style="font-size: 8pt">transcription</a>&nbsp;&nbsp;
      <a href="tags/transcriptomics.html" style="font-size: 8pt">transcriptomics</a>&nbsp;&nbsp;
      <a href="tags/transgenic_mosquitoes.html" style="font-size: 10pt">transgenic mosquitoes</a>&nbsp;&nbsp;
      <a href="tags/translation.html" style="font-size: 8pt">translation</a>&nbsp;&nbsp;
      <a href="tags/trinity.html" style="font-size: 8pt">Trinity</a>&nbsp;&nbsp;
      <a href="tags/twitter.html" style="font-size: 8pt">twitter</a>&nbsp;&nbsp;
      <a href="tags/vector_control.html" style="font-size: 10pt">vector control</a>&nbsp;&nbsp;
      <a href="tags/xpermutations_py.html" style="font-size: 8pt">xpermutations.py</a>
</div></section></aside></div> <!-- #main --></div> <!-- #main-container -->

        <div class="footer-container"><footer class="wrapper">&copy; Copyright 2013, W Augustine Dunn. Powered by <a href="http://www.tinkerer.me/">Tinkerer</a> and <a href="http://sphinx.pocoo.org/">Sphinx</a>.</footer></div> <!-- footer-container -->

      </div> <!--! end of #container --><script type="text/javascript">    var disqus_shortname = "movingthelamppost";    disqus_count();</script><!--[if lt IE 7 ]>
          <script src="//ajax.googleapis.com/ajax/libs/chrome-frame/1.0.3/CFInstall.min.js"></script>
          <script>window.attachEvent('onload',function(){CFInstall.check({mode:'overlay'})})</script>
        <![endif]-->
    </body>
</html>