<?xml version="1.0" encoding="utf-8"?>
<rss version="2.0">
    <channel>
        <title>moving the lamppost</title>
        <link>http://movingthelamppost.com/blog/html/</link>
        <description>random musings of a molecular biologist turned code jockey in the era of big data and open science.</description>
        <language>en-us</language>
        <pubDate>Thu, 04 Jul 2013 00:00:00 -0700</pubDate>
        
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/07/04/first_pass_trinity_results_for_anopheles_stephensi_midgut_rna_seq__non_bloodfed_.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/07/04/first_pass_trinity_results_for_anopheles_stephensi_midgut_rna_seq__non_bloodfed_.html</guid>
            <title><![CDATA[First pass Trinity Results for Anopheles stephensi midgut RNA-seq (non-bloodfed)]]></title>
            <description><![CDATA[<div class="section" id="first-pass-trinity-results-for-anopheles-stephensi-midgut-rna-seq-non-bloodfed">
<h1>First pass Trinity Results for Anopheles stephensi midgut RNA-seq (non-bloodfed)</h1>
<div class="section" id="exploration-of-the-first-set-of-results-from-a-trinity-run-with-default-settings-on-one-library-of-anopheles-stephensi-midgut-rna-seq-as-00-lib-one">
<h2>Exploration of the first set of results from a Trinity run with default settings on one library of Anopheles stephensi midgut RNA-seq (As_00 lib one)</h2>
<div class="section" id="analyze-lengths-of-assembled-contigs-transcripts">
<h3>Analyze Lengths Of Assembled Contigs/Transcripts</h3>
<p>Function to count all transcripts of length <em>X</em>:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="kn">from</span> <span class="nn">collections</span> <span class="kn">import</span> <span class="n">defaultdict</span>
<span class="kn">from</span> <span class="nn">rSeq.utils.files</span> <span class="kn">import</span> <span class="n">ParseFastA</span>

<span class="k">def</span> <span class="nf">measure_and_count_transcripts</span><span class="p">(</span><span class="n">fasta_path</span><span class="p">):</span>

    <span class="n">tx_lens</span> <span class="o">=</span> <span class="n">defaultdict</span><span class="p">(</span><span class="nb">list</span><span class="p">)</span>

    <span class="n">p</span> <span class="o">=</span> <span class="n">ParseFastA</span><span class="p">(</span><span class="n">fasta_path</span><span class="p">)</span>
    <span class="n">seqs</span> <span class="o">=</span> <span class="n">p</span><span class="o">.</span><span class="n">to_dict</span><span class="p">()</span>

    <span class="k">for</span> <span class="n">name</span><span class="p">,</span><span class="n">seq</span> <span class="ow">in</span> <span class="n">seqs</span><span class="o">.</span><span class="n">iteritems</span><span class="p">():</span>
        <span class="n">tx_lens</span><span class="p">[</span><span class="nb">len</span><span class="p">(</span><span class="n">seq</span><span class="p">)]</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">name</span><span class="p">)</span>

    <span class="k">print</span> <span class="s">&quot;The number of different transcript sizes encountered: </span><span class="si">%s</span><span class="s">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">tx_lens</span><span class="p">))</span>
    <span class="k">return</span> <span class="n">tx_lens</span>
</pre></div>
</div>
<p>Now I count and plot the size data.</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="c"># Prep to plot a Histogram of the numbers of transcripts in each length bin</span>

<span class="n">trinity_tx_lens</span> <span class="o">=</span> <span class="n">measure_and_count_transcripts</span><span class="p">(</span><span class="s">'/home/gus/Dropbox/common/projects/trinity_As_one_lib/Trinity.fasta'</span><span class="p">)</span>

<span class="n">trinity_lengths</span> <span class="o">=</span> <span class="p">[]</span>

<span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">trinity_tx_lens</span><span class="o">.</span><span class="n">iteritems</span><span class="p">():</span>
    <span class="n">trinity_lengths</span><span class="o">.</span><span class="n">extend</span><span class="p">([</span><span class="n">item</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">*</span> <span class="nb">len</span><span class="p">(</span><span class="n">item</span><span class="p">[</span><span class="mi">1</span><span class="p">]))</span>

<span class="n">trinity_lengths</span><span class="o">.</span><span class="n">sort</span><span class="p">()</span>
</pre></div>
</div>
<div class="highlight-python"><pre>The number of different transcript sizes encountered: 4632</pre>
</div>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">hist</span><span class="p">(</span><span class="n">trinity_lengths</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">histtype</span><span class="o">=</span><span class="s">'stepfilled'</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">'transcript length'</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">'number of transcripts'</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x645b290&gt;</pre>
</div>
<img alt="../../../_images/_fig_03.png" src="http://movingthelamppost.com/blog/html/_images/_fig_03.png"/>
<div id="more"> </div><p>Now I calculate the N50*length* and N50*index* for the trinity assembly.</p>
<p><strong>N50*index*:</strong> the number of largest contigs(transcripts) whose summed
lengths equal at least 50% of the sum of <strong>ALL</strong> contigs (smaller
numbers are ‘better’)</p>
<p><strong>N50*length*:</strong> the length of the last added contig from above (larger
numbers are ‘better’)</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="k">def</span> <span class="nf">calc_N50s</span><span class="p">(</span><span class="n">lengths</span><span class="p">):</span>
    <span class="n">total_length</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">lengths</span><span class="p">)</span>
    <span class="k">print</span> <span class="s">&quot;total_length:</span><span class="se">\t\t</span><span class="si">%s</span><span class="s">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">total_length</span><span class="p">)</span>

    <span class="n">running_total</span> <span class="o">=</span> <span class="mi">0</span>
    <span class="n">collected_lengths</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">n50_i</span> <span class="o">=</span> <span class="bp">None</span>

    <span class="k">for</span> <span class="n">length</span> <span class="ow">in</span> <span class="nb">reversed</span><span class="p">(</span><span class="n">lengths</span><span class="p">):</span>
        <span class="n">running_total</span> <span class="o">+=</span> <span class="n">length</span>
        <span class="n">collected_lengths</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">length</span><span class="p">)</span>

        <span class="k">if</span> <span class="n">running_total</span> <span class="o">&gt;=</span> <span class="n">total_length</span> <span class="o">*</span> <span class="mf">0.5</span><span class="p">:</span>
            <span class="n">n50_i</span> <span class="o">=</span> <span class="nb">len</span><span class="p">(</span><span class="n">collected_lengths</span><span class="p">)</span>
            <span class="n">n50_l</span> <span class="o">=</span> <span class="n">length</span>
            <span class="k">break</span>

    <span class="k">print</span> <span class="s">&quot;running_total:</span><span class="se">\t\t</span><span class="si">%s</span><span class="s">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">running_total</span><span class="p">)</span>
    <span class="k">print</span> <span class="s">&quot;N50_index:</span><span class="se">\t\t</span><span class="si">%s</span><span class="s"> of </span><span class="si">%s</span><span class="s">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">n50_i</span><span class="p">,</span><span class="nb">len</span><span class="p">(</span><span class="n">lengths</span><span class="p">))</span>
    <span class="k">print</span> <span class="s">&quot;N50_length:</span><span class="se">\t\t</span><span class="si">%s</span><span class="s">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">n50_l</span><span class="p">)</span>
    <span class="k">print</span> <span class="s">&quot;median contig length</span><span class="se">\t</span><span class="si">%s</span><span class="s">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">median</span><span class="p">(</span><span class="n">lengths</span><span class="p">))</span>
</pre></div>
</div>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">calc_N50s</span><span class="p">(</span><span class="n">trinity_lengths</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><pre>total_length:               31858558
running_total:              15931068
N50_index:                  3927 of 25167
N50_length:                 2469
median contig length        647.0</pre>
</div>
</div>
<div class="section" id="analyze-lengths-of-predicted-transcripts">
<h3>Analyze Lengths Of Predicted Transcripts</h3>
<div class="section" id="astes1-0">
<h4>1. AsteS1.0</h4>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">asteS1_0_tx_lengths</span> <span class="o">=</span> <span class="n">measure_and_count_transcripts</span><span class="p">(</span><span class="s">'/home/gus/genome_data/AsteS/Anopheles-stephensi-SDA-500_TRANSCRIPTS_AsteS1.0.fa'</span><span class="p">)</span>

<span class="n">asteS1_0_lengths</span> <span class="o">=</span> <span class="p">[]</span>

<span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">asteS1_0_tx_lengths</span><span class="o">.</span><span class="n">iteritems</span><span class="p">():</span>
    <span class="n">asteS1_0_lengths</span><span class="o">.</span><span class="n">extend</span><span class="p">([</span><span class="n">item</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">*</span> <span class="nb">len</span><span class="p">(</span><span class="n">item</span><span class="p">[</span><span class="mi">1</span><span class="p">]))</span>

<span class="n">asteS1_0_lengths</span><span class="o">.</span><span class="n">sort</span><span class="p">()</span>
</pre></div>
</div>
<div class="highlight-python"><pre>The number of different transcript sizes encountered: 4058</pre>
</div>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">hist</span><span class="p">(</span><span class="n">asteS1_0_lengths</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">histtype</span><span class="o">=</span><span class="s">'stepfilled'</span><span class="p">,</span><span class="n">color</span><span class="o">=</span><span class="s">'c'</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">'transcript length'</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">'number of transcripts'</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x3f2ced0&gt;</pre>
</div>
<img alt="../../../_images/_fig_08.png" src="http://movingthelamppost.com/blog/html/_images/_fig_08.png"/>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">calc_N50s</span><span class="p">(</span><span class="n">asteS1_0_lengths</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><pre>total_length:               25155329
running_total:              12580251
N50_index:                  2617 of 13251
N50_length:                 2838
median contig length        1344.0</pre>
</div>
</div>
<div class="section" id="astei1-0">
<h4>2. AsteI1.0</h4>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">asteI1_0_tx_lengths</span> <span class="o">=</span> <span class="n">measure_and_count_transcripts</span><span class="p">(</span><span class="s">'/home/gus/genome_data/AsteI/Anopheles-stephensi-Indian_TRANSCRIPTS_AsteI1.0.fa'</span><span class="p">)</span>

<span class="n">asteI1_0_lengths</span> <span class="o">=</span> <span class="p">[]</span>

<span class="k">for</span> <span class="n">item</span> <span class="ow">in</span> <span class="n">asteI1_0_tx_lengths</span><span class="o">.</span><span class="n">iteritems</span><span class="p">():</span>
    <span class="n">asteI1_0_lengths</span><span class="o">.</span><span class="n">extend</span><span class="p">([</span><span class="n">item</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">*</span> <span class="nb">len</span><span class="p">(</span><span class="n">item</span><span class="p">[</span><span class="mi">1</span><span class="p">]))</span>

<span class="n">asteI1_0_lengths</span><span class="o">.</span><span class="n">sort</span><span class="p">()</span>
</pre></div>
</div>
<div class="highlight-python"><pre>The number of different transcript sizes encountered: 5227</pre>
</div>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">hist</span><span class="p">(</span><span class="n">asteI1_0_lengths</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">histtype</span><span class="o">=</span><span class="s">'stepfilled'</span><span class="p">,</span><span class="n">color</span><span class="o">=</span><span class="s">'g'</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">'transcript length'</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">'number of transcripts'</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x6d37710&gt;</pre>
</div>
<img alt="../../../_images/_fig_13.png" src="http://movingthelamppost.com/blog/html/_images/_fig_13.png"/>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">calc_N50s</span><span class="p">(</span><span class="n">asteI1_0_lengths</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><pre>total_length:               35859205
running_total:              17930131
N50_index:                  3231 of 23287
N50_length:                 3279
median contig length        717.0</pre>
</div>
</div>
</div>
</div>
</div>]]></description>
            <category><![CDATA[ My Research ]]></category>
             <pubDate>Thu, 04 Jul 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/06/29/speak__for_the_cup_of_hemlock_is_not_yet_on_your_lips.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/06/29/speak__for_the_cup_of_hemlock_is_not_yet_on_your_lips.html</guid>
            <title><![CDATA[Speak, for the Cup of Hemlock is not yet on Your Lips]]></title>
            <description><![CDATA[<div class="section" id="speak-for-the-cup-of-hemlock-is-not-yet-on-your-lips">
<h1>Speak, for the Cup of Hemlock is not yet on Your Lips</h1>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>I am not eager to turn this blog toward religious/athiest discussions.
I want it to avoid the distractions that flame wars and endless volleys of <em>“Nu-Uh!!!”</em> and <em>“Ya-hunh!!”</em> inevitably bring.
However, this story needs to be told.</p>
<p>It is the horrible story of Bangladeshi atheist blogger Asif Mohiuddin, and how his government has silenced him for <em>“hurting religious beliefs”</em>.
He has been detained without formal charges since the 3rd of April, and his blog shutdown.
His health is failing, and this is a show of solidarity to him and all other voices that dare to risk their freedom and, as you will see, their very lives to speak what they believe to be the truth.</p>
<p>There has been a call to repost his final entry before being silenced by his government.
I am participating.
It is an act that symbolizes the futility of governments silencing dissent in this age.
The interconnectivity of the internet is, if nothing else, a powerful modern embodiment of <a class="reference external" href="http://en.wikipedia.org/wiki/Lernaean_Hydra">the Lernaean Hydra</a>.
This is the amplification of the voice they tried to silence.
This is the failure of containing the situation.
This is more attention and anger being focused on his complaints rather than less.
This is what a global distributed network does when a whisper in one tiny corner of its web gets silenced.</p>
<p class="last"><strong>That barely noticeable whisper gets 10,000 more voices.</strong></p>
</div>
<div class="admonition attention">
<p class="first admonition-title">Attention</p>
<p class="last">The words that follow <strong>are not my own</strong>.
I am posting a liberally excerpted text from a post by the <a class="reference external" href="http://iheu.org/story/speak-cup-hemlock-not-yet-your-lips-jailed-atheist-blogger-asif-mohiuddin">International Humanist and Ethical Union</a> that gives some background and includes Mohiuddin’s final post titled: <strong>Speak, for the Cup of Hemlock is not yet on Your Lips</strong>.</p>
</div>
<p>Bangladeshi atheist blogger Asif Mohiuddin has been in prison since 3 April for “hurting religious beliefs”.</p>
<p>The formal charges and a trial are still pending. A court order to release him on bail due to deteriorating health has at last been issued, but not yet enacted.</p>
<p>Below, we republish one of Asif’s final blog posts before his imprisonement.</p>
<div class="figure">
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/asif-mohiuddin.jpg"><img alt="../../../_images/asif-mohiuddin.jpg" src="http://movingthelamppost.com/blog/html/_images/asif-mohiuddin.jpg" style="width: 300px;"/></a>
<p class="caption"><strong>Asif Mohiuddin</strong></p>
</div>
<p>The blog was published originally in Bangla at Mukto-Mona.com, who have also provided this English translation by Allachalaina. Mukto-Mona describes Asif Mohiuddin as being known “as one of the most outspoken atheist and humanist bloggers of Bangladesh. His writing—which was heavily critical of religious dogma, bigotry and superstition—and his political activism including the Jagannath University protests angered the government, marked the beginning of the threats he received from fundamentalists.”</p>
<p>Published on March 8, 2013 this was his last post at Mukto-Mona Banglablog under the title ... “Speak, for the cup of hemlock is not yet on your lips”. The post contains some graphic images below.</p>
<div id="more"> </div><div class="section" id="id1">
<h2>Speak, for the Cup of Hemlock is not yet on Your Lips</h2>
<p><em>by Asif Mohiuddin. Translation by Allachalaina.</em></p>
<p>I was traveling to work to start a long night shift that night. I was traveling by a three-wheeler and from distance I saw three men standing at the gate of my office-building, not a very unusual sight. Therefore, not pay much attention to them I went to pay the fare of the three-wheeler. I was facing back at them. Then, suddenly from behind I felt a hammer-blow at the back of my head. It was extremely painful and debilitating. Before I could pay much attention to that pain, I felt a stranglehold locking my head tight. Immediately afterwards I felt a bunch of stabbing wounds on my back. Terrorists usually fail to hide their excitement during an attack, they usually swear and shout. But, these terrorists were exception, they did all the things in complete silence without saying a word, not even the signature “Allahu Akbor”.</p>
<p>By that time I had already suffered a hail of stabbing from knives and machetes on my back and neck. Every single of those stabs were pointed vertically downwards on my body with an apparent objective of nailing the entire knife inside. Later from my doctor I came to know that two of the wounds in my neck were the most severe; one 6 centimeter deep and the other 4 centimeter; both of which missed my spinal cord just by half an inch.</p>
<blockquote>
<div>my entire body and the spot where the attack took place were completely drench in blood</div></blockquote>
<p>Meanwhile the man holding me in stranglehold was trying to slit my throat. I understood if he successfully severs my carotid artery I would have no chance. And this understanding probably made me to fight back. I employed the entire strength of my body and got out of the stranglehold repulsing the attackers. One of the attackers came back and blew his machete to my abdomen, but luckily I was able to dodge it. Then realizing that I will also be fighting back, the attackers fled and ran away into the darkness. A very important thing to notice in this attack is- the attackers were not robbers at all; they did not come for my mobile phone, my tablet or my wallet; they just came for my life. This observation confirmed my suspicion that it was an Islamist attack.</p>
<p>My entire body and the spot where the attack took place were completely drenched in blood. I was bleeding badly; still, I was trying my best not to collapse. I realized I must live. This sheer will to live out-shadowed my sensation of pain and injury. After a minute or two a couple of men came by from a nearby shop. I requested one of them to find my spectacles which I lost midst the attack. After getting my glasses back I asked one of them where is the nearest hospital. Luckily there was a small clinic just next to the main road, very nearby. Enforced with the aforesaid ‘sheer will to live’ I ran to that clinic.</p>
<blockquote>
<div>to my shock and surprise I found nobody did anything</div></blockquote>
<p>As I entered the clinic I shouted, “I am loosing blood at a shocking rate, please stop my bleeding, or else I will die”. But, to my shock and surprise I found nobody did anything, they rather told me to wait for the police to come and that they can treat me only in the presence of the police. I do not know, probably they thought me being an outlaw of some sort caught in a street battle or something! Anyway finally the police arrived and my wounds were washed. The clinic decided that my wounds needed to be treated surgically. So, they transferred me to the nearby Monsur Ali hospital, a surgical facility. On my way to that hospital I called my sister and fellow blogger Baki Billah and Ananya Azad, I informed them that I have been attacked.</p>
<p>Immediately the news of me being attacked was posted in the ‘Somewherein’ blog. Some of the bloggers gathered in the Monsur Ali Hospital. I saw blogger Sharat talking on his mobile crying frantically. Also Selim Anwar, Mahbub Rashid, Shafiul Joy, Anindya, Ananya Azad and my nephew and brother in law was there. Doctors stapled the wounds in my body closed. However the two wounds in by neck were too deep to be stapled. I remember a doctor pushing his finger inside those wounds in order to assess the depth of the wound and saying ‘probably we will not be able to save this patient for the wound is so deep that it can house the whole length of my finger inside’! Doctors decided to transfer me to a bigger hospital, Dhaka Medical College Hospital. We were waiting for the ambulance. Ananya Azad was crying frantically and suddenly out of emotion kissed me very brotherly. In return I tried my best to smile at him.</p>
<p>Finally the ambulance arrived and we boarded and set off. My nephew Anindya was trying to keep me awake and was holding my head tightly with his hands. Later I came to know this was for my spinal cord which was at significant risk, if the road bumps and the vibration of the ambulance could injure my spinal cord that would have resulted in permanent paralysis of parts or whole of my body. By this time my sisters Mimu, Moni, Chanda, Chandana and Mumu reached Dhaka Medical College Hospital, awaiting our arrival.</p>
<p>Finally we arrived in the Dhake Medical College Hospital to find Moniruddin Topu, Faruk Wasif, Mahbub Shakil already there. They were making arrangements and were preparing for our arrival. There I met Dr. Imran for the first time. He told me he reads my writings regularly and knowing the news of me being stabbed from Facebook has rushed to the hospital to see me.</p>
<blockquote>
<div>I was concerned whether I was going to live or not</div></blockquote>
<p>I was sent into the X-ray chamber. By the time I came back from there, a little crowd of people had gathered in my hospital ward. I can not remember very clearly for the sedatives, but there were Mahbuh Rashid, Babu Ahmed, Taosif Hamim, Badhan Shapnyakathak and a lot others. Albeit the strong sedatives, I was trying to listen to their conversation in order for figuring out the seriousness of my condition. I was concerned whether I was going to live or not. The conversation I heard was not very positive. The realization that there is a significant probability of me dying, suddenly made me feel that I needed a cigarette. To that little crowd I casually communicated this supposed last wish of mine. But, unfortunately nobody seemed very much keen to fulfill this last deathbed wish of mine; ‘a cigarette’ seemed a rather bizarre and completely unacceptable deathbed wish to them for some reasons!</p>
<div class="figure">
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/asif-hospital.jpg"><img alt="../../../_images/asif-hospital.jpg" src="http://movingthelamppost.com/blog/html/_images/asif-hospital.jpg" style="width: 300px;"/></a>
<p class="caption"><strong>In Hospital</strong></p>
</div>
<p>Before beginning my treatment the doctors had to get a consent form signed by my sister who was my next to keen. That form explicitly stated- if the course of my treatment results in my death not due to any criminal or medical negligence but due to the very natural chain of events, then the doctors can not be made legally liable. After this Dr. Pratap, another blogger doctor who goes by the pseudonym ‘incognito’ and a lady doctor started my treatment. First, it was the initial assessment of my medical condition. Here again the gloved finger of Dr. Pratap penetrated the two deep wounds in my neck. This was extremely painful because my neck by that time had inflamed and swelled badly. I was screaming in pain and hearing that blood chilling screams of mine, I later came to know, Babu Ahmed and my nephew Himalaya fainted outside my hospital ward. The result of the initial assessment was again not very optimistic. Therefore, the doctors made my sister sign another acknowledgement form which stated- my medical condition is particularly vulnerable and surgical treatment has significant possibility of resulting in my death. I can picture how difficult it was for my beloved sister to sign a form of this sort! Soon after the initial assessment I was taken to the surgical chamber and was sedated again. The last thing I can remember before losing consciousness is the the anxious glances of Dr. Imran, Dr. Pratap and that lady doctor pouring over me.</p>
<p>In the morning when I regained my consciousness I was informed that my lover is here to see me, however she is not allowed in, she is waiting outside. I was very weak that time, yet with all my energy I yelled ‘let her come in’. She entered crying frantically and held my hand. It was that moment when I finally realized my life has been saved; I am not going to die, not this time! It took 70 stitches to staple my wounds close. Catheters were placed in everywhere in my body. I was sedated and badly disoriented and needless to say extremely extremely weak. Yet I felt as refreshed as ever when my girlfriend was by my side.</p>
<blockquote>
<div>I never lost faith in humanity nor will I ever</div></blockquote>
<p>I had been helped by a lot of friends and well-wishers to whom I have a great deal of thanks to say. At some point in my stay in the hospital I discovered the bed behind me in the ward that I was staying was vacant. I could not figure how come a bed in the busiest hospital in Bangladesh can ever remain vacant. I never knew who it was but later I came to know some good Samaritan paid for that vacant seat and made sure it remains vacant so that I do not have to share the hospital ward with someone else! I never lost faith in humanity nor will I ever  - the men who assaulted me for the sole purpose of killing me were human beings. On the other hand, the men who spent their entire sleepless nights looking after me, working hard to save my life praying and crying, were also human beings. In fact this endless stream of love of them is what hijacked my soul back from the black grips of death and restored it in the living earth, ever colorful, once again. They were countless in number- laymen, bloggers, well-wishers and what not! Some of them came and talked to me crying; while some others came to see that I was still alive and breathing, stood far and remained standing not saying a word. I can not remember all of them for the anesthetics induced amnesia. But I do know, cured when i see them again I will be struggling to find appropriate words for expressing my thankfulness towards them; no word I know could be appropriate, no word enough! At some point the doctors and nurses were forced to say that if not the almost mortal wounds, then this endless stream of visitors and well-wishers is going to kill me; for they were thousands in number. This large crowd of visitors also made them figure quite erroneously that I must be some very influential person which I certainly was not!</p>
<p>Although I had been helped by a lot of friends; but unfortunately some other body was not so helpful nor so friendly! One of the most circulated daily newspaper named Prothom Alo which claims itself to be a progressive daily, very tactfully omitted my name from being published in one of its news cover about the Deutsche Welle award. Although they published the names of other winners, they did not publish my name despite me being an award winner. They also published the news of me being attacked as a mere ‘robbery’ as opposing to an ‘Islamist fundamentalist attack’. Also they published that news in one of the least visited corners of the newspaper as opposing to the first page headlines. In a country like Bangladesh being an outspoken atheist is not really a very good idea it seems!</p>
<blockquote>
<div>blogger Rajib Hayder was not so lucky as me</div></blockquote>
<p>I was attacked but I was lucky, I did not have to die. But, blogger Rajib Hayder was not so lucky as me. Rajib was attacked later that month. His throat was slashed, the attackers left him only after ensuring his death. Those senseless attackers of Rajib had no quarrel with him, they did not even know him. Still they decided to attack and kill him because they figured his writing is somehow challenging and threatening the all powerful position of Allah and the only true religion called Islam! They figured this challenge should be countered with knives and machetes for the holy purpose of the protection of almighty Allah and the holiest religion Islam. Although they believe Allah to be all powerful, omnipresent and omnipotent- they still figured he needs some protection from the literary assault of an innocent writer who writes against what he considers to be bigoted, wrong, poisonous. Although the weapon of that alleged assault of Rajib on their all powerful god was nothing more than pen and paper; the weapons with which those silly soldiers of Allah decided to counter that assault was much more dangerous, namely knives and machetes. They slashed Rajib’s throat saying ‘Allaho Akbor’. ‘god is great’ and in doing so thought they had successfully protected Allah and Islam from a very very dreadful threat! And what those threats were? Well, some articles written by an outspoken atheist who does not want to told how he should live his life! However, if my reaction upon hearing these horrible sets of news becomes- ‘mad, barbaric, ignorant’, fingers would be pointed at me; my reaction will be considered more offensive than the actual act of the slashers of Rajib’s throat and how can I not feel enraged, how can not feel helpless; how any sensible person can not?</p>
<div class="figure">
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/asif-injuries.jpg"><img alt="../../../_images/asif-injuries.jpg" src="http://movingthelamppost.com/blog/html/_images/asif-injuries.jpg" style="width: 300px;"/></a>
<p class="caption"><strong>Asif’s Injuries</strong></p>
</div>
<p>I do not know whether I am going to be attacked again. So far all the alleged hit-lists published by Islamists contain my name, it is very possible that I am going to be attacked again. Humayun Azad once said - “Speak, for the cup of hemlock is not yet on your leaps”. Therefore, I will keep speaking, I will be writing as long as I am alive, as long as the cup of hemlock is not pushed to my lips.</p>
<blockquote>
<div>I never block any religious fundamentalist from my Facebook ... the doors of thoughtfulness should never be closed</div></blockquote>
<p>To me death is a very natural thing. I am therefore prepared to accept it as naturally as I can. It has been years since I started writing against religious fundamentalism. In this long struggle I have made a lot of friends, comrades; quite naturally have made a lot of enemies as well. I have repeatedly seen ideological enmity turning to the personal one. I also may become a victim of this enmity of nasty sort. I know how to take criticism and know how to respect the freedom of speech of every individual. That is why I never block any religious fundamentalist from my Facebook although they very often threaten me with death and call names. I do this because I believe the doors of thoughtfulness should never be closed, the calls of taking the path of progressiveness should never be barred. If I block them from my Facebook, if I bar them from reading what I have to write, then probably they will remain ever deprived of that very call that I believe every individual should hear and respond to. I have no hatred towards the people who tried to kill me. This is because I know they do not know what they are doing, they are senseless, at least for the time being. They have been taught and trained in this way, they have been brainwashed into hating and killing. My attempted murderers be brought into trial and punished is not the first thing that I want. I rather want this senseless practice of indoctrination and brainwashing be abolished someday. Only when this happens I would consider my attempted murderers have been punished and justice been done.</p>
<p>I have a humble request to make to you all. I do not know how long am I going to survive this struggle against religious fundamentalist bigotry. However, if I fall, please make sure nobody successfully claims that I ever was a theist, or I ever in anyway was respectful towards any superstitious middle age beliefs commonly known as religion. I do not want to be defeated by the superstitions against which I spent my entire life fighting, not even after my death. I do not want any funeral or any sort of religious service for me after my death. I lived my life with my ideals and I want to die with them too; I do not want death to separate me from the ideology that I upheld throughout all my life with all my efforts. I have always been outspoken as opposing to being passive. I do not want death to distort my true face to any comparatively more politically correct and generally acceptable form. I am what I always have been and I want to remain so even after my death.</p>
<blockquote>
<div>I love this planet so very much. And therefore I fight for it and am ready to die for it</div></blockquote>
<p>And please make sure my corpse is donated to a medicine school. My eyes and kidneys and every other transplantable organs are to be donated to the people who need them the most. What can be more joyful than the thought that my deceased eyes will enable a blind person in seeing again, my deceased heart will be pumping blood into the arteries of a sick person? I want the rest of my body, my bones and muscles, be donated to a medicine school so that the pupils there can master surgery by cutting incisions on my deceased skin. I never believed in heaven and hell. I rather believed in earth, the planet where we all make our stand. I love this planet so very much. And therefore I fight for it and am ready to die for it, for it being the way I always dream it to be. In this hard battle I do not want to surrender, nor do I want to retreat.</p>
</div>
</div>]]></description>
             <pubDate>Sat, 29 Jun 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/06/14/arthropod_genomics_symposium_2013____notre_dame.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/06/14/arthropod_genomics_symposium_2013____notre_dame.html</guid>
            <title><![CDATA[Arthropod Genomics Symposium 2013 -- Notre Dame]]></title>
            <description><![CDATA[<div class="section" id="arthropod-genomics-symposium-2013-notre-dame">
<h1>Arthropod Genomics Symposium 2013 – Notre Dame</h1>
<p>My notes on talks from the <a class="reference external" href="http://globalhealth.nd.edu/7th-annual-arthropod-genomics-symposium">7th Annual Arthropod Genomics Symposium</a>  If you are interested, you can follow live tweets from <a class="reference external" href="https://twitter.com/search?q=%2313arthgen&amp;src=typd">#13ArthGen</a>.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<ol class="arabic simple">
<li><strong>COMING SOON:</strong> links to or embedded versions of the recorded talks for each section. These will be added as soon as I can get the location of where they will be hosted.</li>
<li><strong>My Spelling:</strong> Obviously, my spelling is gonna suck on this.  Please don’t hold it against me.</li>
<li><strong>Linking to Specific Sections:</strong> If you want to link to a specific person’s notes you can use ‘anchor’ notation as illustrated here for <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/06/14/arthropod_genomics_symposium_2013____notre_dame.html#jennifer-brisson"><em>Jennifer Brisson</em></a>:</li>
</ol>
<div class="last highlight-none"><div class="highlight"><pre>http://movingthelamppost.com/blog/html/2013/06/14/arthropod_genomics_symposium_2013____notre_dame.html#jennifer-brisson
</pre></div>
</div>
</div>
<div class="section" id="general-session">
<h2>General Session</h2>
<div class="section" id="dan-lawson-ebi">
<span id="dan-lawson"/><h3>Dan Lawson (EBI)</h3>
<ul class="simple">
<li>Federated curation and hosting of genomes plus standardized best practices and data formats needed for #i5k</li>
</ul>
<hr class="docutils"/>
</div>
<div class="section" id="steven-fringy-richards-baylor-college-of-medicine">
<span id="steven-richards"/><h3>Steven “Fringy” Richards (Baylor College of Medicine)</h3>
<ul class="simple">
<li>ND50 10kb basic bar to reach for good annotation through Maker</li>
</ul>
<hr class="docutils"/>
</div>
<div class="section" id="terence-murphy-ncbi">
<span id="terence-murphy"/><h3>Terence Murphy (NCBI)</h3>
<p><em>How to annotate 5000 genomes: lessons from NCBI genome annotation pipeline</em></p>
<hr class="docutils"/>
</div>
<div class="section" id="monica-munoz-torres-lawrence-berkeley-national-laboratory">
<span id="monica-munoz-torres"/><h3>Monica Munoz-Torres (Lawrence Berkeley National Laboratory)</h3>
<p><em>Web Apollo web based annotation editing platform</em></p>
<ul class="simple">
<li>first realtime genome collaborative annotation editor on the web</li>
<li>MANUAL annotation is NECESSARY not just desirable: without it many downstream experiments will be poisoned</li>
<li>democratization of genome assembly and annotation require a new model of collaborative annotation</li>
</ul>
<div id="more"> </div></div>
</div>
<hr class="docutils"/>
<div class="section" id="emerging-genomes">
<h2>Emerging Genomes</h2>
<div class="section" id="amy-toth">
<span id="id1"/><h3>Amy Toth</h3>
<p><em>Genome of primitively social wasp “Polistes dominula”: insights/opportunity for understanding genomic basis of eusociality</em></p>
<ul>
<li><p class="first">low repetitive elements</p>
</li>
<li><p class="first">even lower TE content</p>
</li>
<li><dl class="first docutils">
<dt>genomic suspects for social evolution</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>genetic toolkits – conserved gene sets</dt>
<dd><ul class="first last">
<li><p class="first">evo-devo approach –comparative</p>
</li>
<li><dl class="first docutils">
<dt>conservation of Hox genes etc</dt>
<dd><ul class="first last simple">
<li>cis regulatory changes very important as changes in expression vs gene function important in evolution/speciation</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>transcriptomic look at</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>foraging</dt>
<dd><ul class="first last simple">
<li>foraging transcriptome profiles agree with apis mellifera</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>dominance</dt>
<dd><ul class="first last simple">
<li>compared transcriptomes of brains in dominant vs submissive individuals</li>
<li>found correlations of aggressive behavior in fruit flies and even mammals</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>epigenetics</dt>
<dd><ul class="first last simple">
<li>controls wild diff in phenotypic models – same genome gives VERY diff morphologies</li>
<li>cast related diff in DNA methylation are present</li>
<li>treatment with methyl inhibitor causes more worker like phenotypes</li>
<li>however dna meth not likely to be related to cast diff through diff expression, but maybe through splicing control</li>
<li>Dnmt3 (usually needed for DNA meth) not found yet in this genome</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">novel genes</p>
</li>
</ul>
</dd>
</dl>
</li>
</ul>
<hr class="docutils"/>
</div>
<div class="section" id="dave-denlinger-ohio-state">
<span id="dave-denlinger"/><h3>Dave Denlinger (Ohio State)</h3>
<p><em>Sequencing the Antarctic Midge, Belgica antarctica: the smallest insect genome</em></p>
<ul>
<li><dl class="first docutils">
<dt>poly extremophile</dt>
<dd><ul class="first last simple">
<li>cold – frozen solid 8-9 months a year</li>
<li>70% loss of body water</li>
<li>salinity</li>
<li>pH</li>
<li>UV EXPOSURE</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Heat shock proteins on almost constitutively</p>
</li>
<li><p class="first">Dessication 30% helps them survive low temps</p>
</li>
<li><p class="first">initial genome size of ~ 85 Mb?! – then confirmed</p>
</li>
<li><p class="first">13,500 protein encoding genes anyway</p>
</li>
<li><dl class="first docutils">
<dt>uniqueness of the genome:</dt>
<dd><ul class="first last">
<li><p class="first">small size – probly due to extremophile life</p>
</li>
<li><p class="first">consistently small introns</p>
</li>
<li><p class="first">few repetitive elements</p>
</li>
<li><p class="first">few TEs – usually old and probably inactive</p>
</li>
<li><p class="first">47% GC</p>
</li>
<li><dl class="first docutils">
<dt>function enrich:</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>up</dt>
<dd><ul class="first last simple">
<li>membrane</li>
<li>ATP binding</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>dwn</dt>
<dd><ul class="first last simple">
<li>oderant binding</li>
<li>oderant receptors</li>
<li>sensory perception of smell in general</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>pathway of dessication protection</dt>
<dd><ul class="first last simple">
<li>up regulation of autophagy</li>
<li>inhibit apoptosis</li>
<li>metabolism genes shutdown</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">lead to focus on aquaporins as related to dehydration responses in diff tissues</p>
</li>
<li><dl class="first docutils">
<dt>How does clock work in these extreme light/dark cycles</dt>
<dd><ul class="first last simple">
<li>clock genes not correlated with light cycle at all</li>
<li>locomotor activity shows no diurnal bias</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">genes still there and still expressed but the cyclic nature of transcription is NOT observed</p>
</li>
</ul>
<hr class="docutils"/>
</div>
<div class="section" id="virpi-ahola">
<span id="id2"/><h3>Virpi Ahola</h3>
<p><em>melitaea cinxia genome</em></p>
</div>
</div>
<hr class="docutils"/>
<div class="section" id="epigenetics">
<h2>Epigenetics</h2>
<div class="section" id="jennifer-brisson-u-of-neb-lincoln">
<span id="jennifer-brisson"/><h3>Jennifer Brisson (U of Neb - Lincoln)</h3>
<p><em>Linking DNA methylation to phenotype in the pea aphid</em></p>
<ul>
<li><p class="first">meth found in gene bodies primarily in exons</p>
</li>
<li><p class="first">Methylation associated with higher expression (only at global level?)</p>
</li>
<li><dl class="first docutils">
<dt>differences bt morphs:</dt>
<dd><ul class="first last simple">
<li>methyl patterns plainly morph specific by PCA 2D</li>
<li>intron specific CHH methylation flat in wingless/asex but very pronounced in Sexual morphs</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>function of intergenic DNA methylation:</dt>
<dd><ul class="first last">
<li><p class="first">skipped exons should have higher methylations bc CTCF cant bind the exon to slow down the machinerey to allow that exon to be recognized as NOT intron</p>
</li>
<li><p class="first">this IS detected by the speaker</p>
</li>
<li><dl class="first docutils">
<dt>role for DNA methylation in dcerning paralogs?</dt>
<dd><ul class="first last simple">
<li>example used is RNAi genes (Dicer etc)</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
</ul>
<hr class="docutils"/>
</div>
<div class="section" id="micheal-goodisman-ga-tech">
<span id="micheal-goodisman"/><h3>Micheal Goodisman (GA Tech)</h3>
<p><em>Function of DNA methylation in insects</em></p>
<ul>
<li><dl class="first docutils">
<dt>DNA methylation lost in some insects</dt>
<dd><ul class="first last simple">
<li>NO METHEYL IN DIPTERA?!  <em>Did I know that?</em></li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">What is the function of methyl in insects?</p>
</li>
<li><p class="first"><strong>Note:</strong> look up CpG_o/e</p>
</li>
<li><dl class="first docutils">
<dt>in Apis and Pea Aphid:</dt>
<dd><ul class="first last simple">
<li>uniform expression between conditions = high methylation</li>
<li>differential expression bt condition = low methylation</li>
<li>says its a strong repeating pattern</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>fire ant diploid vs haploid males:</dt>
<dd><ul class="first last simple">
<li>increased gene expression tends to track with increased DNAmeth</li>
<li>as variation in gene expression increases DNAmeth decreases</li>
<li>most DNAmeth seen in Haploid males... (not sure I got that right)</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Conservation of methylation and other epigenetic data (honey bee vs fire ant and Dmel [no DNAmeth but yes histone modification])</dt>
<dd><ul class="first last">
<li><p class="first">over all: within ants, correlatiuon of genes with DNAmeth are very high (r ~&gt;0.7)</p>
</li>
<li><p class="first">same in bees</p>
</li>
<li><p class="first">between ants/bees: still relatively high correlation (r ~ 0.6)</p>
</li>
<li><dl class="first docutils">
<dt>are DNAmeth in genes in ants/bees correleated with specific histone mods in Dmel:</dt>
<dd><ul class="first last simple">
<li>Answer is YES</li>
<li>no time to document which hist mods he showed</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Summary:</dt>
<dd><ul class="first last simple">
<li>phenoype specific genes are unmethylated</li>
<li>ubiquitously genes are methylated</li>
<li>DNAmeth associated with ploidy</li>
<li>Patterns of DNAmeth conserved within and bewtween species</li>
<li>DNAmeth tends to be associated with specific histone mods</li>
<li>DNAmeth tied to alternative splicing</li>
</ul>
</dd>
</dl>
</li>
</ul>
<hr class="docutils"/>
</div>
<div class="section" id="greg-hunt-purdue">
<span id="greg-hunt"/><h3>Greg Hunt (Purdue)</h3>
<p><em>Parent of origin effects in gene expression in honey bees</em></p>
<ul class="simple">
<li>hybrids between african and european bees sting intermediate of the parents (but if the father is African hybrids are more similar to african bees?)  <strong>NOT SURE I GOT THAT RIGHT</strong></li>
<li>Used multiple different mapping methods but didnt say how – <em>DONT LIKE THAT</em></li>
<li>Truethfully I dunno if I can talk about the rest because I can’t tell if his biases are real or due to not using the same mapper in all conditions...</li>
</ul>
<hr class="docutils"/>
</div>
<div class="section" id="susan-weiner-iowa-state-univ">
<span id="susan-weiner"/><h3>Susan Weiner (Iowa State Univ)</h3>
<p><em>DNA methylation in the primitively social wasp *Polistes dominulus*</em></p>
<ul>
<li><p class="first">primitavely eusocial = not morpholagical difference (queen/works look the same)</p>
</li>
<li><p class="first">independent origin of socialality? (Did i hear that right?)</p>
</li>
<li><p class="first">Polistes has way more DNAmeth than all other insects looked at</p>
</li>
<li><dl class="first docutils">
<dt>is there a bias to sites DNAmeth in caste?</dt>
<dd><ul class="first last simple">
<li>yes.... <strong>but</strong> PCA 2D was not <strong>THAT</strong> impressive</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Zebularine treatment (<em>inhibits DNAmeth</em>) in multiple replications and variations tend to cause individuals make them more worker-like (opposite as in honey bees)</p>
</li>
<li><p class="first">DNMT3 has not been found yet (DNAmethyltransferase i think)</p>
</li>
<li><p class="first">DNMT1 and 2 are there</p>
</li>
<li><dl class="first docutils">
<dt>question asked: was zebularine validated that it is working uniformly</dt>
<dd><ul class="first last simple">
<li>answer: working on that</li>
</ul>
</dd>
</dl>
</li>
</ul>
</div>
</div>
<hr class="docutils"/>
<div class="section" id="comparative-genomics">
<h2>Comparative Genomics</h2>
<div class="section" id="rob-waterhouse-univ-geneva-mit">
<span id="rob-waterhouse"/><h3>Rob Waterhouse (Univ Geneva/MIT)</h3>
<p><em>Orthology-based genome annotation and interpretation</em></p>
<ul>
<li><p class="first">where we have come since Dmel in early 2000s</p>
</li>
<li><p class="first">what is it you want from your genome BEFORE you start to make sure that you have the quality you need</p>
</li>
<li><dl class="first docutils">
<dt>once your annotation is done:</dt>
<dd><ul class="first last simple">
<li>how do you asses the completness of your annotation?</li>
<li>once complete, orthology becomes useful tool</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">orthoDB: most comprehensive source for orthology amoung arthropods (&gt;57 species)</p>
</li>
<li><dl class="first docutils">
<dt>Assesing completion:</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>BUSCOs -&gt; <strong>needs definition later</strong></dt>
<dd><ul class="first last">
<li><p class="first">orthos with single copy ortho in 90% of other species</p>
</li>
<li><dl class="first docutils">
<dt>expectation:</dt>
<dd><ul class="first last simple">
<li>most should be found in your assembly</li>
<li>most should be single copy</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>implementation:</dt>
<dd><ul class="first last">
<li><p class="first">blast</p>
</li>
<li><dl class="first docutils">
<dt>then:</dt>
<dd><ul class="first last simple">
<li>best BLAST regions</li>
<li>next-best BLAST</li>
<li>homology based gene predictions</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>example:</dt>
<dd><ul class="first last">
<li><p class="first">15 Mosquito species</p>
</li>
<li><p class="first">fewer than 10 BUSCOs in most Mosq species</p>
</li>
<li><dl class="first docutils">
<dt>Aedes:</dt>
<dd><ul class="first last simple">
<li>16 missing</li>
<li>244 multi copy</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Culex:</dt>
<dd><ul class="first last simple">
<li>47 missing</li>
<li>126 multi copy</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>also look at the length differentials between orthologs</dt>
<dd><ul class="first last simple">
<li>helps see whether the fragmentation of assembly is affecting your assembly/gene models</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>mapping: new feature for orthoDB</dt>
<dd><ul class="first last simple">
<li>allows new genomes to be mapped to current frozen set of orthologs</li>
<li>private interface if required</li>
<li>also allows you to predict how many of the BUSCOs are missing to decide whether to make current gene set public</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>orthology to infer gene function:</dt>
<dd><ul class="first last">
<li><p class="first">functional traits</p>
</li>
<li><p class="first">evolutionary traits</p>
</li>
<li><dl class="first docutils">
<dt><strong>CAVEAT EMPTOR</strong>:</dt>
<dd><ul class="first last simple">
<li>orthology does not strictly define function</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Added a <strong>BUNCH</strong> of cool extra information including synteny blocks and relative evolutionary rate</p>
</li>
<li><dl class="first docutils">
<dt><strong>if:</strong></dt>
<dd><ul class="first last simple">
<li>single copy</li>
<li>kept in most species</li>
<li>slow evolutionary rate</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt><strong>then:</strong></dt>
<dd><ul class="first last simple">
<li>functional assumptions are warranted</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
</ul>
<hr class="docutils"/>
</div>
<div class="section" id="chris-winchell-uc-berkley">
<span id="chris-winchell"/><h3>Chris Winchell (UC Berkley)</h3>
<p><em>Genome and germline of emerging genome of crustacean __Parhyale hawaiensis__</em></p>
<ul>
<li><dl class="first docutils">
<dt>BAC library:</dt>
<dd><ul class="first last simple">
<li>129 024 clones</li>
<li>median insert ~140 kb</li>
<li>aprox coverage 5X</li>
<li>70 have been seq’d: 53 genes found</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>germline <em>REPLACEMENT?!</em></dt>
<dd><ul class="first last simple">
<li>you can ablate the original developing germline cells and they will MAKE MORE!! (convert somatic cells to germline?)</li>
<li>very elegant transgenic demonstration of where the new germline cells are originating in the blastoderm</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first"><strong>not</strong> a “normal” event, but may be a failsafe for rare germline loss of function in development</p>
</li>
</ul>
<hr class="docutils"/>
</div>
<div class="section" id="michael-brewer-uc-berkley">
<span id="michael-brewer"/><h3>Michael Brewer (UC Berkley)</h3>
<p><em>Evolutionary transcriptomics associated with developmental color switching in an adaptive radiation of Hawaiian Tetragnatha spiders (Araneae: Tetragnathidae)</em></p>
<ul>
<li><p class="first">orb weavers that don’t weave orbs anymore and adopt purely active hunting lifestyle</p>
</li>
<li><dl class="first docutils">
<dt>2 ecomorph:</dt>
<dd><ul class="first last simple">
<li>green</li>
<li>maroon</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">green individuals switch to maroon and their diet changes upon maturity</p>
</li>
<li><p class="first">they can control whether/when they shift?</p>
</li>
<li><p class="first">can date the species due to isolation on the islands</p>
</li>
<li><dl class="first docutils">
<dt>METHOD:</dt>
<dd><ul class="first last simple">
<li>RNAseq</li>
<li>trinity assembly</li>
<li>annotate by blast</li>
<li>trinity MM ORF</li>
<li>RecipBestBlast</li>
<li>Something else: slide change...</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">... got caught up in other things ...</p>
</li>
<li><p class="first"><strong>He has many centipede/millipede RNA-seq libraries but they are being neglected in the community: if you are interested in the results from these libraries please contact him</strong></p>
</li>
</ul>
<hr class="docutils"/>
</div>
<div class="section" id="igor-sharakhov-virgina-tech">
<span id="igor-sharakhov"/><span id="pirna"/><h3>Igor Sharakhov (Virgina Tech)</h3>
<p><em>Ornganization and evolution of piRNAs in _Anopheles gambiae_</em></p>
<ul>
<li><p class="first">in aedes piRNAs bind mostly to GENES not transposable elements</p>
</li>
<li><dl class="first docutils">
<dt>tissue subcell localization:</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>Dmel:</dt>
<dd><ul class="first last simple">
<li>diff localization for each piRNA protein</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>composition of piRNA in Ag</dt>
<dd><ul class="first last">
<li><p class="first">aedes piRNA clusters overlap GENES</p>
</li>
<li><p class="first">most piRNAs in Ag seem to have signiture of Ping Pong amplification</p>
</li>
<li><dl class="first docutils">
<dt>Localization in Ag seems mostly localized with transposable elements not genes</dt>
<dd><ul class="first last simple">
<li>but on X many map with genes...</li>
</ul>
</dd>
</dl>
</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Genomic chromosomal localization of piRNA in Ag</dt>
<dd><ul class="first last">
<li><p class="first">they have Abs for PIWI1</p>
</li>
<li><p class="first">PIWI1 localization on chromo looks more like Dmel PIWI</p>
</li>
<li><p class="first">PIWI2 (aub) not succeeded</p>
</li>
<li><dl class="first docutils">
<dt>aedes: most piRNA clusters NOT near centromeres like Dmel</dt>
<dd><ul class="first last simple">
<li>mostly in euchromatin</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>Ag: most piRNA peaks in centromereic but many outside of centromeric</dt>
<dd><ul class="first last simple">
<li>however still mostly in heterochromatic</li>
<li>M and S forms look pretty similar in localization of clusters</li>
<li>again in X chromosome many more clusters mapping in euchromatin</li>
<li>M and S forms NOT as similar in X chromosome</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">aedes piRNAs clusters as much as 20% of genome</p>
</li>
</ul>
</dd>
</dl>
</li>
</ul>
</div>
</div>
<hr class="docutils"/>
<div class="section" id="systems-biology-population-genomics">
<h2>Systems Biology/Population Genomics</h2>
<div class="section" id="andy-clark-cornell">
<span id="andy-clark"/><h3>Andy Clark (Cornell)</h3>
<p><em>Population genomic and metabolic inference from a global diversity reference panel of _Drosophila melanogaster_</em></p>
<ul>
<li><p class="first">using flies as model for diabetes: insulin signaling and TOR</p>
</li>
<li><dl class="first docutils">
<dt>lines isofemale lines (&gt;90 lines):</dt>
<dd><ul class="first last simple">
<li>Ithica</li>
<li>Netherlands</li>
<li>Beijing</li>
<li>Zimbabwe</li>
<li>Tasmania</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">significant metabolic phenotypic variation between the lines</p>
</li>
<li><dl class="first docutils">
<dt>scored for MANY types of info:</dt>
<dd><ul class="first last simple">
<li>enzyme activity</li>
<li>fat storage</li>
<li>more</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>modeling objective:</dt>
<dd><ul class="first last simple">
<li>fit models predicting endpoints like total fat storage from genome</li>
<li>assess sensitivity of component of models</li>
<li>explain genetic architecture</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="first docutils">
<dt>modeling:</dt>
<dd><ul class="first last">
<li><dl class="first docutils">
<dt>multiple regression</dt>
<dd><ul class="first last simple">
<li>observed vs expected: r^2 = 0.23</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Flux balance Analysis (FBA)</p>
</li>
<li><dl class="first docutils">
<dt>Structural Equation MOodels (SEM)</dt>
<dd><ul class="first last simple">
<li>obsvd vs exp: r^2 = 0.44</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Bayesian hierarchical model</p>
</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">3 cam 120 frames/sec linked in 3D</p>
</li>
<li><p class="first">perturbations by flying through magnetic field with magnet glued to their back</p>
</li>
<li><p class="first">recovery from perturbation begins in a single wing beat</p>
</li>
<li><p class="first"><strong>VERY COOL VIDEOS</strong></p>
</li>
<li><p class="first">Data suggesting that the Drosophila melanogaster population variance may mirror the human data supporting the idea that we brought Dmel with us as we left Africa and they follow similar variance constrictions as they followed the particular human population as they split</p>
</li>
<li><p class="first">New descriptions of hybrid dysgenesis through cross population mating allowing exploration of other mobile elements and their effects (similar to P-element)</p>
</li>
<li><p class="first">Wolbachia data suggests a single invasion of Dmel</p>
</li>
</ul>
<hr class="docutils"/>
</div>
<div class="section" id="peter-cherbas-indiana-univ">
<span id="peter-cherbas"/><h3>Peter Cherbas (Indiana Univ)</h3>
<p><em>modENCODE and the potential of systems biology in _Drosophila_</em></p>
<ul>
<li><p class="first">1938 new transcribed regions (not linked to any other gene model)</p>
</li>
<li><p class="first">almost every gene included new 5’ or 3’ UTR...</p>
</li>
<li><p class="first">promoter switching linked to certain lines where Broad in activated or deactivated by ecdysone</p>
</li>
<li><p class="first">environmental perturbations reveal totally new genes</p>
</li>
<li><p class="first">neural specific expressed isoforms have LONG 3’ extensions</p>
</li>
<li><p class="first">18 loci where there is almost exactly overlapped (sense/antisense) transcription (seems protein coding too but I am not sure)</p>
</li>
<li><p class="first">in general: the <strong>least certain</strong> part of any annotation is the non-coding regions</p>
</li>
<li><p class="first"><strong>RESOURCE:</strong> MANY cell lines used to produced transcriptomes for public use</p>
</li>
<li><dl class="first docutils">
<dt>cell lines unique and represents different “test tubes” for biological questions</dt>
<dd><ul class="first last simple">
<li>unique complement of transcription factors etc</li>
<li>sufficiently differentiated to be homologous in behavior and components</li>
</ul>
</dd>
</dl>
</li>
</ul>
<hr class="docutils"/>
</div>
<div class="section" id="jeremy-lynch-univ-illinois-chicago">
<span id="jeremy-lynch"/><h3>Jeremy Lynch (Univ Illinois Chicago)</h3>
<p><em>Global analysis of the dorsal-ventral patterning regulatory network in the wasp _Nasonia vitripennis_ using quantitative transcriptomics</em></p>
<ul>
<li><p class="first">opening statement: <strong>DON’T BE AFRAID OF THE COMMAND LINE!!!!</strong></p>
</li>
<li><dl class="first docutils">
<dt>Nasonia embryo layed out very similar to that of Dmel</dt>
<dd><ul class="first last simple">
<li>morphology and gene expression</li>
<li>most likely convergent evolution</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">Nasonia patterning systems more dynamic</p>
</li>
<li><p class="first">BMP provides most patterning information</p>
</li>
<li><p class="first">also once cells start migrating in the embryo there are major divergences from Dmel as well</p>
</li>
<li><dl class="first docutils">
<dt>does Nasonia use the same genes as Dmel?</dt>
<dd><ul class="first last simple">
<li>knock out BMP: lose dorsal gene expression</li>
<li>KO toll: lose ventral gene expression</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">used tuxedo protocol</p>
</li>
<li><dl class="first docutils">
<dt>also used trinity to see what happened de novo followed by DESeq</dt>
<dd><ul class="first last simple">
<li>works mostly ok but has a lot of nonsense as well</li>
</ul>
</dd>
</dl>
</li>
</ul>
<hr class="docutils"/>
</div>
<div class="section" id="sara-mitchell-harvard-public-health">
<span id="sarah-mitchell"/><h3>Sara Mitchell (Harvard Public Health)</h3>
<p><em>Genetic pathways induced by mating have a key role in the reproductive biology of _Anopheles gambiae_</em></p>
<ul class="simple">
<li>many discoveries regarding elements that are required for mating plug formation as well as signals effecting the female post mating changes (sperm is not required)</li>
<li>found changes in the midgut after mating</li>
<li>looking at early middle and late matting reponses (3, 12, 24 h)</li>
<li>mostly induction of genes in mated vs virgins</li>
<li>wow.  LOTS of information very fast!  Not keeping up with typing very well!</li>
</ul>
</div>
</div>
<hr class="docutils"/>
<div class="section" id="ecological-genomics">
<h2>Ecological Genomics</h2>
<div class="section" id="leslie-vosshall-the-rockefeller-univ">
<span id="leslie-vosshall"/><h3>Leslie Vosshall (The Rockefeller Univ)</h3>
<p><em>Dissecting mosquito host-seeking behavior through loss-of-function genetics</em></p>
<ul>
<li><dl class="first docutils">
<dt>two clases of insect oder recepters</dt>
<dd><ul class="first last simple">
<li>OR</li>
<li>IR</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">each insect has different numbers of these receptors</p>
</li>
<li><p class="first">“Aedes has a ‘horrifically’ enormous genome”</p>
</li>
<li><p class="first">KO OR-coeffector effectively removes function of all ORs</p>
</li>
<li><p class="first">used Zinc Fingers to do the KO</p>
</li>
<li><dl class="first docutils">
<dt>Behavior assay:</dt>
<dd><ul class="first last simple">
<li>tube with a human scent at the end, see if the females choose the right tube.</li>
<li>ORCO-/- cant see human scent alone but can see if CO2 is added</li>
<li>IRs must also be involved</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">given choice between human or guinee pig the double mutants are much less good at choosing the human</p>
</li>
<li><p class="first">ORco not needed to choose human over nothing but are important for choosing human vs non-human</p>
</li>
<li><dl class="first docutils">
<dt>Vienette 2:</dt>
<dd><ul class="first last simple">
<li>many cues moisture, heat, smell, C02 etc</li>
<li>moving on to gustatory receptors (GR)</li>
<li>GR3 knocks out all???</li>
<li>GR3-/- blind to CO2</li>
<li>CO2 + Heat causes biting</li>
<li>both needed</li>
<li>GR3- means no CO2 signal</li>
<li>mosqs need more than one signal to trigger feeding</li>
<li>mutants severly impaired in lab but in semi field conditions much LESS impaired</li>
</ul>
</dd>
</dl>
</li>
</ul>
<hr class="docutils"/>
</div>
<div class="section" id="brian-lazzaro-cornell-univ">
<span id="brian-lazzaro"/><h3>Brian Lazzaro (Cornell Univ)</h3>
<p><em>Complexity in the function and evolution of insect immunity</em></p>
<ul>
<li><dl class="first docutils">
<dt>how does diet affect immunity?</dt>
<dd><ul class="first last">
<li><p class="first">it does btw...</p>
</li>
<li><dl class="first docutils">
<dt>?? Could it not be that the more sugar you add to the food the more you make it easier for the bacteria? (you feed them more?)</dt>
<dd><ul class="first last simple">
<li>there seems to be some link to genotype</li>
<li>could be that the genetic component is how well the fly converts sugar to less usable substrates</li>
<li>?? could you use a sugar that the flies can use but is not easy for the bacteria to use?</li>
<li>?? a bactaeria that doesnt eat glucose?</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">4th PC out of 5 deminsions in PCA?</p>
</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">how does mating effect immunity?</p>
</li>
</ul>
</div>
</div>
<hr class="docutils"/>
<div class="section" id="my-own-personal-remember-this-list">
<h2>My Own Personal “Remember This” List</h2>
<ul>
<li><dl class="first docutils">
<dt>GMAP to map de novo assembled transcriptomes to the genome</dt>
<dd><ul class="first last simple">
<li>plan to convert to GTF and merge with cufflinks GTF using cuffmerge to help define UTRs better for CRM/CRE discovery in my Mosquito species</li>
</ul>
</dd>
</dl>
</li>
</ul>
</div>
</div>]]></description>
            <category><![CDATA[ Conferences ]]></category>
             <pubDate>Fri, 14 Jun 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/05/18/fundamentals_of_gene_expression__the__central_dogma__of_molecular_biology.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/05/18/fundamentals_of_gene_expression__the__central_dogma__of_molecular_biology.html</guid>
            <title><![CDATA[Fundamentals of Gene Expression: the 'Central Dogma' of Molecular Biology]]></title>
            <description><![CDATA[<div class="section" id="fundamentals-of-gene-expression-the-central-dogma-of-molecular-biology">
<h1>Fundamentals of Gene Expression: the ‘Central Dogma’ of Molecular Biology</h1>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>This is an adaptation/update of a post I wrote a long time ago on my other blog <a class="reference external" href="http://scipher.wordpress.com">scipher.wordpress.com</a>.  Here is <a class="reference external" href="http://scipher.wordpress.com/2007/07/19/the-central-dogma-of-molecular-biology">the original post</a> in case you would like to read it.</p>
<p class="last">The purpose of this post is to serve as a background foundation for me to point to when I reference certain aspects of gene expression in posts on this blog, especially as related to descriptions in my ‘dissertation’ posts.</p>
</div>
<div class="section" id="the-dogma-of-gene-expression">
<h2>The “Dogma” of Gene Expression?</h2>
<p>There is a concept in molecular biology that is referred to as “the central dogma”. I do not like this name as I feel that the language is too close to that of religion, but history has firmly cemented its use to describe the concept. In general, it is described as:</p>
<blockquote>
<div>Genetic information flows from nucleic acid (DNA and RNA) to nucleic acid, or from nucleic acid to protein, but does not flow from protein to nucleic acid.</div></blockquote>
<p>I believe that Francis Crick was one of the first to assert this.
<a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/05/18/fundamentals_of_gene_expression__the__central_dogma__of_molecular_biology.html#figure1"><em>Figure 1</em></a> reproduces a diagram of the central dogma from circa 1958, as reconstructed by Francis Crick.
There had been a strong debate at the time over what molecule was transferring the genetic information from old cell to new cell.
The two types of molecules involved were proteins and nucleic acids.
The legendary description of the double helix by Crick and Watson went a long way towards supporting DNA as the general data keeper, and the data has born this out.</p>
<p>In this post I will go over some of the implications of the central dogma, and provide an introduction into how DNA, RNA, and proteins interact to enable the life processes of our cells.</p>
<div class="figure" id="figure1">
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/Cricks_1958_central_dogma.jpg"><img alt="../../../_images/Cricks_1958_central_dogma.jpg" src="http://movingthelamppost.com/blog/html/_images/Cricks_1958_central_dogma.jpg" style="width: 300px;"/></a>
<p class="caption"><strong>Figure 1</strong></p>
<div class="legend">
<a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/05/18/fundamentals_of_gene_expression__the__central_dogma__of_molecular_biology.html#crick1970" id="id1">[Crick1970]</a></div>
</div>
<div id="more"> </div></div>
<div class="section" id="dna-as-database">
<h2>DNA as Database</h2>
<p>There are a couple reasons why DNA seems to be the main storage molecule.
DNA is more stable than RNA due to its molecular structure.
In addition, molecules of DNA are usually found in the famous double stranded helix orientation which yields a much more stable complex than single stranded nucleic acids.
The double helix also provides a very useful way to duplicate genetic information as a cell divides.</p>
<p>Nucleic acids store information.
Ok. What kind of information?
I said that information flows to proteins.
What does that mean?
Well DNA stores the instructions for how to make proteins.
Both DNA and proteins are polymers.
Like a metal chain, which is made up of repeating links of metal, DNA and proteins are made of smaller repeating building blocks or residues.
Unlike a chain, which usually has only one type of link, DNA and proteins have multiple types of building blocks.
Nucleic acids generally use four different residues called nucleotides.
Proteins are made up of about 20 different residues called amino acids.
So the information I was talking about is encoded as a particular pattern of the building blocks arranged in a sequence.
Just as a sentence in English can encode a virtually infinite number of meanings using sequences of only 26 letters, a protein uses a set of 20 amino acids to encode a virtually infinite number of patterns.</p>
<p>The specific pattern of amino acids determines the properties of the protein and therefore its function in the cell.
There are many reasons that the sequence determines function, but for now, suffice it to say that the sequence of amino acids that make up a particular protein is vital to its function.
It is this vital information that is encoded in the DNA.</p>
<p><strong>Proteins do the work of life.</strong>
They digest our food; they repair tissue damage; they let our cells communicate; they attack intruding organisms.
DNA is how our cells know how to make these molecules.</p>
</div>
<div class="section" id="a-typical-example">
<h2>A Typical Example</h2>
<p>What follows is a simplified version of how the information for how to make a protein gets from the DNA to being manifested as a functional protein.
I will describe how it happens in a <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-eukaryote"><em class="xref std std-term">eukaryotic</em></a> cell because that is the type of cells that humans are made of.
Eukaryotes (organisms with eukaryotic cells) separate their DNA from the rest of the cell into what is called a nucleus (<a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/05/18/fundamentals_of_gene_expression__the__central_dogma__of_molecular_biology.html#figure2"><em>Figure 2</em></a>).
This protects the DNA from the cellular processes that happen in the cell and allows for an extra level of control when it comes to gene regulation.</p>
<div class="figure" id="figure2">
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/Animal_cell_structure_en.svg.png"><img alt="../../../_images/Animal_cell_structure_en.svg.png" src="http://movingthelamppost.com/blog/html/_images/Animal_cell_structure_en.svg.png" style="width: 550px;"/></a>
<p class="caption"><strong>Figure 2</strong></p>
<div class="legend">
<p>The parts of an eukaryotic cell. (Click to enlarge)</p>
<p>[image credit: <a class="reference external" href="http://en.wikipedia.org/wiki/File:Animal_cell_structure_en.svg">Mariana Ruiz</a>]</p>
</div>
</div>
<div class="section" id="the-library-analogy">
<h3>The Library Analogy</h3>
<p>You can think of the nucleus as a library.
The DNA inside is like a giant set of reference books.
In a library, you usually cannot check out the reference books right?
You have to copy the information that you need into notes, and you will use these to write your paper.
It is the same with the DNA in the nucleus.
So what do cells copy notes with?
RNA.</p>
<p>When RNA is used in this fashion, we call it messenger RNA (mRNA) because it carries the message of how to make a protein.
There are certain types of proteins in the the nucleus that can unzip the double stranded helix of DNA.
Once this happens a copy of that region of the DNA is made, but with RNA instead of DNA.
This process of copying the DNA sequence into RNA sequence is called <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-transcription"><em class="xref std std-term">transcription</em></a>.</p>
<div class="topic">
<p class="topic-title first">DNA Transcription (Basic) (<a class="reference external" href="https://www.youtube.com/watch?v=SMtWvDbfHLo">Advanced</a>)</p>
<div style="padding-bottom: 56.250000%; padding-top: 30px; position: relative; width: 100%">
<iframe src="http://www.youtube.com/embed/5MfSYnItYvg" style="border: 0; height: 100%; left: 0; position: absolute; top: 0; width: 100%">
</iframe></div><p>I recommend watching both the basic <strong>and</strong> advanced versions of these videos. They each contain certain aspects that are left out of the other.</p>
<p>[video credit: <a class="reference external" href="http://www.dnalc.org">Cold Spring Harbor Laboratory - DNA Learning Center</a>]</p>
</div>
<p>Only one strand is copied.
This single stranded RNA copy of the gene then leaves the nucleus.
Outside the nucleus, another set of proteins reads the sequence of mRNA and gathers free floating amino acids to fuse them into a chain.
The sequence of the mRNA determines the order in which each amino acid is incorporated into the growing protein.
The process of translating the mRNA sequence into a protein sequence is called <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-translation"><em class="xref std std-term">translation</em></a>.</p>
<div class="topic">
<p class="topic-title first">mRNA Translation (Basic) (<a class="reference external" href="https://www.youtube.com/watch?v=TfYf_rPWUdY">Advanced</a>)</p>
<div style="padding-bottom: 56.250000%; padding-top: 30px; position: relative; width: 100%">
<iframe src="http://www.youtube.com/embed/8dsTvBaUMvw" style="border: 0; height: 100%; left: 0; position: absolute; top: 0; width: 100%">
</iframe></div><p>[video credit: <a class="reference external" href="http://www.dnalc.org">Cold Spring Harbor Laboratory - DNA Learning Center</a>]</p>
</div>
<p>That’s pretty much it! Not that complicated eh?</p>
<p>Not all organism’s use DNA as their storage molecule, however.
Some viruses encode their genome with RNA, others use DNA but only a single strand of it whereas the norm is double stranded DNA.
I have described what is typical, but you should know that there are forms of life that use variations of this.</p>
</div>
</div>
<div class="section" id="conclusion">
<h2>Conclusion</h2>
<p>The central dogma of molecular biology is the observation that genetic information generally can be transmitted from DNA/RNA to other DNA/RNA or from DNA/RNA to protein, but not from protein to DNA/RNA
I should say here that we cannot say that it is impossible for information to flow from protein to nucleic acid; it may be.
However, no one has postulated a mechanism whereby this might happen, and has never been observed.
It may be discovered someday, but that is VERY unlikely.
DNA is usually the storage molecule for genetic information which is then encoded into RNA so that it can ultimately be translated into a protein.</p>
<hr class="docutils"/>
<p class="rubric"><strong>Footnotes:</strong></p>
<hr class="docutils"/>
<p class="rubric"><strong>Citations:</strong></p>
<table class="docutils citation" frame="void" id="crick1970" rules="none">
<colgroup><col class="label"/><col/></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id1">[Crick1970]</a></td><td>Crick, F. (1970). Central dogma of molecular biology. Nature, 227(5258), 561-3. Retrieved from <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/4913914">http://www.ncbi.nlm.nih.gov/pubmed/4913914</a></td></tr>
</tbody>
</table>
</div>
</div>]]></description>
            <category><![CDATA[ Background Science ]]></category>
             <pubDate>Sat, 18 May 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/05/15/new_release__blacktie_v0_2_1.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/05/15/new_release__blacktie_v0_2_1.html</guid>
            <title><![CDATA[NEW RELEASE: Blacktie v0.2.1]]></title>
            <description><![CDATA[<div class="section" id="new-release-blacktie-v0-2-1">
<h1>NEW RELEASE: Blacktie v0.2.1</h1>
<p>I have released a new version of blacktie (0.2.1) which has a number of minor fixes and <strong>NOW INCLUDES</strong> <a class="reference external" href="http://compbio.mit.edu/cummeRbund">cummeRbund</a> <strong>SUPPORT!!!</strong></p>
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/basic_cummerbund_plots.png"><img alt="../../../_images/basic_cummerbund_plots.png" src="http://movingthelamppost.com/blog/html/_images/basic_cummerbund_plots.png" style="width: 600px;"/></a>
<ul class="simple">
<li>github repo: <a class="reference external" href="https://github.com/xguse/blacktie">https://github.com/xguse/blacktie</a></li>
<li>documentation: <a class="reference external" href="http://xguse.github.io/blacktie">http://xguse.github.io/blacktie/</a></li>
<li><strong>issue tracking and feature requests:</strong> <a class="reference external" href="https://github.com/xguse/blacktie/issues">https://github.com/xguse/blacktie/issues</a></li>
</ul>
<p>Thanks and again <strong>please</strong> let me know what you think.  I need your input to make blacktie better and easier to use.</p>
<p>Gus</p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p>One major change was introduced in v0.2.0rc1: the yaml config file was changed slightly to facilitate the inclusion of biological replicate data.  So please take a look at the demo config file in the docs posted at: <a class="reference external" href="http://xguse.github.io/blacktie/tutorial.html#the-configuration-file">http://xguse.github.io/blacktie/tutorial.html#the-configuration-file</a></p>
<p class="last">This is the last time that this warning will be included.</p>
</div>
<div class="section" id="news">
<h2>News</h2>
<div class="section" id="id1">
<h3>0.2.1</h3>
<p><em>Release date: 2013-05-15</em></p>
<ul class="simple">
<li>git tag: ‘v0.2.1’</li>
<li>added new script named blacktie-cummerbund to run cummeRbund</li>
<li>added new class in calls.py CummerbundCall to use blacktie-cummerbund script to add cummeRbund plots to blacktie script</li>
<li>checks for R and rpy2 installations</li>
<li>if cummeRbund R library not found, it walks you through installing it</li>
<li><span class="docutils literal"><span class="pre">src/blacktie/utils/calls.py</span></span>: - fixed _flag_out_dir() so that if the outdir has not been created yet it gracefully moves on</li>
<li><span class="docutils literal"><span class="pre">examples/blacktie_config_example.yaml</span></span>: - added cummerbund_options</li>
<li><span class="docutils literal"><span class="pre">requirements.txt</span></span>: - added rpy2</li>
<li>updated docs</li>
</ul>
</div>
<div class="section" id="rc1">
<h3>0.2.0rc1</h3>
<p><em>Release date: 2013-04-19</em></p>
<ul>
<li><p class="first">git tag: ‘v0.2.0rc1’</p>
</li>
<li><p class="first">Added support for handling biological replicates in cuffdiff runs.</p>
</li>
<li><dl class="first docutils">
<dt>Major changes to yaml config:</dt>
<dd><ul class="first last simple">
<li>condition_queue[index].group_id –&gt; condition_queue[index].experiment_id</li>
<li>addition of condition_queue[index].replicate_id to track replicate data</li>
<li>condition_queue[index].name now represents description of an ‘experiemental condition’ and will be shared by replicates.</li>
</ul>
</dd>
</dl>
</li>
</ul>
</div>
</div>
</div>]]></description>
            <category><![CDATA[ My Code ]]></category>
             <pubDate>Wed, 15 May 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/05/13/r__it_thinks_it_s_cooler_than_me__it_s_not_wrong.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/05/13/r__it_thinks_it_s_cooler_than_me__it_s_not_wrong.html</guid>
            <title><![CDATA[R: it thinks it's cooler than me; it's not wrong]]></title>
            <description><![CDATA[<div class="section" id="r-it-thinks-it-s-cooler-than-me-it-s-not-wrong">
<h1>R: it thinks it’s cooler than me; it’s not wrong</h1>
<p>I do a lot of RNA-seq analysis for my research and we mostly run the popular “Tuxedo” suite of analysis programs (<a class="reference external" href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">bowtie</a>, <a class="reference external" href="http://tophat.cbcb.umd.edu">tophat</a>, <a class="reference external" href="http://cufflinks.cbcb.umd.edu">cufflinks</a>, and now the <strong>REALLY COOL</strong> <a class="reference external" href="http://compbio.mit.edu/cummeRbund/manual_2_0.html">cummeRbund</a>).
So cummeRbund, as the capitalization suggests, is written in the <a class="reference external" href="http://www.r-project.org">R</a> programming language (meant mostly for doing statistics <strong>really</strong> well).  So I have been working in R a bit more than usual lately and I realized something. But before I get into that, you should know something:</p>
<p><strong>R and I have a *love:hate* relationship...</strong></p>
<p>It really does have almost mind-blowing capabilities, and it has been thoroughly embraced by at least the gene expression and related wings of the bioinformatics community.
This has resulted in the conglomerate family of R libraries housed under the umbrella of the <a class="reference external" href="http://www.bioconductor.org">bioconductor</a> project.
It’s data display capabilities are spectacular; to which anyone who has seen plots generated by the <a class="reference external" href="http://ggplot2.org">ggplot2</a> library can attest. <a class="reference external" href="http://goo.gl/1GBWV">Take a look.</a>.
Oh and did I mention does all of it’s world-class awesomeness completely <em>pro bono</em>?
Suffice it to say that, R is plain awesome and this once niche language initially popular with academics who couldn’t or simply wouldn’t spring for expensive licenses for more mainstream stats software is actually starting to take the rest of the data-analysis world by storm [<a class="reference external" href="http://java.sys-con.com/node/2288420">link1</a>, <a class="reference external" href="http://r4stats.com/articles/popularity">link2</a>].</p>
<p><strong>So that was some of the love side of the relationship.</strong></p>
<p>What I said I came to realize, at this beginning of the post, deals more with the <em>other</em> side of the relationship.
Basically, R is like that athletic, pretty, popular kid at your high school.
I know that sounds like an extraordinary claim, but bear with me.
R, like that kid, knows that what it has is pretty darn awesome.
And like that kid, it has grown up with people bending over backwards to ask it to the dance no matter how well or poorly it treated its suitors.
But I gotta tell ya, as a Python-guy, <em>it’s almost offensive to me to work in R</em>.
I do it.
And don’t get me wrong, I am glad it is there.
But it does not let you forget that you need <strong>it</strong>, not the other way around.</p>
<p>Now I must admit that if I started with R, instead of Perl and Python (hell, or even BASH), I might feel the opposite, but I am not sure about that.
It also might be that I am just spoiled by working in Python most of my time since that language bends over backwards <strong>for me</strong>.
But Holy Hell, it seems like R is the classic example of design by committee!</p>
<p>Here are some examples of things that drive me crazy.  Please pipe in if you object to or can provide rebuttals that might change my mind about them.</p>
<div class="topic">
<p class="topic-title first">Things that drive me crazy about R</p>
<ol class="arabic simple">
<li>Object methods seem to live in the global namespace.</li>
<li>Documentation is simply hit or miss at best and downright less-than-useless in many cases (and its not just the authors’ fault)</li>
<li>Slicing arrays is overly complicated</li>
<li>The convention of naming variables with a ‘.’ instead of an ‘_’</li>
<li>Assignment operator is twice as many characters as it should be</li>
</ol>
</div>
<p>Each point will probably end up getting its own short post.</p>
<div class="section" id="object-methods-seem-to-live-in-the-global-namespace">
<h2>Object methods seem to live in the global namespace.</h2>
<p>This one might be me not being familiar enough with how R treats scope or namespace, but it sure seems that if I want to call an object method (<span class="docutils literal"><span class="pre">objMethod</span></span>) for an R object (<span class="docutils literal"><span class="pre">rObj</span></span>), I call <span class="docutils literal"><span class="pre">objMethod</span></span> like a global function:</p>
<div class="highlight-none"><div class="highlight"><pre>objMethod(rObj)
</pre></div>
</div>
<p>That sure <strong>looks</strong> like it lives in the global namespace rather than in its object’s namespace.
Here is how this would be done in Python for comparison:</p>
<div class="highlight-none"><div class="highlight"><pre>pyObj.objMethod()
</pre></div>
</div>
<p>You specify the local namespace (<span class="docutils literal"><span class="pre">pyObj</span></span>) then call the method that lives in that namespace.  Its an object so it knows that it is operating on itself.  That’s kinda the <strong>whole idea of objects</strong>.</p>
<p>Now this might sound like a trivial difference, and to some extent, it is.
But it is also kinda dumb in my opinion.
For one thing, it needlessly creates confusion when trying to learn a code library.
There is nothing in the code that tells me that I can not call <span class="docutils literal"><span class="pre">objectMethod</span></span> on non-<span class="docutils literal"><span class="pre">rObj</span></span> variables: that it <strong>belongs</strong> to only objects of the same type as <span class="docutils literal"><span class="pre">rObj</span></span>.</p>
<p>In python this is unmistakable.</p>
<p>This is actually a major gripe of mine about R; it really doesn’t seem to want you to have an easy time learning new libraries.
However, there is another problem.
Because the <span class="docutils literal"><span class="pre">objectMethod</span></span> lives in global namespace, it can be overwritten if you load another library that has an object with a method of the same name.
<strong>This is exactly why object methods live in the local namespace of their objects in most languages</strong>.</p>
<p>Anyway, if I am wrong about this, please set me straight.  If nothing else, I will learn something new about R that will let me get better at it.</p>
</div>
</div>]]></description>
             <pubDate>Mon, 13 May 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html</guid>
            <title><![CDATA[Elements of transgenic disease-vector control (Part 2: altering the mosquitoes)]]></title>
            <description><![CDATA[<div class="section" id="elements-of-transgenic-disease-vector-control-part-2-altering-the-mosquitoes">
<h1>Elements of transgenic disease-vector control (Part 2: altering the mosquitoes)</h1>
<p id="altering-the-mosquitoes"><a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/21/elements_of_transgenic_disease_vector_control_part1.html#setting-the-scene"><em>Part one of this series can be found here</em></a></p>
<div class="admonition attention">
<p class="first admonition-title">Attention</p>
<p class="last">I am serious when I say that I want your criticisms, suggestions, comments, reference suggestions, etc. Please <strong>do</strong> visit the comments section and help me out! The most helpful commenters will be acknowledged in the final dissertation document.</p>
</div>
<div class="topic">
<p class="topic-title first">The four fundamentals of transgenic vector control:</p>
<ol class="arabic simple">
<li>the ability to introduce <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-transgene"><em class="xref std std-term">transgenes</em></a> into the mosquito’s genome in a heritable way</li>
<li>discovery or construction <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-effector-gene"><em class="xref std std-term">effector genes</em></a></li>
<li>the ability to control when and where the effector genes are expressed</li>
<li>the ability of your transgenic mosquitoes to have <strong>swift</strong>, dramatic, and clinically meaningful effects on the native vector population (<a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-gene-drive"><em class="xref std std-term">gene drive</em></a>)</li>
</ol>
</div>
<div class="section" id="introducing-inheritable-transgenes-into-the-mosquito-genome">
<h2>Introducing Inheritable Transgenes into the Mosquito Genome</h2>
<p>Before one can employ transgenic methods for vector control, one must of course have the ability to make transgenic vectors.
For mosquitoes, this is quite a bit more labor intensive than for other fly species like <em>Drosophila melanogaster</em>.
However, it has been accomplished, and protocols to reliably introduce transgenes into many mosquito species now exist.
Unfortunately, the relative difficulty is still high, and it takes roughly 2-3 months before a stable line is established.
So while this step can be filed under “accomplished”, work is constantly being done to improve its ease and efficiency.</p>
<div id="more"> </div><p>The method of gene insertion into the genome generally relies on re-purposing the activities of small, “parasitic”, mobile DNA elements called <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-transposon"><em class="xref std std-term">transposons</em></a> or <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-transposable-element"><em class="xref std std-term">transposable elements</em></a>.
These elements have the ability to cut themselves out of one part of the DNA and re-insert themselves into another.
We can engineer them in the lab to carry our genes instead of the genes that they usually contain.
This allows our genes to be inserted into the genome instead of the contents of the original transposon.</p>
<p>Until recently, most if not all transposons used inserted the genes into more-or-less random regions, and in single, double or <strong>multiple</strong> copies and/or locations <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html#adelman2004" id="id1">[Adelman2004]</a>,<a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html#sethuraman2007" id="id2">[Sethuraman2007]</a>.
This makes characterizing the effects of the effector genes complicated.
However, with the successful implementation of the <a class="reference external" href="http://en.wikipedia.org/wiki/Site-specific_recombinase_technology#PhiC31_Integrase">phic31 site specific integration system</a> in <em>Aedes aegypti</em> and <em>Anopheles stephensi</em> <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html#thorpe1998" id="id3">[Thorpe1998]</a>,<a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html#nimmo2006" id="id4">[Nimmo2006]</a>,<a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html#isaacs2012" id="id5">[Isaacs2012]</a>, we can now reliably target specific, known regions of the genome for transgene integration of a single copy.</p>
<p>The “heritable” part is achieved through which cells take up the transgene.
Only certain types of cells have the ability to project their genetic code into the next generation.
In order to have a transgene be heritable by the following generations we must make sure that at least some of the cells that get the transgene are germ line cells (sperm or egg cells).</p>
<p>A peculiarity about fly development is that as the embryo is developing the cells that will become the germ line always position themselves at one end (or pole) of the embryo.
They earned the moniker “pole cells”.
This makes targeting these cells for the transgene feasible.
When the solution containing the transgene construct is injected into the embryos, it is done where the pole cells are.
This means that the first set of mosquitoes that might be fully transgenic is not the set that you injected but the first generation of offspring that your injected mosquitoes produce.</p>
</div>
<div class="section" id="effector-genes">
<h2>Effector Genes</h2>
<p>With the ability to integrate transgene constructs into the vectors’ genomes, we need constructs that carry genes that will effect the response that we desire in the vector.
Two broad strategies are commonly applied when designing an anti-transmission <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-phenotype"><em class="xref std std-term">phenotype</em></a>.</p>
<div class="section" id="vector-population-reduction">
<h3>Vector Population Reduction:</h3>
<div class="align-center figure align-right">
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/onlyGoodMosq.png"><img alt="../../../_images/onlyGoodMosq.png" src="http://movingthelamppost.com/blog/html/_images/onlyGoodMosq.png" style="width: 250px;"/></a>
<p class="caption"><em>The only</em> <strong>“good”</strong> <em>mosquito is a...</em></p>
</div>
<p>The goal of vector population reduction is the same as the conventional vector control modalities.
<em>Reduce the size of the vector population to decrease the probability of infectious interactions between vectors and humans.</em></p>
<p><strong>Conventional tactics</strong> include removal of breeding sites through draining of swamps, removal of items that store water (tires, buckets, etc) where larvae develop from populated areas, and of course, chemical insecticidal campaigns. Another, more recent tactic is sterile insect technique (see the relevant <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/21/elements_of_transgenic_disease_vector_control_part1.html#topic-sit"><em>topic</em></a> from part one of this series).</p>
<p>A <strong>transgenic tactic</strong> for the population reduction strategy is illustrated by <a class="reference external" href="http://www.oxitec.com">Oxitec’s</a> <em>Aedes aegypti</em> strain <a class="reference external" href="http://www.oxitec.com/health/our-products/aedes-agypti-ox3604c">OX3604C</a> developed with support from the <a class="reference external" href="http://www.grandchallenges.org/Pages/Default.aspx">Bill and Melinda Gates Foundation’s Grand Challenges for Global Health Initiative</a>.
OX3604C, represents a female-specific <abbr title="Release of Insects carrying a Dominant Lethal phenotype">RIDL</abbr> approach which uses a poison transgene that kills any female adult expressing the transgene unless the female is being fed the “antidote” through its water supply.</p>
<p>Releasing enough of these mosquitoes will affect the local mosquito species population in a way that is analogous to <abbr title="Sterile Insect Technique">SIT</abbr>.</p>
</div>
<div class="section" id="vector-population-replacement-conversion">
<h3>Vector Population Replacement (Conversion):</h3>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>The more standard term is vector population replacement.
However, I prefer <strong>vector population conversion</strong> because it is more accurate.
We will not necessarily even <em>want</em> to replace the local population with purely lab-produced mosquitoes.
What will most likely be happening is that the population goes from having no frequency of our designed trait, to having a high frequency of our trait, but that the rest of the genetic information in the population is still derived from the genomes present in the <em>original population</em>.</p>
<p class="last">It would be as if a brown eyed family moved into a valley where people only had blue eyes.
Since the <em>brown eyes</em> trait is usually dominant over the <em>blue eyes</em> trait, the brown eyed phenotype will spread through the valley as generations interbreed.
However, it would not be accurate to say that the original brown eyed family <strong>replaced</strong> the native population.</p>
</div>
<p>Vector population conversion is a novel strategy for vector control that only exists within the context of genetically altering the vectors.
This strategy actually promises to be the most long-lasting vector-focused intervention; because unlike <strong>all</strong> vector population reduction tactics, there is a theoretical point in a conversion intervention (the anti-transmission trait works and is present in local populations at near 100%) when the human interaction can be ceased but the intervention continues to function.
For population reduction to approach this result, the vector species must not simply be eliminated from the local area, but approach elimination on a continental scale, or more realistically achieve global eradication.</p>
<p>The reason is that these mosquitoes (especially <em>Aedes aegypti</em>) can and do travel long distances, as eggs or larvae, in the backs of trucks (between villages) or in the pools of water collected in super-tankers (transcontinentally).
So a local village is only free of the vectors until more migrate into the area.
But in a conversion scenario, those migrants mate with the local vector population, and their offspring are assimilated into transmission-deficient mosquitoes.
The protection of the local village can be preserved <em>even if some surrounding villages fail to maintain control of their mosquitoes</em>.</p>
<p>An example of a <strong>transgenic tactic</strong> for population conversion of <em>Aedes aegypti</em> into a transmission-deficient phenotype involves an effector gene that codes for double stranded portion of the target dengue virus as RNA <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html#franz2006" id="id6">[Franz2006]</a>,<a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html#mathur2010" id="id7">[Mathur2010]</a>.
Because most animal cells have a system that detects and degrades double stranded RNA<a class="footnote-reference" href="#dsrna" id="id8">[1]</a> in a sequence specific manner, this primes the mosquito cells’ antiviral response to specifically attack the dengue virus if the effector gene is expressed in the cell before it gets infected.</p>
</div>
</div>
<div class="section" id="controlling-when-and-where-the-effector-genes-are-expressed">
<h2>Controlling When and Where the Effector Genes are Expressed</h2>
<div class="topic">
<p class="topic-title first">Crash Course in Gene Expression:</p>
<p>The following is a <strong>very</strong> brief introduction to how the expression of genes are usually controlled.
Pay specific attention to the parts that mention <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-transcription-factor"><em class="xref std std-term">transcription factors</em></a> and enhancer regions.</p>
<div style="padding-bottom: 56.250000%; padding-top: 30px; position: relative; width: 100%">
<iframe src="http://www.youtube.com/embed/SMtWvDbfHLo" style="border: 0; height: 100%; left: 0; position: absolute; top: 0; width: 100%">
</iframe></div><p>If that was a little over your head, <a class="reference external" href="https://www.youtube.com/watch?v=5MfSYnItYvg">watch this version first</a>, then go back and watch the advanced on again.</p>
</div>
<p>Even armed with an effector gene that clears 100% of the pathogen 100% of the time, you will not be successful in limiting transmission if it is not turned on in the right time and place.
If your effector gene works best when the pathogen is in the midgut, but your gene is only expressed in the antennae, you have wasted your time.</p>
<p>The region of DNA directly before the sequence of the gene is <em>usually</em> the most influential determinant of the pattern of expression and is referred to as the <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-promoter"><em class="xref std std-term">promoter</em></a>.
It determines when the gene will be turned on (let’s say directly following the ingestion of a bloodmeal), and in which tissue type (let’s say the midgut).
The way that this is accomplished is due to the binding of special proteins called transcription factors that recognize specific DNA sequences.
Once bound to the promoter they recruit the special machinery needed for the gene to be turned on.</p>
<p>If we wanted to control a transgene in a specific way (turned on after a bloodmeal in the midgut), one way to go about it would be to identify genes in the mosquito that already have a similar expression pattern to the ideal that you want.
Then we could copy the promoter from that gene and paste it in front of our transgene<a class="footnote-reference" href="#copypaste" id="id9">[2]</a>.
Because we will have replicated the specific transcription factor binding sites (TFBS) that control the original gene, our transgene should inherit a very similar expression pattern.</p>
<p>This is a very common process used to engineer the expression patterns of real transgenes in mosquitoes.
In <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html#moreira2000" id="id10">[Moreira2000]</a>, the promoter sequences of a gene called carboxypeptidase (normally expressed in the midgut after a meal to help digest it) from <em>Aedes aegypti</em> and <em>Anopheles gambiae</em> were pasted in front of a transgene that causes the cells that express the gene to light up.
This type of transgene is called a <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-reporter-gene"><em class="xref std std-term">reporter gene</em></a> because it allows the researchers to visualize the activity of the promoter used to drive its expression.</p>
<p>From the abstract of <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html#moreira2000" id="id11">[Moreira2000]</a>:</p>
<blockquote>
<div>Six independent transgenic lines were obtained with the <abbr title="Ae = Aedes aegypti; CP = carboxypeptidase gene">AeCP</abbr> construct and one with the <abbr title="Ag = Anopheles gambiae; CP = carboxypeptidase gene">AgCP</abbr> construct. Luciferase mRNA and protein were abundantly expressed in the guts of transgenic mosquitoes in four of the six AeCP lines and in the AgCP line. Expression of the reporter gene was gut-specific and reached peak levels at about 24 h post-blood ingestion.</div></blockquote>
</div>
<div class="section" id="achieving-swift-dramatic-and-clinically-meaningful-effects-on-the-native-vector-population">
<h2>Achieving Swift, Dramatic, and Clinically Meaningful Effects on the Native Vector Population</h2>
<p>In many ways, this is the most difficult part of the puzzle.
In order for the transgene to have its self-sustaining properties as well as achieve effective anti-transmission results, it must spread through the native mosquito population to the point that the percent of individuals possessing the gene approaches 100%.</p>
<p>Optimistically assuming that the transgene carries a negligible fitness cost<a class="footnote-reference" href="#fitness" id="id12">[3]</a>, or even a slight fitness advantage, achieving near 100% conversion could take decades.
Funding terms for efforts like this in poor nations can be closer to 5 years or less.
To enable the population conversion strategies to work, we must come up with genetic “tricks” that cause the gene to spread through a mosquito population <strong>much</strong> faster than could happen naturally.
Efforts to discover or design this <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-gene-drive"><em class="xref std std-term">gene drive</em></a> system are an on going and active area in this field.</p>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last">The topic of gene drive deserves its own post, so I will use this space to link to any future post that tackles the subject in substantial depth.</p>
</div>
<hr class="docutils"/>
<p class="rubric"><strong>Footnotes:</strong></p>
<table class="docutils footnote" frame="void" id="dsrna" rules="none">
<colgroup><col class="label"/><col/></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id8">[1]</a></td><td>Double stranded RNA generally signals that a virus is active in the cell.</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="copypaste" rules="none">
<colgroup><col class="label"/><col/></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id9">[2]</a></td><td>The story is <strong>of course</strong> more complicated than this, and the replication of the original expression pattern may not always be perfect with this simplified method; however, it is suitably explanatory for our purposes at the moment.</td></tr>
</tbody>
</table>
<table class="docutils footnote" frame="void" id="fitness" rules="none">
<colgroup><col class="label"/><col/></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id12">[3]</a></td><td>By fitness cost/advantage here, I mean that the transgene causes the mosquitoes that inherit it to be either less or more successful at producing offspring, respectively.</td></tr>
</tbody>
</table>
<hr class="docutils"/>
<p class="rubric"><strong>Citations:</strong></p>
<table class="docutils citation" frame="void" id="adelman2004" rules="none">
<colgroup><col class="label"/><col/></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id1">[Adelman2004]</a></td><td>Adelman, Z. N., Jasinskiene, N., Vally, K. J. M., Peek, C., Travanty, E. A., Olson, K. E., Brown, S. E., et al. (2004). Formation and loss of large, unstable tandem arrays of the piggyBac transposable element in the yellow fever mosquito, Aedes aegypti. Transgenic research, 13(5), 411–25. Retrieved from <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/15587266">http://www.ncbi.nlm.nih.gov/pubmed/15587266</a></td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="sethuraman2007" rules="none">
<colgroup><col class="label"/><col/></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id2">[Sethuraman2007]</a></td><td>Sethuraman, N., Fraser, M. J., Eggleston, P., &amp; O’Brochta, D. A. (2007). Post-integration stability of piggyBac in Aedes aegypti. Insect biochemistry and molecular biology, 37(9), 941–51. doi:10.1016/j.ibmb.2007.05.004</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="thorpe1998" rules="none">
<colgroup><col class="label"/><col/></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id3">[Thorpe1998]</a></td><td>Thorpe, H. M., &amp; Smith, M. C. (1998). In vitro site-specific integration of bacteriophage DNA catalyzed by a recombinase of the resolvase/invertase family. Proceedings of the National Academy of Sciences of the United States of America, 95(10), 5505–10. Retrieved from <a class="reference external" href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=20407&amp;tool=pmcentrez&amp;rendertype=abstract">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=20407&amp;tool=pmcentrez&amp;rendertype=abstract</a></td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="nimmo2006" rules="none">
<colgroup><col class="label"/><col/></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id4">[Nimmo2006]</a></td><td>Nimmo, D. D., Alphey, L., Meredith, J. M., &amp; Eggleston, P. (2006). High efficiency site-specific genetic engineering of the mosquito genome. Insect molecular biology, 15(2), 129–36. doi:10.1111/j.1365-2583.2006.00615.x</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="isaacs2012" rules="none">
<colgroup><col class="label"/><col/></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id5">[Isaacs2012]</a></td><td>Isaacs, A. T., Jasinskiene, N., Tretiakov, M., Thiery, I., Zettor, A., Bourgouin, C., &amp; James, A. A. (2012). Transgenic Anopheles stephensi coexpressing single-chain antibodies resist Plasmodium falciparum development. Proceedings of the National Academy of Sciences of the United States of America, 109(28), E1922–30. doi:10.1073/pnas.1207738109</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="franz2006" rules="none">
<colgroup><col class="label"/><col/></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id6">[Franz2006]</a></td><td>Franz, A. W. E., Sanchez-Vargas, I., Adelman, Z. N., Blair, C. D., Beaty, B. J., James, A. A., &amp; Olson, K. E. (2006). Engineering RNA interference-based resistance to dengue virus type 2 in genetically modified Aedes aegypti. Proceedings of the National Academy of Sciences of the United States of America, 103(11), 4198–203. doi:10.1073/pnas.0600479103</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="mathur2010" rules="none">
<colgroup><col class="label"/><col/></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id7">[Mathur2010]</a></td><td>Mathur, G., Sanchez-Vargas, I., Alvarez, D., Olson, K. E., Marinotti, O., &amp; James, a a. (2010). Transgene-mediated suppression of dengue viruses in the salivary glands of the yellow fever mosquito, Aedes aegypti. Insect molecular biology, 1. doi:10.1111/j.1365-2583.2010.01032.x</td></tr>
</tbody>
</table>
<table class="docutils citation" frame="void" id="moreira2000" rules="none">
<colgroup><col class="label"/><col/></colgroup>
<tbody valign="top">
<tr><td class="label">[Moreira2000]</td><td><em>(<a class="fn-backref" href="#id10">1</a>, <a class="fn-backref" href="#id11">2</a>)</em> Moreira, L. a, Edwards, M. J., Adhami, F., Jasinskiene, N., James, a a, &amp; Jacobs-Lorena, M. (2000). Robust gut-specific gene expression in transgenic Aedes aegypti mosquitoes. Proceedings of the National Academy of Sciences of the United States of America, 97(20), 10895–8. Retrieved from <a class="reference external" href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=27120&amp;tool=pmcentrez&amp;rendertype=abstract">http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=27120&amp;tool=pmcentrez&amp;rendertype=abstract</a></td></tr>
</tbody>
</table>
</div>
</div>]]></description>
            <category><![CDATA[ My Research ]]></category>
            <category><![CDATA[ My Dissertation ]]></category>
             <pubDate>Sun, 12 May 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/05/09/comparison_of_aedes_aegypti_and_anopheles_gambiae_midgut_expression_post_bloodmeal.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/05/09/comparison_of_aedes_aegypti_and_anopheles_gambiae_midgut_expression_post_bloodmeal.html</guid>
            <title><![CDATA[Comparison of Aedes aegypti and Anopheles gambiae midgut gene expression post bloodmeal: an overview]]></title>
            <description><![CDATA[<div class="section" id="comparison-of-aedes-aegypti-and-anopheles-gambiae-midgut-gene-expression-post-bloodmeal-an-overview">
<h1>Comparison of <em>Aedes</em> <em>aegypti</em> and <em>Anopheles</em> <em>gambiae</em> midgut gene expression post bloodmeal: an overview</h1>
<a class="reference external image-reference" href="/blog/html/_static/talks/2013-05-08_myWork/2013-05-08_myWork/index.html"><img alt="../../../_images/Screenshot.png" class="align-center" src="http://movingthelamppost.com/blog/html/_images/Screenshot.png" style="width: 600px;"/></a>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The links on the discussion slides are now fixed!</p>
</div>
<div class="section" id="introduction-to-new-post-category-my-presentations">
<h2>Introduction to New Post Category: My Presentations</h2>
<p>This is my first post in a new category of posts called “My Presentations” which shockingly will consist of as many of my talks that I am able to share online.
It will most likely only contain talks that I write going forward since most of my other talks are not in ‘website’ form, and I do not have time to convert them.</p>
<p>However, I recently discovered an approach to writing presentations that is pretty similar to how I write this blog (in <a class="reference external" href="http://docutils.sourceforge.net/rst.html">ReST</a> or <a class="reference external" href="http://daringfireball.net/projects/markdown">Markdown</a> then conversion to HTML and JavaScript with a special tool: <a class="reference external" href="http://www.tinkerer.me">Tinkerer</a> for this blog and <a class="reference external" href="https://github.com/egonSchiele/mdpress">mdpress</a> for the presentations).
Therefore the resulting presentations are webpages and easily shareable!
Yay <a class="reference external" href="http://opensciencefederation.com">Open Science</a>!</p>
<div id="more"> </div><p>So far I REALLY like the presentations it produces, but it <strong>IS</strong> a little flashy, and I have found some differences in how it is displayed depending on the browser being used (so sorry if it sucks for you).
The underlying technology behind the presentations is a combination of JavaScript and CSS3.
Specifically, an engine written by <a class="reference external" href="https://github.com/bartaz">Bartek Szopka</a> (hereafter referred to as his <a class="reference external" href="http://github.com">GitHub</a>/coder handle: bartaz) called <a class="reference external" href="https://github.com/bartaz/impress.js">impress.js</a>.
Here is his <a class="reference external" href="http://bartaz.github.io/impress.js">introduction to it</a>, written <strong>in</strong> it.</p>
<p>However, beware!
In its native form, you write your whole presentation in HTML and CSS.
It is not for folks that are looking for a flashier version of powerpoint.
That’s where mdpress comes in (THANK GOD).
<a class="reference external" href="https://github.com/egonSchiele">Aditya Bhargava</a> wrote mdpress and it is awesome.</p>
<p>I write my presentation <strong>mostly</strong> in markdown (certain more complicated things require jumping into HTML and/or editing the CSS style-sheets), and mdpress reads my markdown file and writes the HTML code for me.
Also, as you saw if you played with his introduction, <em>which you should</em>, you might easily get carried away and end up distracting your audience at best, or nauseating them if you are not careful.</p>
<blockquote>
<div><p>With great power comes great responsibility</p>
<p>– Uncle Ben to Peter Parker (<em>Spiderman</em>)</p>
</div></blockquote>
<p>Anyway, enough introduction.</p>
</div>
<div class="section" id="presentation-background">
<h2>Presentation Background</h2>
<p>This is a talk I gave to my lab yesterday at our weekly lab meeting.
It represents my effort to update the group on my projects and provides an opportunity for lab-mates to ask questions, give comments and advice, etc.
This data is part of my dissertation project (which really should get its own full introduction post soon...).
Here is a bit of context.</p>
<p>These are preliminary <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-rna-seq"><em class="xref std std-term">RNA-seq</em></a> results from an effort I am undergoing to characterize the degree to which conservation of the regulation of bloodmeal triggered gene expression may be described across the ~150 million year divergence time between the last common ancestor of my subject species (<em>Anopheles gambiae,</em> <em>Anopheles stephensi</em>, <em>Aedes aegypti</em>, and <em>Culex quinquefasciatus</em>).</p>
<div class="figure align-center">
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/mosqPhyloTree.png"><img alt="../../../_images/mosqPhyloTree.png" src="http://movingthelamppost.com/blog/html/_images/mosqPhyloTree.png" style="width: 600px;"/></a>
</div>
<p>The talk also served as a way for me to introduce them to the analyses and plots generated by <a class="reference external" href="http://compbio.mit.edu/cummeRbund">cummeRbund</a> that I have integrated into my development version of <a class="reference external" href="https://github.com/xguse/blacktie">Blacktie</a> my automated <a class="reference external" href="http://tophat.cbcb.umd.edu">Tophat</a>/<a class="reference external" href="http://cufflinks.cbcb.umd.edu">Cufflinks</a>/<a class="reference external" href="http://compbio.mit.edu/cummeRbund">cummeRbund</a> RNA-seq analysis pipeline.</p>
</div>
</div>]]></description>
            <category><![CDATA[ My Research ]]></category>
            <category><![CDATA[ My Presentations ]]></category>
             <pubDate>Thu, 09 May 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/04/21/elements_of_transgenic_disease_vector_control_part1.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/04/21/elements_of_transgenic_disease_vector_control_part1.html</guid>
            <title><![CDATA[Elements of transgenic disease-vector control (Part 1: setting the scene)]]></title>
            <description><![CDATA[<div class="section" id="elements-of-transgenic-disease-vector-control-part-1-setting-the-scene">
<h1>Elements of transgenic disease-vector control (Part 1: setting the scene)</h1>
<span class="target" id="setting-the-scene"/><p><a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html#altering-the-mosquitoes"><em>Part two of this series can be found here</em></a></p>
<div class="admonition attention">
<p class="first admonition-title">Attention</p>
<p class="last">I am serious when I say that I want your criticisms, suggestions, comments, reference suggestions, etc. Please <strong>do</strong> visit the comments section and help me out! The most helpful commenters will be acknowledged in the final dissertation document.</p>
</div>
<div class="section" id="foreword">
<h2>Foreword</h2>
<p>This is the first installment in my endeavor to <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/15/blogging_my_dissertation.html"><em>“Blog My Dissertation”</em></a>.
I am afraid that I have not quite settled on a format for this project or even a solid governing theme on how to organize and execute the bloggification...
So I ask you to please bear with me as I experiment with how technical to be; how polished; how formal/informal; and “dissertation-ready” to make the prose etc.</p>
<p>This post will cover part of the dissertation’s first chapter (Introduction and Background).
The long term goal of the Anthony James lab at UCI is to produce genetically modified mosquitoes that do not transmit <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-malaria"><em class="xref std std-term">malaria</em></a> or <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-dengue-fever"><em class="xref std std-term">dengue fever</em></a> and will spread this trait swiftly through local mosquito populations.
In order to produce mosquitoes that will perform these functions, we use a process called <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-transgenesis"><em class="xref std std-term">transgenesis</em></a> that introduces a set of <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-exogenous"><em class="xref std std-term">exogenous</em></a> genes into the mosquitoes’ <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-genome"><em class="xref std std-term">genome</em></a> and tells them when and where to be expressed.
All of my work should be interpreted in the scheme of aiding the lab’s ability to improve this process; however, there is real basic science that it aims to further as well.</p>
<p>This is part one of a few posts that will discuss the fundamental elements of transgenic disease-<a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-vector"><em class="xref std std-term">vector</em></a> control.
It will set the scene by explaining what a disease-vector is and how our specific models of disease transmission influence how we can approach the problem of malaria and dengue fever by addressing the mosquitoes rather than treating the humans.</p>
<div id="more"> </div></div>
<div class="section" id="mosquito-borne-disease">
<h2>Mosquito borne disease</h2>
<div class="topic">
<p class="topic-title first"><abbr title="Too Long; Didn't Read">TL;DR</abbr>:</p>
<ul class="simple">
<li>there are many but we focus on Dengue Fever and Malaria</li>
<li>causative agents are RNA viruses and Plasmodium protists, respectively</li>
<li>people don’t give people these two diseases</li>
<li>mosquitoes generally don’t give mosquitoes these diseases</li>
</ul>
</div>
<p>There are many mosquito borne diseases that afflict the world’s population; however, the James Lab focuses on two of the major ones: dengue fever and malaria.
The causative agents of these two diseases are a suite of four species of RNA viruses (<a class="reference external" href="http://en.wikipedia.org/wiki/Dengue_fever#Virology">Dengue Virus 1-4</a>) and a suite of five species of protist from the genus <a class="reference external" href="http://en.wikipedia.org/wiki/Plasmodium"><em>Plasmodium</em></a>, respectively.
However, in any <strong>single</strong> location they are generally transmitted by a single species of mosquito:</p>
<table border="1" class="docutils">
<colgroup>
<col width="34%"/>
<col width="23%"/>
<col width="43%"/>
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Disease</th>
<th class="head">Region</th>
<th class="head">Mosquito Vector</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>Dengue Viruses</td>
<td>Worldwide</td>
<td><em>Aedes aeygypti</em></td>
</tr>
<tr class="row-odd"><td rowspan="2">Human Malaria</td>
<td>Africa</td>
<td><em>Anopheles gambiae</em></td>
</tr>
<tr class="row-even"><td>Asia</td>
<td><em>Anopheles stephensi</em></td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br/></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The multiple species nature of these two diseases makes addressing the problem of preventing human disease caused by these pathogens quite difficult: especially when focusing on the human aspect.</p>
</div>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last">Other posts will go into more detail on the specifics of the distribution and effects of the actual diseases on local populations. Once completed, they will be linked to from here.</p>
</div>
<p>Another very important aspect of both of these diseases is that people do not become infected through interaction with other people, and mosquitoes do not become infected through interaction with other mosquitoes.
The <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-transmission-cycle"><em class="xref std std-term">transmission cycle</em></a> of both diseases requires that the pathogen be passed from human to mosquito or from mosquito to human.
We will soon see that this pattern has <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/21/elements_of_transgenic_disease_vector_control_part1.html#pub-health"><em>important consequences that we can exploit</em></a> to prevent the transmission cycle from churning out infected individuals.</p>
<div class="section" id="transmission-cycle">
<h3>Transmission cycle</h3>
<div class="topic">
<p class="topic-title first"><abbr title="Too Long; Didn't Read">TL;DR</abbr>:</p>
<ol class="arabic simple">
<li>a female mosquito bites infected person</li>
<li>the pathogen is taken into mosquito midgut with bloodmeal</li>
<li>the pathogen must escape the midgut and gain access to the mosquito’s circulatory system <em>(the animal is now</em> <strong>infected</strong><em>)</em></li>
<li>the pathogen must gain access to the mosquito’s salivary glands <em>(the animal is now</em> <strong>infectious</strong><em>)</em></li>
<li>the infectious female mosquito bites an uninfected human and pathogens in her saliva are introduced to the person’s body</li>
</ol>
</div>
<p>While the specifics of how viruses and protists live and reproduce in mosquitoes and humans are quite different, the transmission cycles of dengue and malaria are very similar in their major events.
The transmission of either pathogen from one infected human to another can both be summarized into five fundamental events.
First, a female mosquito feeds on an infected human’s blood taking up the pathogen as well.
The bloodmeal is digested in the mosquito’s <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-midgut"><em class="xref std std-term">midgut</em></a>, and as you might expect, the midgut is the first stop for the pathogen too.
Escape of the midgut is the first critical step for the survival of the pathogen inside the mosquito.
If it is trapped in the midgut, it will eventually be passed as waste after the bloodmeal is digested.
If it manages to escape to the <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-hemolymph"><em class="xref std std-term">hemolymph</em></a>, the pathogen must gain access to the mosquito’s salivary glands if it is to infect another human.
If this step is unsuccessful, then the mosquito, while itself <strong>infected</strong>, is not <strong>infectious</strong> to other humans.
For this, the pathogen must be injected into the bloodstream of the next human along with the contents of the mosquito’s saliva.</p>
<div class="figure align-center">
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/mosqXsection.png"><img alt="../../../_images/mosqXsection.png" src="http://movingthelamppost.com/blog/html/_images/mosqXsection.png" style="width: 585px;"/></a>
<p class="caption"><strong>Cross section of female mosquito with tissues important to the transmission cycle highlighted.</strong></p>
</div>
</div>
</div>
<div class="section" id="public-health-multiple-targets">
<span id="pub-health"/><h2>Public health: <em>multiple targets</em></h2>
<div class="topic">
<p class="topic-title first"><abbr title="Too Long; Didn't Read">TL;DR</abbr>:</p>
<ul class="simple">
<li>The specifics of the transmission cycle of these diseases provides multiple targets for public health interventions</li>
<li>focus on the human</li>
<li>focus on the vector</li>
</ul>
</div>
<p>Because the arrow of transmission always<a class="footnote-reference" href="#nonhuman-transmission" id="id1">[1]</a> points from mosquito to human or from human to mosquito, if either of those arrows are broken, the cycle will collapse and the area would eventually be cleared of the pathogen provided the intervention is maintained.
This model suggests that there are two primary targets for public health interventions aiming to reduce the <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-incidence"><em class="xref std std-term">incidence</em></a> (followed by <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-prevalence"><em class="xref std std-term">prevalence</em></a>) of infection in a population.</p>
<ol class="arabic simple">
<li>prevent mosquitoes from infecting humans</li>
<li>prevent humans from infecting mosquitoes</li>
</ol>
<p>However, many of the interventions can not be clearly divided into addressing purely the human or vector side of the cycle.
For example, any efforts to reduce the number of bites that humans receive from mosquitoes affects both the probabilities of the vector <em>and</em> the human becoming infected.
This is one reason that vector control rather than purely human-based interventions are almost always part of prevention strategies.
For the purposes of this document, I will define human-based interventions to be those that are directly administered to the human’s body.
Essentially: medical interventions.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">It should be noted that in this respect, the long term goals of public health are more focused on prevention than the treatment of acute cases. Of course sick people need to be treated, and finding and clearing people who are infected is a part of preventing mosquitoes from acquiring the pathogen from people.  However, in the long game it is much more effective to prevent the infection .  These efforts are what we will focus on.</p>
</div>
<div class="section" id="options-for-the-humans">
<h3>Options for the humans</h3>
<p>Because of the definition of human-based interventions that I am using, there are relatively few effective options in this category for malaria and dengue fever.
Normally, this section would include vaccines and swift, effective patient identification and treatment to clear the infection.
For reasons that will be discussed in a later post, these interventions simply do not exist on the market or in a cost-effective form applicable to the isolated and impoverished areas that are most affected.</p>
<div class="admonition seealso">
<p class="first admonition-title">See also</p>
<p class="last">This subject should also be given a deeper treatment in the future. Links to those posts will show up here.</p>
</div>
</div>
<div class="section" id="conventional-options-for-the-vectors">
<h3>Conventional options for the vectors</h3>
<p>Because medical options are generally quite scarce, most attention in the field is directed toward controlling access of the vector populations to human contact.
This can include removal of nearby mosquito breeding sites (usually standing water), spraying of insecticides, introduction of biological predators, and/or bed nets, etc.
One fairly novel approach that has implications for the next post in this series is sterile insect technique.</p>
<div class="topic" id="topic-sit">
<p class="topic-title first">Sterile Insect Technique:</p>
<p><a class="reference external" href="http://en.wikipedia.org/wiki/Sterile_insect_technique">Sterile insect technique (SIT)</a> exploits a peculiar aspect of some insects’ reproductive behavior.
In many insects, it is only (or at least primarily) the first mating event that “matters”.
Subsequent mating events contribute little to no genetic material to the females progeny, <strong>even when the first event involves a sterile male</strong>.
This means that if massive numbers of sterilized males are introduced into a native population, any wild female that mates first with one of the sterile males will be effectively sterilized herself.
This can have dramatic effects on the local population.
For a famously effective SIT campaign, look up <a class="reference external" href="http://goo.gl/DF7bv">screwworm eradication</a> on google.</p>
</div>
</div>
</div>
<div class="section" id="transgenic-options-for-the-vectors">
<h2>Transgenic options for the vectors</h2>
<p>As you saw above, most conventional vector control strategies involve what might be termed <strong>vector population reduction</strong>.  As we get into what transgenics can do for vector control, we will see that in addition to population reduction we have a new strategy available which has quite exciting implications to the sustainability of the vector control aspect of public health interventions for dengue and malaria.
This could be called <strong>vector population conversion</strong>.  Tune in next time to read all about it!</p>
<p><a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html#altering-the-mosquitoes"><em>Part two of this series can be found here</em></a></p>
<div class="line-block">
<div class="line"><br/></div>
<div class="line"><br/></div>
<div class="line"><br/></div>
</div>
<p class="rubric"><strong>Footnotes:</strong></p>
<table class="docutils footnote" frame="void" id="nonhuman-transmission" rules="none">
<colgroup><col class="label"/><col/></colgroup>
<tbody valign="top">
<tr><td class="label"><a class="fn-backref" href="#id1">[1]</a></td><td>While some mosquito to non-human vertebrate transmission may occur, it is not thought to be sufficient to maintain the pathogen if the mosquito:human loop is broken.</td></tr>
</tbody>
</table>
</div>
</div>]]></description>
            <category><![CDATA[ My Research ]]></category>
            <category><![CDATA[ My Dissertation ]]></category>
             <pubDate>Sun, 21 Apr 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/04/19/new_release__blacktie_v0_2_0rc1.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/04/19/new_release__blacktie_v0_2_0rc1.html</guid>
            <title><![CDATA[NEW RELEASE: Blacktie v0.2.0rc1]]></title>
            <description><![CDATA[<div class="section" id="new-release-blacktie-v0-2-0rc1">
<h1>NEW RELEASE: Blacktie v0.2.0rc1</h1>
<p>I have released a new version of blacktie (0.2.0rc1) which has a number of minor fixes and new features.</p>
<ul class="simple">
<li>github repo: <a class="reference external" href="https://github.com/xguse/blacktie">https://github.com/xguse/blacktie</a></li>
<li>documentation: <a class="reference external" href="http://xguse.github.io/blacktie">http://xguse.github.io/blacktie/</a></li>
<li>issue tracking and feature requests: <a class="reference external" href="https://github.com/xguse/blacktie/issues">https://github.com/xguse/blacktie/issues</a></li>
</ul>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">One major gotcha will be that the yaml config file has been changed slightly to facilitate the inclusion of biological replicate data.  So please take a look at the demo config file in the docs posted at: <a class="reference external" href="http://xguse.github.io/blacktie/tutorial.html#the-configuration-file">http://xguse.github.io/blacktie/tutorial.html#the-configuration-file</a></p>
</div>
<p>Thanks and <strong>please</strong> let me know what you think.  I need your input to make improvements.</p>
<p>Gus</p>
<hr class="docutils"/>
<div class="section" id="news">
<h2>News</h2>
<div class="section" id="rc1">
<h3>0.2.0rc1</h3>
<p><em>Release date: 2013-04-19</em></p>
<ul>
<li><p class="first">git tag: ‘v0.2.0rc1’</p>
</li>
<li><p class="first">email_notification is now adjustable for other email servers</p>
</li>
<li><p class="first">Added support for handling biological replicates in cuffdiff runs.</p>
</li>
<li><dl class="first docutils">
<dt>Major changes to yaml config:</dt>
<dd><ul class="first last simple">
<li>condition_queue[index].group_id –&gt; condition_queue[index].experiment_id</li>
<li>addition of condition_queue[index].replicate_id to track replicate data</li>
<li>condition_queue[index].name now represents description of an ‘experimental condition’ and will be shared by replicates.</li>
</ul>
</dd>
</dl>
</li>
</ul>
</div>
</div>
</div>]]></description>
            <category><![CDATA[ My Code ]]></category>
             <pubDate>Fri, 19 Apr 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html</guid>
            <title><![CDATA[Genome-Level Analysis of Selective Constraint without Apparent Sequence Conservation]]></title>
            <description><![CDATA[<div class="section" id="genome-level-analysis-of-selective-constraint-without-apparent-sequence-conservation">
<h1>Genome-Level Analysis of Selective Constraint without Apparent Sequence Conservation</h1>
<div class="section" id="full-citation">
<h2>Full Citation</h2>
<p>Vakhrusheva, O. A, Bazykin, G. A, &amp; Kondrashov, A. S. (2013). <em>Genome-level analysis of selective constraint without apparent sequence conservation</em>. <strong>Genome biology and evolution</strong>, 5(3), 532–541. doi:10.1093/gbe/evt023</p>
<p>Links: <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/23418180">PubMed</a> – <a class="reference external" href="http://gbe.oxfordjournals.org/content/5/3/532.full.pdf+html">Full Text (PDF)</a> – <a class="reference external" href="http://gbe.oxfordjournals.org/content/5/3/532.full">Full Text (HTML)</a></p>
<p><a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#license-information-pertaining-to-the-original-article">License information pertaining to the original article</a></p>
</div>
<div class="section" id="elevator-speech">
<h2>Elevator Speech</h2>
<div class="section" id="academic-significance">
<h3>Academic Significance</h3>
<p>Other studies have demonstrated individual instances of conservation of sequence elements that are no longer <strong>globally alignable</strong> due to extended phylogenetic divergence.  Familiar examples include the demonstration that some transgenic enhancer regions from orthologous genes in distantly related species are able to drive correct gene expression of the host species even though <strong>global alignment</strong> of the enhancer regions was impossible.</p>
<p>However the authors claim that this is the first genome-scale study they know of that supports the phenomena of selective constraint acting on sequence elements absent of identifiable sequence conservation.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">By terms like ‘identifiable sequence conservation’, they really seem to be referring to the ability to generate meaningful <strong>global alignment</strong> since they plainly use <strong>local alignment</strong> to identity the sequences they propose are being selectively retained.</p>
</div>
</div>
<div class="section" id="main-hypotheses">
<h3>Main Hypotheses</h3>
<p><span class="math">\(H_1\)</span>: the presence of a ‘conserved’ segment between two less divergent species should <em>overall</em> predict the presence of corresponding conserved segments in a separate pair of similarly divergent species that together have a much larger pair-to-pair divergence distance (see <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#f1"><em>Figure 1</em></a> for graphical representation). <strong>[relevant figures:</strong> <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#f2"><em>Figure 2</em></a> <strong>]</strong></p>
<p><span class="math">\(H_2\)</span>: if <em>empirically validated</em> regulatory elements persist longer than the sequence similarity of the corresponding DNA segments, introns containing a regulatory element in pair 1 should carry a corresponding segment in pair 2 at rates higher than expected by chance. <strong>[relevant figures:</strong> <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#f3"><em>Figure 3</em></a> <strong>]</strong></p>
<p><span class="math">\(H_3\)</span>: presence of a functional element in an orthologous intron in one species pair should correlate with a reduced instance of intron loss when compared to an outgroup serving as a way to infer the presence/absence of the functional element in a distant common ancestor. <strong>[relevant figures:</strong> <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#f4"><em>Figure 4</em></a> <strong>]</strong></p>
<p><span class="math">\(H_4\)</span>: because others have already established that longer introns are more likely to carry a segment of conservation than short introns, a function conserved in all four species should be expected to be associated with larger intron length  when conserved in pair 1 <em>and</em> pair 2, compared with the introns lost in pair 2. <strong>[relevant figures:</strong> <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#f5"><em>Figure 5</em></a> <strong>]</strong></p>
<div id="more"> </div><div class="line-block">
<div class="line"><br/></div>
</div>
<hr class="docutils"/>
<div class="line-block">
<div class="line"><br/></div>
</div>
</div>
</div>
<div class="section" id="highlights-by-section">
<h2>Highlights by Section</h2>
<div class="section" id="introduction">
<h3>Introduction</h3>
<ul>
<li><p class="first">Conservation of function is usually perceived to be associated with conservation of sequence (the main discovery process for orthologous genes is centered on sequence similarity even)</p>
</li>
<li><p class="first">As far as I can tell, when they say “sequence conservation” they mean successful sequence alignment across genomes</p>
</li>
<li><dl class="first docutils">
<dt>This coupling of function with primary sequence conservation is not strictly required. Examples given:</dt>
<dd><ul class="first last simple">
<li>unalignable proteins can have very similar 3D structures (and even active site locations)</li>
<li>single stranded RNA molecules can form identical structures regardless of alignability</li>
<li>unalignable enhancers from mice can correctly drive gene expression in zebra fish</li>
</ul>
</dd>
</dl>
</li>
</ul>
<div class="line-block">
<div class="line"><br/></div>
</div>
<hr class="docutils"/>
<div class="line-block">
<div class="line"><br/></div>
</div>
</div>
<div class="section" id="materials-and-methods">
<h3>Materials and Methods</h3>
<div class="section" id="data">
<h4>Data</h4>
<ul class="simple">
<li>two sets of species quartets were chosen (one dipteran and one vertebrate quartet)</li>
<li>each quartet represents a basic evolutionary structure characterized by two species couplets and an outgroup</li>
<li>each couplet shows less evolutionary divergence between the species than exists between the two couplet groups of each quartet (see <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#f1"><em>Figure 1</em></a>)</li>
<li>orthologs for all species were obtained from the <a class="reference external" href="http://inparanoid.sbc.su.se/cgi-bin/index.cgi">INPARANOID</a> web resource</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>The following was a bit tricky at first reading:</p>
<p class="last">It seems there were two classes of analyses performed. The first with just the four experimental species (what level of ‘conservation’ of small local similarity exists between and within the couplets of the two quartets), and one to assess cases of intron loss which required use of the outgroup.</p>
</div>
<ul>
<li><dl class="first docutils">
<dt>ortholog inclusion depended on several criteria including but not limited to:</dt>
<dd><ul class="first last simple">
<li>all four or five species contain a version of the ortholog</li>
<li>no ortholog set was allowed to have any member contain a premature stop codon</li>
</ul>
</dd>
</dl>
</li>
<li><p class="first">the resulting ortholog set sizes were:</p>
</li>
</ul>
<table border="1" class="docutils">
<colgroup>
<col width="24%"/>
<col width="26%"/>
<col width="50%"/>
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Quartet</th>
<th class="head">Experimental</th>
<th class="head">Plus Outgroup</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>dipterans</td>
<td>5,183</td>
<td>3,541</td>
</tr>
<tr class="row-odd"><td>vertebrates</td>
<td>8,159</td>
<td>2,518</td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br/></div>
<div class="line"><br/></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">evolutionary divergence is defined here as the amount of expected synonymous substitutions per site in each species (<span class="math">\(Ks\)</span>)</p>
</div>
<div class="line-block">
<div class="line"><br/></div>
<div class="line"><br/></div>
</div>
<div class="figure align-center" id="f1">
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/F1.large.jpg"><img alt="../../../_images/F1.large.jpg" src="http://movingthelamppost.com/blog/html/_images/F1.large.jpg" style="width: 600px;"/></a>
<p class="caption"><strong>Figure 1</strong></p>
<div class="legend">
<em>Two quartets of species used in the analysis, of 1) dipterans and 2) vertebrates, together with the corresponding outgroup species. Evolutionary distances within each pair of species, characterized by the estimated per site number of synonymous substitutions Ks, are presented.</em></div>
</div>
<div class="line-block">
<div class="line"><br/></div>
</div>
<hr class="docutils"/>
<div class="line-block">
<div class="line"><br/></div>
</div>
</div>
<div class="section" id="identification-and-analysis-of-orthologous-introns">
<h4>Identification and Analysis of Orthologous Introns</h4>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">orthologous introns here <strong>very basically</strong> means that the introns share the same flanking exons in all the species</p>
</div>
<ul>
<li><p class="first">a series of filters were applied to all the introns of the orthologous genes from each quartet</p>
</li>
<li><p class="first">the purpose of the filters were to focus on introns that were least likely to have conserved sequence (<strong>by alignability</strong>) while most likely to actually share common ancestry.</p>
</li>
<li><dl class="first docutils">
<dt>the final useful intron counts were:</dt>
<dd><ul class="first last simple">
<li>dipterans: 5,367</li>
<li>vertebrates: 51,844</li>
</ul>
</dd>
</dl>
</li>
</ul>
<div class="line-block">
<div class="line"><br/></div>
</div>
<hr class="docutils"/>
<div class="line-block">
<div class="line"><br/></div>
</div>
</div>
<div class="section" id="calculation-of-expected-number-of-introns-carrying-a-segment-of-similarity-within-both-pairs">
<h4>Calculation of Expected Number of Introns Carrying a Segment of Similarity within Both Pairs</h4>
<div class="topic">
<p class="topic-title first">General Procedure</p>
<ol class="arabic simple">
<li>Orthologous introns in each species pair are blasted against each other looking for regions of <strong>local alignment</strong></li>
<li>Definition of successful <strong>local alignment</strong> are set by using progressively more stringent E-value thresholds</li>
<li>For each E-vale threshold: the number of orthologous introns producing successful <strong>local alignments</strong> are tallied</li>
<li>To determine the number of <em>expected</em> successful <strong>local alignments</strong>, the data are randomly shuffled 10,000 times</li>
<li>Four different shuffling methods for generating the <em>expected</em> successful <strong>local alignments</strong> were employed</li>
</ol>
</div>
<table border="1" class="docutils">
<colgroup>
<col width="81%"/>
<col width="19%"/>
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Shuffling Methods</th>
<th class="head">Figure</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>no accounting for confounding factors</td>
<td>S1</td>
</tr>
<tr class="row-odd"><td>accounting for intron length</td>
<td>2</td>
</tr>
<tr class="row-even"><td>accounting for gene identity</td>
<td>S3</td>
</tr>
<tr class="row-odd"><td>accounting for gene identity <strong>and</strong> intron order</td>
<td>S4</td>
</tr>
</tbody>
</table>
<div class="line-block">
<div class="line"><br/></div>
</div>
<hr class="docutils"/>
<div class="line-block">
<div class="line"><br/></div>
</div>
</div>
<div class="section" id="analysis-of-data-on-chromatin-modifications">
<h4>Analysis of Data on Chromatin Modifications</h4>
<p>Because so far the idea of ‘functional’ has been simply inferred on the basis of coincidental inclusion of similar local sequence identity, the authors attempt to harvest experimentally validated functional regions from <em>Homo sapiens</em> and <em>Drosophila melanogaster</em> using the <a class="reference external" href="http://genome.ucsc.edu/ENCODE">ENCODE</a> and <a class="reference external" href="http://www.modencode.org">modENCODE</a> projects data.</p>
<p>For humans they focus on relevant strong enhancers (states 4 and 5) or insulators (state 8).  For flies they focus on enhancers (state 3) or ‘active introns’ (state 4).</p>
<div class="line-block">
<div class="line"><br/></div>
</div>
<hr class="docutils"/>
<div class="line-block">
<div class="line"><br/></div>
</div>
</div>
<div class="section" id="identification-of-intron-losses">
<h4>Identification of Intron Losses</h4>
<p>Direct quote:</p>
<blockquote>
<div>For the third and the fourth tests, to trace the losses of introns on the phylogeny, the tree corresponding to each species quartet was rooted with an outgroup species ([<a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#f1"><em>Figure 1</em></a>]). We then selected introns present in both species from pair 1 and in the outgroup species, assuming that such introns were present in the last common ancestor of both considered species pairs. Again, orthologous introns with significant similarity (bl2seq E-value ≤ 0.0001) between sequences from different pairs were excluded from analysis. These introns were then subdivided into 1) those also present in both pair 2 species and 2) those that had been lost in at least one of the pair 2 species. (Introns absent in one of the pair 2 species were usually also absent in the other, implying loss on the branch separating pair 2 from its common ancestor with pair 1; cases of intron loss mapping to external branches were rare.) For groups of introns indicated in 1) and 2), we compared the distributions of E-values within pair 1, and the distributions of intron lengths within pair 1. Reciprocal tests were performed analogously.</div></blockquote>
<div class="line-block">
<div class="line"><br/></div>
</div>
<hr class="docutils"/>
<div class="line-block">
<div class="line"><br/></div>
</div>
</div>
</div>
<div class="section" id="results-and-discussion">
<h3>Results and Discussion</h3>
<div class="figure align-center" id="f2">
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/F2.large.jpg"><img alt="../../../_images/F2.large.jpg" src="http://movingthelamppost.com/blog/html/_images/F2.large.jpg" style="width: 600px;"/></a>
<p class="caption"><strong>Figure 2</strong></p>
<div class="legend">
<em>Introns that carry a segment of high similarity between species of one pair are more likely to also carry a segment of high similarity between species of the other pair within a quartet. Each blue dot corresponds to a specific BLAST E-value (shown next to the dot), with lower values corresponding to more stringent similarity thresholds. Each E-value was used to detect similar segments within orthologous introns of the two species belonging to pair 1, and of the two species belonging to pair 2. Horizontal axis, fraction of introns that carry similar segments within pair 1, among introns present in all four species. Vertical axis, number (A, B) or observed-to-expected ratio (C, D) for the number of introns that carry similar segments both within pair 1 and 2. Top, dipterans; bottom, vertebrates. Observed-to-expected ratio was defined as the ratio of the observed number of introns with similarity within both pairs to the expected number if the segments of similarity were distributed randomly over all introns, controlling for intron lengths (see text). The red line and the gray area correspond to the mean and 95% confidence intervals for the expected values calculated in 10,000 resampling trials.</em></div>
</div>
<div class="line-block">
<div class="line"><br/></div>
</div>
<hr class="docutils"/>
<div class="line-block">
<div class="line"><br/></div>
</div>
<div class="figure align-center" id="f3">
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/F3.large.jpg"><img alt="../../../_images/F3.large.jpg" src="http://movingthelamppost.com/blog/html/_images/F3.large.jpg" style="width: 600px;"/></a>
<p class="caption"><strong>Figure 3</strong></p>
<div class="legend">
<em>Introns that carry a segment of similarity in pair 2 are more likely to overlap regulation-associated elements within pair 1. Each blue dot corresponds to a specific BLAST E-value (shown next to the dot). Each E-value was used to detect similar segments within orthologous introns of the two species belonging to pair 2. Horizontal axis, fraction of introns that carry similar segments within pair 2, among introns present in all four species. Vertical axis, observed-to-expected ratio for the number of introns that carry a regulation-associated element within pair 1, according to modENCODE (Kharchenko et al. 2010; Roy et al. 2010) (A, B) or ENCODE (ENCODE Project Consortium 2004; Ernst et al. 2011) (C–F) data, and also carry a segment of similarity within pair 2. (A, B) Dipterans; (C–F) vertebrates. (A) Active introns, any cell line; (B) enhancers, any cell line; (C) insulators, all adult cell lines; (D) insulators, both embryonic cell lines; (E) strong enhancers, all adult cell lines; (F) strong enhancers, both embryonic cell lines. Observed-to-expected ratio was defined as the ratio of the observed number of introns with a regulation-associated element within pair 1 and similarity within pair 2 to the same number expected if the regulation-associated elements and the segments of similarity were distributed randomly over all introns, controlling for intron lengths. The red line and the gray area correspond to the mean and 95% confidence intervals for the expected values calculated in 10,000 resampling trials.</em></div>
</div>
<div class="line-block">
<div class="line"><br/></div>
</div>
<hr class="docutils"/>
<div class="line-block">
<div class="line"><br/></div>
</div>
<div class="figure align-center" id="f4">
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/F4.large.jpg"><img alt="../../../_images/F4.large.jpg" src="http://movingthelamppost.com/blog/html/_images/F4.large.jpg" style="width: 600px;"/></a>
<p class="caption"><strong>Figure 4</strong></p>
<div class="legend">
<em>Introns preserved in both pair 2 species are more likely to carry a conserved segment within pair 1. For each quartet, the distribution of E-values within pair 1 are shown for introns preserved in both pair 2 species (green), and for introns lost in at least one of the pair 2 species (blue). E-values indicated below the horizontal axis correspond to the lower E-value threshold. Insets show the fraction of introns with at least a marginal (E-value ≤ 1) similarity observed, for the same two groups.</em></div>
</div>
<div class="line-block">
<div class="line"><br/></div>
</div>
<hr class="docutils"/>
<div class="line-block">
<div class="line"><br/></div>
</div>
<div class="figure align-center" id="f5">
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/F5.large.jpg"><img alt="../../../_images/F5.large.jpg" src="http://movingthelamppost.com/blog/html/_images/F5.large.jpg" style="width: 600px;"/></a>
<p class="caption"><strong>Figure 5</strong></p>
<div class="legend">
<em>Introns preserved in both pair 2 species tend to be longer within pair 1 species. For each quartet, the distribution of lengths within pair 1 are shown for introns preserved in both pair 2 species (green), and for introns lost in at least one of the pair 2 species (blue). Lengths indicated below the horizontal axis correspond to the lower length threshold. The length of the shorter of the two orthologous introns in pair 1 was used.</em></div>
</div>
</div>
</div>
<div class="section" id="license-information-pertaining-to-the-original-article">
<h2>License information pertaining to the original article</h2>
<p>Unless otherwise noted, all figures and text in the captions are the work of the original authors.  I claim no authorship to these whatsoever. The research was originally licensed and <a class="reference external" href="http://www.oxfordjournals.org/our_journals/gbe/terms.html">distributed under the terms of the Creative Commons Attribution Non-Commercial License (http://creativecommons.org/licenses/by-nc/3.0/), which permits unrestricted non-commercial use, distribution, and reproduction in any medium, provided the original work is properly cited.</a></p>
</div>
</div>]]></description>
            <category><![CDATA[ Journal Club ]]></category>
            <category><![CDATA[ Journal Club (shorts) ]]></category>
             <pubDate>Tue, 16 Apr 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/04/15/blogging_my_dissertation.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/04/15/blogging_my_dissertation.html</guid>
            <title><![CDATA[Blogging my dissertation]]></title>
            <description><![CDATA[<div class="section" id="blogging-my-dissertation">
<h1>Blogging my dissertation</h1>
<p>I am not surprised or disappointed that <a class="reference external" href="http://chronicle.com/article/How-Blogging-Helped-Me-Write/136893">this idea has already been done</a>.
In fact, I am quite encouraged that I read that others that have done this so far have had such great experiences with it.
I began to think about using my blog to help me get over my reluctance to simply sit down and write for quite a few years.
However, I somehow never made the connection that I could also use it to help me actually write the documents themselves.
It was only recently, after <a class="reference external" href="http://movingthelamppost.com/blog/html/2013/04/04/hello__world_.html">making a personal promise</a> to engage more frequently and vigorously in the <a class="reference external" href="http://en.wikipedia.org/wiki/Open_science">open science</a> endeavor, that I started to research how the idea of blogging a <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-dissertation"><em class="xref std std-term">dissertation</em></a> had worked out for other people.</p>
<div id="more"> </div><div class="section" id="it-s-been-normalizing">
<h2>It’s been normalizing</h2>
<p>Gladly, I discover passages about many all-to-familiar feelings I have been having about writing that mystic tome that will represent the sum of my time as a graduate student, yet will most likely be read by <strong>maybe</strong> ten people...
Like many other grad students, I find writing, along with the associated stresses and insecurities, quite intimidating.
I fall prey to avoiding such tried and true methods of addressing this as <em>write everyday (even if for just 20 min)</em> because I can’t think of anything to write or because I convince myself that it would be a waste of my time or I feel like if it’s not perfect the first time out of my head then I am a failure or... <em>ad infinitum</em>.</p>
</div>
<div class="section" id="blogging-as-document-idea-development-tool">
<h2>Blogging as document/idea development tool</h2>
<p>It’s actually kind of a perfect device for me; I think.
Blogs are naturally pretty bite-sized which removes the <em>“Holy FSM! I am</em> <strong>NEVER</strong> <em>going to be able to finish this!”</em> brake that is a personal favorite nightmare of mine.
Because of this, I <strong>should</strong> be much more likely to write regularly.
They are generally aimed at curious <em>non-experts</em> which silences the <em>“What if I show it to my adviser and he/she thinks</em> <strong>its so horrible</strong> <em>and tells me to</em> <strong>start over</strong>?!” demon.
Blogs have built-in feedback mechanisms in their comments sections which allow input from people you may never have expected.
This is great for two reasons:</p>
<ol class="arabic">
<li><p class="first">criticism from people you don’t know can be easier to recover from than from folks you feel like you <strong>need</strong> to impress</p>
</li>
<li><p class="first">it is exactly these people that a good scientist-communicator <strong>should</strong> be hearing from, and on a frequent basis</p>
<blockquote>
<div><ul class="simple">
<li>shared jargon, background knowledge, etc makes it too easy to write for yourself or other experts in your tiny corner of the community if you truly want to improve your ability to communicate science.</li>
<li>it can also be kind of dangerous: group-think is the enemy of innovation and you never know how well you communicated your ideas until you get questions/comments from non-experts.</li>
</ul>
</div></blockquote>
</li>
</ol>
<p>It’s not bad practice for grant writing either since you need to make sure that <strong>everyone</strong> on that study section has the best chance possible of understanding the purpose, approach, and <em>awesomeness</em> of your ideas regardless of the referee’s specialty.</p>
</div>
<div class="section" id="the-opposite-of-wasting-my-time">
<h2>The opposite of wasting my time</h2>
<p>I have been trying to figure out what to do with the many but <strong>short</strong> bits of time that I have to myself when I am being a full-time dad two and a half days a week while the little dude is or <strong>should</strong> be napping.
I find that I can not code in that time because I will lose my place and spend half the time reminding myself what I was doing when I left off the last time.
This way I am actually making progress towards my major goals and improving an essential skill for any scientist.
Granted: I will not be able to simply cut and paste my work from here into submission formats for journals or my committee as a final dissertation.
But my thoughts will have been laid out, organized, reworked, and hopefully tested through community involvement in the comments.</p>
</div>
<div class="section" id="what-s-it-about">
<h2>What’s it about?</h2>
<p>Thanks for asking!</p>
<p>Here is a snippet explaining it:</p>
<blockquote>
<div><p><strong>Comparative transcriptomics of bloodmeal regulated gene expression in mosquito midguts across 150 million years of divergence.</strong></p>
<p>Hematophagy (blood-feeding) is a trait believed to have been present in the common ancestor of modern mosquitoes.  The females of many mosquito species must obtain a suitable bloodmeal in order to become competent to mature the first clutch of fertilized eggs, a fact that drives the transmission of many tropical mosquito-borne diseases.  20-hydroxyecdysone (20E) is a well known steroid hormone that is involved in molting and other developmental switches across all of Arthropoda; it is also known to participate in the transcriptional control of bloodmeal regulated genes in mosquitoes.</p>
<p>This work aims to characterize the level of conservation of transcriptional patterns in genes of four mosquito species (<em>Anopheles gambiae, Anopheles stephensi, Aedes aegypti</em>, and <em>Culex quinquefasciatus</em>) as measured using RNA-seq.  It specifically aims to focus on genes that may be regulated by the early transcription factors in the 20E transcriptional program.  RNA was collected from midguts of mosquitoes 0, 4, 6, 8, and 10 hours post bloodmeal.  It represents an effort to combine multiple genome-scale data types including comparative transcriptomics, orthology, evolutionary tree branch lengths (divergence times as old as 150 million years), and putative transcription factor binding sites.</p>
</div></blockquote>
</div>
<div class="section" id="what-are-your-thoughts-experiences-with-blogs-as-academic-tools">
<h2>What are your thoughts/experiences with blogs as academic tools?</h2>
<p>I would love to hear about your experiences with blogs in academia.
Do you have any words of advice or warning?
Do you have any favorite academics that you follow?
Please comment below and give me your thoughts.</p>
</div>
<div class="section" id="follow-me">
<h2>Follow me!</h2>
<p>If you liked what you just read and want to read more as it comes out: <a class="reference external" href="http://feeds.feedburner.com/MovingTheLamppost">subscribe to the rss feed</a>!</p>
</div>
</div>]]></description>
            <category><![CDATA[ My Dissertation ]]></category>
             <pubDate>Mon, 15 Apr 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/04/08/exploring_ptci.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/04/08/exploring_ptci.html</guid>
            <title><![CDATA[Exploring Various ways to combine transcription profile correlation (r), confidence in that correlation (p value associated with r), and phylogenetic distance (w(d)) between species into a useful PTC index (Phylogenetic Transcription Conservation Index)]]></title>
            <description><![CDATA[<div class="section" id="exploring-various-ways-to-combine-transcription-profile-correlation-r-confidence-in-that-correlation-p-value-associated-with-r-and-phylogenetic-distance-w-d-between-species-into-a-useful-ptc-index-phylogenetic-transcription-conservation-index">
<h1>Exploring Various ways to combine transcription profile correlation (<em>r</em>), confidence in that correlation (<em>p value</em> associated with <em>r</em>), and phylogenetic distance (<em>w(d)</em>) between species into a useful PTC index (Phylogenetic Transcription Conservation Index)</h1>
<div class="section" id="original-solution">
<h2>Original solution</h2>
<p><span class="math">\(PTCI = r \cdot (1-p) \cdot w(d)\)</span></p>
<p>Xiaohui objected that <em>1-p</em> is not optimal because the <em>1-p</em> will likely
have a distribution skewed toward <em>1</em>. He suggested loging the variable,
among other options (see below).</p>
<div class="section" id="original-results">
<h3>Original Results</h3>
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/Ag_Aa_Cq_3way_PTCI.png"><img alt="../../../_images/Ag_Aa_Cq_3way_PTCI.png" src="http://movingthelamppost.com/blog/html/_images/Ag_Aa_Cq_3way_PTCI.png" style="width: 600px;"/></a>
<p>This is a histogram of the three way mean PTCIs using the original
solution. Blue=Real data. Grey=Randomizations that abolish the three way
orthology relationships. The ‘Null’ distribution [grey] is significantly
different from the real data [blue] but has significant structure that
is not what one would predict (resemble a normal dist). Notably, there is
a large concentration around -0.5 that I can not explain.</p>
<div id="more"> </div></div>
<div class="section" id="suggested-alterations">
<h3>Suggested alterations</h3>
<ol class="arabic simple">
<li>log the 1-p value</li>
<li>convert to Z scores<ul>
<li>not sure exactly what he meant here</li>
<li>*<span class="math">\(z(r) \cdot (1-p) \cdot w(d)\)</span></li>
<li><span class="math">\(r \cdot z(1-p) \cdot w(d)\)</span></li>
<li><span class="math">\(z(r) \cdot z(1-p) \cdot w(d)\)</span></li>
<li><span class="math">\(z(r \cdot (1-p)) \cdot w(d)\)</span></li>
</ul>
</li>
<li>‘fiddle’ with it till the null distribution makes sense...</li>
</ol>
</div>
<div class="section" id="distributions-of-r-and-1-p">
<h3>Distributions of <em>r</em> and <em>1-p</em></h3>
<p>I will now plot the distributions of the separate variables in the PTCI
to determine if they help explain the structure seen above.</p>
<p>In[1]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="c"># run gfunc_build_n_way_one2one script and capture output in variable 'trap'</span>
<span class="c"># NOTE: for syntax highlighting to work on this blog post, I had to comment out the ipython magic command below:</span>

<span class="c"># %run -i gfunc_build_n_way_one2one.py gfunc/Aa_Ag_Cq_sort_pci.conf</span>

<span class="n">graphHandler</span><span class="p">,</span><span class="n">graphBuilder</span><span class="p">,</span><span class="n">n_way_ortho_table</span><span class="p">,</span><span class="n">ortho_parser</span> <span class="o">=</span> <span class="n">trap</span>

<span class="c"># extract edge data from the graphs</span>
<span class="n">edges</span> <span class="o">=</span> <span class="n">graphHandler</span><span class="o">.</span><span class="n">edge_dict</span><span class="o">.</span><span class="n">values</span><span class="p">()</span>


<span class="c"># save original graphHandler in case we alter it and then want to use the original later</span>
<span class="kn">from</span> <span class="nn">copy</span> <span class="kn">import</span> <span class="n">deepcopy</span>
<span class="n">original_graphHandler</span> <span class="o">=</span>  <span class="n">deepcopy</span><span class="p">(</span><span class="n">graphHandler</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><pre>bestChoose is 'comb' from 'gmpy'.
main() completed.</pre>
</div>
<div class="section" id="here-i-define-a-function-edge-correlation-to-calculate-and-return-r-and-its-p">
<h4>Here I define a function ‘<span class="docutils literal"><span class="pre">edge_correlation()</span></span>‘ to calculate and return <em>r</em> and its <em>p</em></h4>
<p>In[2]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="kn">from</span> <span class="nn">scipy</span> <span class="kn">import</span> <span class="n">stats</span> <span class="k">as</span> <span class="n">sp_stats</span>

<span class="k">def</span> <span class="nf">edge_correlation</span><span class="p">(</span><span class="n">gFunc_edge</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Returns the pearson r value and corresponding p-value</span>
<span class="sd">    for a given edge's nodes.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">node1</span><span class="p">,</span><span class="n">node2</span> <span class="o">=</span> <span class="n">gFunc_edge</span><span class="o">.</span><span class="n">nodes</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="n">r_val</span><span class="p">,</span><span class="n">p_val</span> <span class="o">=</span> <span class="n">sp_stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">node1</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">expression_vector</span><span class="p">,</span> <span class="n">node2</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">expression_vector</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">np</span><span class="o">.</span><span class="n">isnan</span><span class="p">(</span><span class="n">r_val</span><span class="p">):</span>
            <span class="k">pass</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">r_val</span><span class="p">,</span><span class="n">p_val</span>

    <span class="k">except</span> <span class="ne">AttributeError</span> <span class="k">as</span> <span class="n">err</span><span class="p">:</span>
        <span class="k">if</span> <span class="s">&quot;&quot;&quot;'Bunch' object has no attribute&quot;&quot;&quot;</span> <span class="ow">in</span> <span class="n">err</span><span class="o">.</span><span class="n">message</span><span class="p">:</span>
            <span class="c"># if this executes then one of the nodes did not have an expression_vector which means no r is possible</span>
            <span class="c"># in this case return None</span>
            <span class="k">return</span> <span class="bp">None</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="c"># if this executes then something ELSE went wrong: thus I will fail.</span>
            <span class="k">raise</span> <span class="n">err</span>
</pre></div>
</div>
<p>In[3]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="c"># collect all the results using edge_correlation()</span>
<span class="n">edge_r_and_p_values</span> <span class="o">=</span> <span class="p">[</span><span class="n">edge_correlation</span><span class="p">(</span><span class="n">edge</span><span class="p">)</span> <span class="k">for</span> <span class="n">edge</span> <span class="ow">in</span> <span class="n">edges</span><span class="p">]</span>

<span class="c"># get rid of any results that equal None</span>
<span class="n">edge_r_and_p_values</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">edge_r_and_p_values</span> <span class="k">if</span> <span class="n">x</span> <span class="o">!=</span> <span class="bp">None</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="section" id="now-we-plot-a-histogram-of-the-r-values">
<h4>Now we plot a histogram of the r values</h4>
<p>In[4]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">r_values</span> <span class="o">=</span> <span class="p">[</span><span class="n">edge_r_and_p_values</span><span class="p">[</span><span class="n">i</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">edge_r_and_p_values</span><span class="p">))]</span>

<span class="n">hist</span><span class="p">(</span><span class="n">r_values</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">'stepfilled'</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">'b'</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">'correlation values'</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">'number of edges in each bin'</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">'r values for pairwise edge comparisons'</span><span class="p">)</span>
</pre></div>
</div>
<p>Out[4]:</p>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x3861e950&gt;</pre>
</div>
<img alt="../../../_images/Exploring_PTCI_fig_00.png" src="http://movingthelamppost.com/blog/html/_images/Exploring_PTCI_fig_00.png"/>
</div>
<div class="section" id="and-now-of-the-1-p-values">
<h4>And now of the (1 - p_values)</h4>
<p>In[5]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">p_values</span> <span class="o">=</span> <span class="p">[</span><span class="n">edge_r_and_p_values</span><span class="p">[</span><span class="n">i</span><span class="p">][</span><span class="mi">1</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">edge_r_and_p_values</span><span class="p">))]</span>
<span class="n">one_minus_pvals</span> <span class="o">=</span> <span class="p">[(</span><span class="mi">1</span><span class="o">-</span><span class="n">p</span><span class="p">)</span> <span class="k">for</span> <span class="n">p</span> <span class="ow">in</span> <span class="n">p_values</span><span class="p">]</span>
<span class="n">hist</span><span class="p">(</span><span class="n">one_minus_pvals</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">'stepfilled'</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">'g'</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">'1-p'</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">'number of edges in each bin'</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">'(1-p) for pairwise edge correlations'</span><span class="p">)</span>
</pre></div>
</div>
<p>Out[5]:</p>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x3ac69b50&gt;</pre>
</div>
<img alt="../../../_images/Exploring_PTCI_fig_01.png" src="http://movingthelamppost.com/blog/html/_images/Exploring_PTCI_fig_01.png"/>
</div>
<div class="section" id="now-the-combined">
<h4>Now the combined <span class="math">\(r \cdot (1-p)\)</span></h4>
<p>In[6]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">r_by_1minusP</span> <span class="o">=</span> <span class="p">[</span><span class="n">r_values</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">*</span> <span class="p">(</span><span class="mi">1</span> <span class="o">-</span> <span class="n">p_values</span><span class="p">[</span><span class="n">i</span><span class="p">])</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">r_values</span><span class="p">))]</span>
<span class="n">hist</span><span class="p">(</span><span class="n">r_by_1minusP</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">'stepfilled'</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">'c'</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">'r * (1-p)'</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">'number of edges in each bin'</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">'r * (1-p) for pairwise edge correlations'</span><span class="p">)</span>
</pre></div>
</div>
<p>Out[6]:</p>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x40ea4090&gt;</pre>
</div>
<img alt="../../../_images/Exploring_PTCI_fig_02.png" src="http://movingthelamppost.com/blog/html/_images/Exploring_PTCI_fig_02.png"/>
<p>In[7]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">hist2d</span><span class="p">(</span><span class="n">r_values</span><span class="p">,</span><span class="n">p_values</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">25</span><span class="p">)</span>
<span class="n">colorbar</span><span class="p">()</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">&quot;r values&quot;</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">&quot;p-values&quot;</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">'2D histogram of r and p-value pairs'</span><span class="p">)</span>
</pre></div>
</div>
<p>Out[7]:</p>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x3862f1d0&gt;</pre>
</div>
<img alt="../../../_images/Exploring_PTCI_fig_03.png" src="http://movingthelamppost.com/blog/html/_images/Exploring_PTCI_fig_03.png"/>
<p>The 2D histogram above shows that as expected the p-values track mostly
but not completely with the r values. Generally, a more extreme r value
will also come with a “better” p-value, but this is not absolute which
warrants keeping the p-value variable in the PTCI in some capacity.
Also, this plot illustrates that the most populated bins are those with
positive r values close to +1.</p>
</div>
<div class="section" id="now-the-full-ptci">
<h4>Now the full PTCI</h4>
<p>In[8]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="c"># define function to calculate the PTCI</span>
<span class="kn">from</span> <span class="nn">gfunc.maths</span> <span class="kn">import</span> <span class="n">weight_d_for_ptci</span> <span class="k">as</span> <span class="n">scale_the_d</span>

<span class="k">def</span> <span class="nf">calc_ptci</span><span class="p">(</span><span class="n">gFunc_edge</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    calculate the PTCI</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="n">r_val</span><span class="p">,</span><span class="n">p_val</span> <span class="o">=</span> <span class="n">edge_correlation</span><span class="p">(</span><span class="n">gFunc_edge</span><span class="p">)</span>

        <span class="n">d_val</span><span class="p">,</span><span class="n">d_min</span><span class="p">,</span><span class="n">d_max</span> <span class="o">=</span> <span class="n">gFunc_edge</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">divergence</span>

        <span class="n">ptci</span> <span class="o">=</span> <span class="n">r_val</span> <span class="o">*</span> <span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">p_val</span><span class="p">)</span> <span class="o">*</span> <span class="n">scale_the_d</span><span class="p">(</span><span class="n">d_val</span><span class="p">,</span><span class="n">d_min</span><span class="p">,</span><span class="n">d_max</span><span class="p">)</span>

        <span class="k">if</span> <span class="ow">not</span> <span class="n">np</span><span class="o">.</span><span class="n">isnan</span><span class="p">(</span><span class="n">ptci</span><span class="p">):</span>
            <span class="c"># If we get a valid ptci store the value in the gFunc_edge object and also return it</span>
            <span class="n">gFunc_edge</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">PTCI</span> <span class="o">=</span> <span class="n">ptci</span>
            <span class="k">return</span> <span class="n">ptci</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="c"># If we get an invalid ptci, store it and return it as None</span>
            <span class="n">gFunc_edge</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">PTCI</span> <span class="o">=</span> <span class="bp">None</span>
            <span class="k">return</span> <span class="bp">None</span>
    <span class="k">except</span> <span class="ne">TypeError</span> <span class="k">as</span> <span class="n">err</span><span class="p">:</span>
        <span class="k">if</span> <span class="nb">str</span><span class="p">(</span><span class="n">err</span><span class="p">)</span> <span class="o">==</span> <span class="s">&quot;'NoneType' object is not iterable&quot;</span><span class="p">:</span>
            <span class="c"># If we get an invalid ptci, store it and return it as None</span>
            <span class="n">gFunc_edge</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">PTCI</span> <span class="o">=</span> <span class="bp">None</span>
            <span class="k">return</span> <span class="bp">None</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">raise</span>
</pre></div>
</div>
<p>In[9]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">pairwise_ptci_vals</span> <span class="o">=</span> <span class="p">[</span><span class="n">calc_ptci</span><span class="p">(</span><span class="n">edge</span><span class="p">)</span> <span class="k">for</span> <span class="n">edge</span> <span class="ow">in</span> <span class="n">edges</span><span class="p">]</span>

<span class="c"># remove any None values</span>
<span class="n">pairwise_ptci_vals</span> <span class="o">=</span> <span class="p">[</span><span class="n">ptci</span> <span class="k">for</span> <span class="n">ptci</span> <span class="ow">in</span> <span class="n">pairwise_ptci_vals</span> <span class="k">if</span> <span class="n">ptci</span> <span class="o">!=</span> <span class="bp">None</span><span class="p">]</span>
</pre></div>
</div>
<p>In[10]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">hist</span><span class="p">(</span><span class="n">pairwise_ptci_vals</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">'stepfilled'</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">'c'</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">'ptci'</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">'number of edges in each bin'</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">'ptci for pairwise edge correlations'</span><span class="p">)</span>
</pre></div>
</div>
<p>Out[10]:</p>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x476b5110&gt;</pre>
</div>
<img alt="../../../_images/Exploring_PTCI_fig_04.png" src="http://movingthelamppost.com/blog/html/_images/Exploring_PTCI_fig_04.png"/>
</div>
</div>
</div>
<div class="section" id="explore-null-distributions-for-pairwise-edge-relationships-randomized-orthology-assignments">
<h2>Explore Null distributions for pairwise edge relationships (randomized orthology assignments)</h2>
<div class="section" id="generate-the-null-distributions">
<h3>Generate the Null distributions</h3>
<p>In[11]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="c"># import edge randomizer function</span>
<span class="kn">from</span> <span class="nn">gfunc.scripts.gfunc_build_n_way_one2one</span> <span class="kn">import</span> <span class="n">reset_random_edges</span>

<span class="c"># Scramble the edges many times and collect the pairwise ptci distributions</span>

<span class="n">reps</span> <span class="o">=</span> <span class="mi">50</span>

<span class="n">null_paired_ptci_distributions</span> <span class="o">=</span> <span class="p">[]</span>

<span class="k">for</span> <span class="n">rep</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">reps</span><span class="p">):</span>
    <span class="c"># scramble edges for this rep and set new r&amp;p vals</span>
    <span class="n">reset_random_edges</span><span class="p">(</span><span class="n">graphHandler</span><span class="p">,</span><span class="n">graphBuilder</span><span class="p">,</span><span class="n">n_way_ortho_table</span><span class="p">,</span><span class="n">ortho_parser</span><span class="p">)</span>
    <span class="n">graphHandler</span><span class="o">.</span><span class="n">measure_relations</span><span class="p">()</span>

    <span class="c"># calculate null ptci vals</span>
    <span class="n">null_edges</span> <span class="o">=</span> <span class="n">graphHandler</span><span class="o">.</span><span class="n">edge_dict</span><span class="o">.</span><span class="n">values</span><span class="p">()</span>
    <span class="n">null_pairwise_ptci_vals</span> <span class="o">=</span> <span class="p">[</span><span class="n">calc_ptci</span><span class="p">(</span><span class="n">edge</span><span class="p">)</span> <span class="k">for</span> <span class="n">edge</span> <span class="ow">in</span> <span class="n">null_edges</span><span class="p">]</span>

    <span class="c"># remove any None values</span>
    <span class="n">null_pairwise_ptci_vals</span> <span class="o">=</span> <span class="p">[</span><span class="n">ptci</span> <span class="k">for</span> <span class="n">ptci</span> <span class="ow">in</span> <span class="n">null_pairwise_ptci_vals</span> <span class="k">if</span> <span class="n">ptci</span> <span class="o">!=</span> <span class="bp">None</span><span class="p">]</span>

    <span class="c"># collect null ptci distribution</span>
    <span class="n">null_paired_ptci_distributions</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">null_pairwise_ptci_vals</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="graph-the-null-distributions">
<h3>Graph the Null distributions</h3>
<p>In[12]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="c"># Show what the actual data looks like for comparison</span>
<span class="n">hist</span><span class="p">(</span><span class="n">pairwise_ptci_vals</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">'stepfilled'</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">'c'</span><span class="p">,</span><span class="n">alpha</span><span class="o">=.</span><span class="mi">7</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s">'Real Data'</span><span class="p">)</span>

<span class="c"># Graph null distributions as grey slightly transparent histograms</span>
<span class="n">null_label</span> <span class="o">=</span> <span class="s">'Null Data'</span>

<span class="k">for</span> <span class="n">null_dist</span> <span class="ow">in</span> <span class="n">null_paired_ptci_distributions</span><span class="p">:</span>
    <span class="n">hist</span><span class="p">(</span><span class="n">null_dist</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">'step'</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">'k'</span><span class="p">,</span><span class="n">alpha</span><span class="o">=.</span><span class="mi">1</span><span class="p">,</span><span class="n">label</span><span class="o">=</span><span class="n">null_label</span><span class="p">)</span>
    <span class="n">null_label</span> <span class="o">=</span> <span class="bp">None</span>




<span class="n">xlabel</span><span class="p">(</span><span class="s">'ptci'</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">'number of edges in each bin'</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">'null distributions for pairwise edge correlations (</span><span class="si">%s</span><span class="s"> reps)'</span> <span class="o">%</span> <span class="p">(</span><span class="n">reps</span><span class="p">))</span>
<span class="n">legend</span><span class="p">()</span>
</pre></div>
</div>
<p>Out[12]:</p>
<div class="highlight-python"><pre>&lt;matplotlib.legend.Legend at 0x35fad690&gt;</pre>
</div>
<img alt="../../../_images/Exploring_PTCI_fig_05.png" src="http://movingthelamppost.com/blog/html/_images/Exploring_PTCI_fig_05.png"/>
</div>
<div class="section" id="observations-and-discussions">
<h3>Observations and Discussions:</h3>
<ol class="arabic simple">
<li>The structure of the null distributions in the pairwise cases are
what I would have predicted<ul>
<li>most pairs show no correlation</li>
<li>there is no bias in either the strong negative or strong positive
correlation zones (near -1 or +1)</li>
</ul>
</li>
<li>[ <strong>QUESTION:</strong> ] It seems that the PTCI math itself is not the
ultimate cause of the strange structure found in the null
distributions of the 3-way comparisons at the start of this document.</li>
<li>[ <strong>QUESTION:</strong> ] This seems to negate the need to use z-score
conversions on any of the PTCI variables??<ul>
<li><strong>ACTUALLY:</strong> this is not totally true, although it will might
<strong>not</strong> fix the shape problem, as the z-score distributions will
all have the same “shape” as the un-converted distributions</li>
<li><strong>BUT:</strong> since the z-score distributions are x-shifted so that “0”
lies at the mean (<em>or alternately: median if I wanted to do it
that way</em>), z-converting the <strong>r values</strong> may still be a good way
to highlight those pairs that are most likely to be positively
correlated? (see figure below)</li>
</ul>
</li>
<li>[ <strong>QUESTION:</strong> ] Would this also suggest that the <strong>shape</strong> of the
p-value distribution is not a primary problem either (that taking the
log of the p-values would not help the shape of the 3-way null
distributions)</li>
<li>[ <strong>PROPOSITION:</strong> ] Perhaps I made a mistake in the way that I
calculated the mean 3-way PTCI values for the null distributions?</li>
<li>[ <strong>PROPOSITION:</strong> ] Perhaps I should exclude any mean PTCI info from
a 3-way ortholog triplicate if any pairwise PTCI is missing (usualy
due to one of the orthologs being undetected). Currently, if one
ortholog expression vector is missing (undetected) I simply take the
mean of the remaining two pairwise comparisons. Maybe this is
introducing strange structural elements in the distribution depending
on which species is missing data?<ul>
<li>I <strong>thought</strong> that the branch length data should correct this but
maybe not?</li>
</ul>
</li>
</ol>
<p>In[13]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="k">def</span> <span class="nf">convert_to_z_scores</span><span class="p">(</span><span class="n">data_vector</span><span class="p">):</span>
    <span class="n">vec_mean</span> <span class="o">=</span> <span class="n">mean</span><span class="p">(</span><span class="n">data_vector</span><span class="p">)</span>
    <span class="n">vec_median</span> <span class="o">=</span> <span class="n">median</span><span class="p">(</span><span class="n">data_vector</span><span class="p">)</span>
    <span class="n">vec_stdev</span> <span class="o">=</span> <span class="n">std</span><span class="p">(</span><span class="n">data_vector</span><span class="p">)</span>

    <span class="n">z_vector</span> <span class="o">=</span> <span class="p">(</span><span class="n">data_vector</span> <span class="o">-</span> <span class="n">vec_median</span><span class="p">)</span> <span class="o">/</span> <span class="p">(</span><span class="n">vec_stdev</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">z_vector</span>
</pre></div>
</div>
<p>In[14]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">z_r_values</span> <span class="o">=</span> <span class="n">convert_to_z_scores</span><span class="p">(</span><span class="n">r_values</span><span class="p">)</span>


<span class="n">subplot</span><span class="p">(</span><span class="mi">211</span><span class="p">)</span>

<span class="n">vert_line_color</span> <span class="o">=</span> <span class="s">'0.75'</span>

<span class="n">hist</span><span class="p">(</span><span class="n">r_values</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">'stepfilled'</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">'b'</span><span class="p">)</span>
<span class="n">axvline</span><span class="p">(</span><span class="n">linewidth</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">vert_line_color</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">'r values'</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">'edges per bin'</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">'r values for edges BEFORE z-convertion'</span><span class="p">)</span>

<span class="n">subplot</span><span class="p">(</span><span class="mi">212</span><span class="p">)</span>
<span class="n">hist</span><span class="p">(</span><span class="n">z_r_values</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">'stepfilled'</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">'b'</span><span class="p">)</span>
<span class="n">axvline</span><span class="p">(</span><span class="n">linewidth</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">vert_line_color</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">'r values'</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">'edges per bin'</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">'r values for edges AFTER z-convertion'</span><span class="p">)</span>

<span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
<img alt="../../../_images/Exploring_PTCI_fig_06.png" src="http://movingthelamppost.com/blog/html/_images/Exploring_PTCI_fig_06.png"/>
</div>
<div class="section" id="next-actions">
<h3>Next Actions:</h3>
<ol class="arabic simple">
<li>Re try calculating the 3-way null distributions: using new code or
code from this notebook rather than gFunc code to see if I simply
made a mistake.</li>
<li>Replot pairwise distribution using z-converted r values.</li>
</ol>
</div>
</div>
</div>]]></description>
             <pubDate>Mon, 08 Apr 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/04/04/hello__world_.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/04/04/hello__world_.html</guid>
            <title><![CDATA[Hello, World!]]></title>
            <description><![CDATA[<div class="section" id="hello-world">
<h1>Hello, World!</h1>
<p>I have recently joined the trend of taking my blog ‘static’ rather than using a
more complicated database driven solution like <a class="reference external" href="http://wordpress.com">Wordpress</a>.  One major reason for
me is that I like to write content in abstraction.  I love <a class="reference external" href="http://en.wikipedia.org/wiki/LaTeX">LaTeX</a> for this reason
and I love Sphinx for the same reason.  I chose <a class="reference external" href="http://www.tinkerer.me">Tinkerer</a> as my static engine
because it is actually built from <a class="reference external" href="http://sphinx-doc.org">Sphinx</a> and it is written in my favorite
language (<a class="reference external" href="http://www.python.org">Python</a>).  Sphinx is a code documentation framework originally built
for python.  I use it to generate my documentation using reStructuredText
markup. I figured since I am more-or-less used to the way Sphinx does its
markup, I would choose a blog engine that worked the same way.  So far, I
like it.  It is a bit less mature than some of the other static engines out
there (<a class="reference external" href="http://octopress.org">Octopress</a> being a VERY popular one) but I think it will grow nicely.</p>
<div class="section" id="the-name-moving-the-lamppost">
<h2>The name <em>“Moving The Lamppost”</em></h2>
<p><em>Moving The Lamppost</em> is a reference to <a class="reference external" href="http://en.wikipedia.org/wiki/Streetlight_effect">the streetlight
effect</a>.  I first heard the
parable from a professor in my Probabilistic Modeling of Biological Data
course.  It goes something like this:</p>
<blockquote>
<div><div id="more"> </div><p>A police officer encounters a drunk fumbling around under a streetlight late
at night and asks him what he is doing.  The drunk responds that he is looking
for his keys.  The officer asks where he lost them. The drunk says he
lost them across the street in the Park. Exasperated, the officer asks him why
on earth he is looking for them over here. “Because this is where the light
is,” responds the drunk.</p>
</div></blockquote>
<p>In one sense this is something of a metaphor for some situations that we find ourselves in
as scientists.  We have a question we want to explore.  An important one.  But
there are times when the tools available to us, or some critical idea or model
related to our problem has yet to be created/formulated.  So we go to work with
the tools we have.  This can cause much wasted time and money, however.  So we
must always have a couple questions in the back of our mind as we pursue our
questions:</p>
<ol class="arabic simple">
<li>“Am I looking where the answer is actually likely to be or simply where the light is?”</li>
<li>“If I am stuck under a lamppost, is there a way to extend the reach of the light so that it includes the answer?” (Can I move this lamppost)</li>
</ol>
<p>From another angle <em>Moving The Lamppost</em> can represent the most basic goal of science: to extend the edges of what is known.</p>
</div>
<div class="section" id="the-purpose-of-this-blog">
<h2>The purpose of this blog</h2>
<p>My main goal for <em>Moving The Lamppost</em> is to serve as a place for me to record random thoughts about my work and what is going on in science at large in such a way that I can engage in dialog with others about it.
Relating to that goal is my desire to foster relationships with other scientists or informed hobbyists that can approach what is termed <a class="reference external" href="http://en.wikipedia.org/wiki/Open_science">Open Science</a>.
I hope to make this my home base for sharing what I am interested in as well as what I am actually working on: including data and figures.</p>
<p>For a nice introduction to the ideals of Open Science take a look at <a class="reference external" href="http://scienceinthesands.blogspot.co.uk/2012/07/7-habits-of-open-scientist.html">the excellent blog series</a> by <a class="reference external" href="http://www.davidketcheson.info">David Ketcheson</a> over at <a class="reference external" href="http://scienceinthesands.blogspot.co.uk">Science in the Sands</a></p>
</div>
</div>]]></description>
             <pubDate>Thu, 04 Apr 2013 00:00:00 -0700</pubDate>
        </item>
    
    </channel>
</rss>