<?xml version="1.0" encoding="utf-8"?>
<rss version="2.0">
    <channel>
        <title>moving the lamppost</title>
        <link>http://movingthelamppost.com/blog/html/</link>
        <description>random musings of a molecular biologist turned code jockey in the era of big data and open science.</description>
        <language>en-us</language>
        <pubDate>Sun, 12 May 2013 00:00:00 -0700</pubDate>
        
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html</guid>
            <title><![CDATA[Elements of transgenic disease-vector control (Part 2: altering the mosquitoes)]]></title>
            <description><![CDATA[<div class="section" id="elements-of-transgenic-disease-vector-control-part-2-altering-the-mosquitoes">
<h1>Elements of transgenic disease-vector control (Part 2: altering the mosquitoes)</h1>
<div class="topic">
<p class="topic-title first">The four fundamentals of transgenic vector control:</p>
<ol class="arabic simple">
<li>the ability to introduce <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-transgene"><em class="xref std std-term">transgenes</em></a> into the mosquito’s genome in a heritable way</li>
<li>discovery or construction <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-effector-gene"><em class="xref std std-term">effector genes</em></a></li>
<li>the ability to control when and where the effector genes are expressed</li>
<li>the ability of your transgenic mosquitoes to have <strong>swift</strong>, dramatic, and clinically meaningful effects on the native vector population (<a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-gene-drive"><em class="xref std std-term">gene drive</em></a>)</li>
</ol>
</div>
<div class="section" id="introducing-inheritable-transgenes-into-the-mosquito-genome">
<h2>Introducing Inheritable Transgenes into the Mosquito Genome</h2>
<p>Before one can employ transgenic methods for vector control, one must of course have the ability to make transgenic vectors.
For mosquitoes, this is quite a bit more labor intensive than for other fly species like <em>Drosophila melanogaster</em>.
However, it has been accomplished, and protocols to reliably introduce transgenes into many mosquito species now exist.
Unfortunately, the relative difficulty is still high, and it takes roughly 2-3 months before a stable line is established.
So while this step can be filed under “accomplished”, work is constantly being done to improve its ease and efficiency.</p>
</div><p><a class="readmore" href="http://movingthelamppost.com/blog/html/2013/05/12/elements_of_transgenic_disease_vector_control_part2.html#more">Read more...</a></p></div>]]></description>
            <category><![CDATA[ My Research ]]></category>
            <category><![CDATA[ My Dissertation ]]></category>
             <pubDate>Sun, 12 May 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/05/09/comparison_of_aedes_aegypti_and_anopheles_gambiae_midgut_expression_post_bloodmeal.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/05/09/comparison_of_aedes_aegypti_and_anopheles_gambiae_midgut_expression_post_bloodmeal.html</guid>
            <title><![CDATA[Comparison of Aedes aegypti and Anopheles gambiae midgut gene expression post bloodmeal: an overview]]></title>
            <description><![CDATA[<div class="section" id="comparison-of-aedes-aegypti-and-anopheles-gambiae-midgut-gene-expression-post-bloodmeal-an-overview">
<h1>Comparison of <em>Aedes</em> <em>aegypti</em> and <em>Anopheles</em> <em>gambiae</em> midgut gene expression post bloodmeal: an overview</h1>
<a class="reference external image-reference" href="/blog/html/_static/talks/2013-05-08_myWork/2013-05-08_myWork/index.html"><img alt="../../../_images/Screenshot.png" class="align-center" src="http://movingthelamppost.com/blog/html/_images/Screenshot.png" style="width: 600px;"/></a>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The links on the discussion slides are now fixed!</p>
</div>
<div class="section" id="introduction-to-new-post-category-my-presentations">
<h2>Introduction to New Post Category: My Presentations</h2>
<p>This is my first post in a new category of posts called “My Presentations” which shockingly will consist of as many of my talks that I am able to share online.
It will most likely only contain talks that I write going forward since most of my other talks are not in ‘website’ form, and I do not have time to convert them.</p>
<p>However, I recently discovered an approach to writing presentations that is pretty similar to how I write this blog (in <a class="reference external" href="http://docutils.sourceforge.net/rst.html">ReST</a> or <a class="reference external" href="http://daringfireball.net/projects/markdown">Markdown</a> then conversion to HTML and JavaScript with a special tool: <a class="reference external" href="http://www.tinkerer.me">Tinkerer</a> for this blog and <a class="reference external" href="https://github.com/egonSchiele/mdpress">mdpress</a> for the presentations).
Therefore the resulting presentations are webpages and easily shareable!
Yay <a class="reference external" href="http://opensciencefederation.com">Open Science</a>!</p>
</div><p><a class="readmore" href="http://movingthelamppost.com/blog/html/2013/05/09/comparison_of_aedes_aegypti_and_anopheles_gambiae_midgut_expression_post_bloodmeal.html#more">Read more...</a></p></div>]]></description>
            <category><![CDATA[ My Research ]]></category>
            <category><![CDATA[ My Presentations ]]></category>
             <pubDate>Thu, 09 May 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/04/21/elements_of_transgenic_disease_vector_control_part1.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/04/21/elements_of_transgenic_disease_vector_control_part1.html</guid>
            <title><![CDATA[Elements of transgenic disease-vector control (Part 1: setting the scene)]]></title>
            <description><![CDATA[<div class="section" id="elements-of-transgenic-disease-vector-control-part-1-setting-the-scene">
<h1>Elements of transgenic disease-vector control (Part 1: setting the scene)</h1>
<div class="section" id="foreword">
<h2>Foreword</h2>
<p>This is the first installment in my endeavor to <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/15/blogging_my_dissertation.html"><em>“Blog My Dissertation”</em></a>.
I am afraid that I have not quite settled on a format for this project or even a solid governing theme on how to organize and execute the bloggification...
So I ask you to please bear with me as I experiment with how technical to be; how polished; how formal/informal; and “dissertation-ready” to make the prose etc.</p>
<p>This post will cover part of the dissertation’s first chapter (Introduction and Background).
The long term goal of the Anthony James lab at UCI is to produce genetically modified mosquitoes that do not transmit <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-malaria"><em class="xref std std-term">malaria</em></a> or <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-dengue-fever"><em class="xref std std-term">dengue fever</em></a> and will spread this trait swiftly through local mosquito populations.
In order to produce mosquitoes that will perform these functions, we use a process called <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-transgenesis"><em class="xref std std-term">transgenesis</em></a> that introduces a set of <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-exogenous"><em class="xref std std-term">exogenous</em></a> genes into the mosquitoes’ <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-genome"><em class="xref std std-term">genome</em></a> and tells them when and where to be expressed.
All of my work should be interpreted in the scheme of aiding the lab’s ability to improve this process; however, there is real basic science that it aims to further as well.</p>
<p>This is part one of a few posts that will discuss the fundamental elements of transgenic disease-<a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-vector"><em class="xref std std-term">vector</em></a> control.
It will set the scene by explaining what a disease-vector is and how our specific models of disease transmission influence how we can approach the problem of malaria and dengue fever by addressing the mosquitoes rather than treating the humans.</p>
</div><p><a class="readmore" href="http://movingthelamppost.com/blog/html/2013/04/21/elements_of_transgenic_disease_vector_control_part1.html#more">Read more...</a></p></div>]]></description>
            <category><![CDATA[ My Research ]]></category>
            <category><![CDATA[ My Dissertation ]]></category>
             <pubDate>Sun, 21 Apr 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/04/19/new_release__blacktie_v0_2_0rc1.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/04/19/new_release__blacktie_v0_2_0rc1.html</guid>
            <title><![CDATA[NEW RELEASE: Blacktie v0.2.0rc1]]></title>
            <description><![CDATA[<div class="section" id="new-release-blacktie-v0-2-0rc1">
<h1>NEW RELEASE: Blacktie v0.2.0rc1</h1>
<p>I have released a new version of blacktie (0.2.0rc1) which has a number of minor fixes and new features.</p>
<ul class="simple">
<li>github repo: <a class="reference external" href="https://github.com/xguse/blacktie">https://github.com/xguse/blacktie</a></li>
<li>documentation: <a class="reference external" href="http://xguse.github.io/blacktie">http://xguse.github.io/blacktie/</a></li>
<li>issue tracking and feature requests: <a class="reference external" href="https://github.com/xguse/blacktie/issues">https://github.com/xguse/blacktie/issues</a></li>
</ul>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">One major gotcha will be that the yaml config file has been changed slightly to facilitate the inclusion of biological replicate data.  So please take a look at the demo config file in the docs posted at: <a class="reference external" href="http://xguse.github.io/blacktie/tutorial.html#the-configuration-file">http://xguse.github.io/blacktie/tutorial.html#the-configuration-file</a></p>
</div>
<p>Thanks and <strong>please</strong> let me know what you think.  I need your input to make improvements.</p>
<p>Gus</p>
<hr class="docutils"/>
<div class="section" id="news">
<h2>News</h2>
<div class="section" id="rc1">
<h3>0.2.0rc1</h3>
<p><em>Release date: 2013-04-19</em></p>
<ul>
<li><p class="first">git tag: ‘v0.2.0rc1’</p>
</li>
<li><p class="first">email_notification is now adjustable for other email servers</p>
</li>
<li><p class="first">Added support for handling biological replicates in cuffdiff runs.</p>
</li>
<li><dl class="first docutils">
<dt>Major changes to yaml config:</dt>
<dd><ul class="first last simple">
<li>condition_queue[index].group_id –&gt; condition_queue[index].experiment_id</li>
<li>addition of condition_queue[index].replicate_id to track replicate data</li>
<li>condition_queue[index].name now represents description of an ‘experimental condition’ and will be shared by replicates.</li>
</ul>
</dd>
</dl>
</li>
</ul>
</div>
</div>
</div>]]></description>
            <category><![CDATA[ My Code ]]></category>
             <pubDate>Fri, 19 Apr 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html</guid>
            <title><![CDATA[Genome-Level Analysis of Selective Constraint without Apparent Sequence Conservation]]></title>
            <description><![CDATA[<div class="section" id="genome-level-analysis-of-selective-constraint-without-apparent-sequence-conservation">
<h1>Genome-Level Analysis of Selective Constraint without Apparent Sequence Conservation</h1>
<div class="section" id="full-citation">
<h2>Full Citation</h2>
<p>Vakhrusheva, O. A, Bazykin, G. A, &amp; Kondrashov, A. S. (2013). <em>Genome-level analysis of selective constraint without apparent sequence conservation</em>. <strong>Genome biology and evolution</strong>, 5(3), 532–541. doi:10.1093/gbe/evt023</p>
<p>Links: <a class="reference external" href="http://www.ncbi.nlm.nih.gov/pubmed/23418180">PubMed</a> – <a class="reference external" href="http://gbe.oxfordjournals.org/content/5/3/532.full.pdf+html">Full Text (PDF)</a> – <a class="reference external" href="http://gbe.oxfordjournals.org/content/5/3/532.full">Full Text (HTML)</a></p>
<p><a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#license-information-pertaining-to-the-original-article">License information pertaining to the original article</a></p>
</div>
<div class="section" id="elevator-speech">
<h2>Elevator Speech</h2>
<div class="section" id="academic-significance">
<h3>Academic Significance</h3>
<p>Other studies have demonstrated individual instances of conservation of sequence elements that are no longer <strong>globally alignable</strong> due to extended phylogenetic divergence.  Familiar examples include the demonstration that some transgenic enhancer regions from orthologous genes in distantly related species are able to drive correct gene expression of the host species even though <strong>global alignment</strong> of the enhancer regions was impossible.</p>
<p>However the authors claim that this is the first genome-scale study they know of that supports the phenomena of selective constraint acting on sequence elements absent of identifiable sequence conservation.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">By terms like ‘identifiable sequence conservation’, they really seem to be referring to the ability to generate meaningful <strong>global alignment</strong> since they plainly use <strong>local alignment</strong> to identity the sequences they propose are being selectively retained.</p>
</div>
</div>
<div class="section" id="main-hypotheses">
<h3>Main Hypotheses</h3>
<p><span class="math">\(H_1\)</span>: the presence of a ‘conserved’ segment between two less divergent species should <em>overall</em> predict the presence of corresponding conserved segments in a separate pair of similarly divergent species that together have a much larger pair-to-pair divergence distance (see <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#f1"><em>Figure 1</em></a> for graphical representation). <strong>[relevant figures:</strong> <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#f2"><em>Figure 2</em></a> <strong>]</strong></p>
<p><span class="math">\(H_2\)</span>: if <em>empirically validated</em> regulatory elements persist longer than the sequence similarity of the corresponding DNA segments, introns containing a regulatory element in pair 1 should carry a corresponding segment in pair 2 at rates higher than expected by chance. <strong>[relevant figures:</strong> <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#f3"><em>Figure 3</em></a> <strong>]</strong></p>
<p><span class="math">\(H_3\)</span>: presence of a functional element in an orthologous intron in one species pair should correlate with a reduced instance of intron loss when compared to an outgroup serving as a way to infer the presence/absence of the functional element in a distant common ancestor. <strong>[relevant figures:</strong> <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#f4"><em>Figure 4</em></a> <strong>]</strong></p>
<p><span class="math">\(H_4\)</span>: because others have already established that longer introns are more likely to carry a segment of conservation than short introns, a function conserved in all four species should be expected to be associated with larger intron length  when conserved in pair 1 <em>and</em> pair 2, compared with the introns lost in pair 2. <strong>[relevant figures:</strong> <a class="reference internal" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#f5"><em>Figure 5</em></a> <strong>]</strong></p>
</div></div><p><a class="readmore" href="http://movingthelamppost.com/blog/html/2013/04/16/vakhrusheva_et_al_gbe_2013.html#more">Read more...</a></p></div>]]></description>
            <category><![CDATA[ Journal Club ]]></category>
            <category><![CDATA[ Journal Club (shorts) ]]></category>
             <pubDate>Tue, 16 Apr 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/04/15/blogging_my_dissertation.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/04/15/blogging_my_dissertation.html</guid>
            <title><![CDATA[Blogging my dissertation]]></title>
            <description><![CDATA[<div class="section" id="blogging-my-dissertation">
<h1>Blogging my dissertation</h1>
<p>I am not surprised or disappointed that <a class="reference external" href="http://chronicle.com/article/How-Blogging-Helped-Me-Write/136893">this idea has already been done</a>.
In fact, I am quite encouraged that I read that others that have done this so far have had such great experiences with it.
I began to think about using my blog to help me get over my reluctance to simply sit down and write for quite a few years.
However, I somehow never made the connection that I could also use it to help me actually write the documents themselves.
It was only recently, after <a class="reference external" href="http://movingthelamppost.com/blog/html/2013/04/04/hello__world_.html">making a personal promise</a> to engage more frequently and vigorously in the <a class="reference external" href="http://en.wikipedia.org/wiki/Open_science">open science</a> endeavor, that I started to research how the idea of blogging a <a class="reference internal" href="http://movingthelamppost.com/blog/html/pages/glossary.html#term-dissertation"><em class="xref std std-term">dissertation</em></a> had worked out for other people.</p>
<p><a class="readmore" href="http://movingthelamppost.com/blog/html/2013/04/15/blogging_my_dissertation.html#more">Read more...</a></p></div>]]></description>
            <category><![CDATA[ My Dissertation ]]></category>
             <pubDate>Mon, 15 Apr 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/04/08/exploring_ptci.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/04/08/exploring_ptci.html</guid>
            <title><![CDATA[Exploring Various ways to combine transcription profile correlation (r), confidence in that correlation (p value associated with r), and phylogenetic distance (w(d)) between species into a useful PTC index (Phylogenetic Transcription Conservation Index)]]></title>
            <description><![CDATA[<div class="section" id="exploring-various-ways-to-combine-transcription-profile-correlation-r-confidence-in-that-correlation-p-value-associated-with-r-and-phylogenetic-distance-w-d-between-species-into-a-useful-ptc-index-phylogenetic-transcription-conservation-index">
<h1>Exploring Various ways to combine transcription profile correlation (<em>r</em>), confidence in that correlation (<em>p value</em> associated with <em>r</em>), and phylogenetic distance (<em>w(d)</em>) between species into a useful PTC index (Phylogenetic Transcription Conservation Index)</h1>
<div class="section" id="original-solution">
<h2>Original solution</h2>
<p><span class="math">\(PTCI = r \cdot (1-p) \cdot w(d)\)</span></p>
<p>Xiaohui objected that <em>1-p</em> is not optimal because the <em>1-p</em> will likely
have a distribution skewed toward <em>1</em>. He suggested loging the variable,
among other options (see below).</p>
<div class="section" id="original-results">
<h3>Original Results</h3>
<a class="reference internal image-reference" href="http://movingthelamppost.com/blog/html/_images/Ag_Aa_Cq_3way_PTCI.png"><img alt="../../../_images/Ag_Aa_Cq_3way_PTCI.png" src="http://movingthelamppost.com/blog/html/_images/Ag_Aa_Cq_3way_PTCI.png" style="width: 600px;"/></a>
<p>This is a histogram of the three way mean PTCIs using the original
solution. Blue=Real data. Grey=Randomizations that abolish the three way
orthology relationships. The ‘Null’ distribution [grey] is significantly
different from the real data [blue] but has significant structure that
is not what one would predict (resemble a normal dist). Notably, there is
a large concentration around -0.5 that I can not explain.</p>
</div></div><p><a class="readmore" href="http://movingthelamppost.com/blog/html/2013/04/08/exploring_ptci.html#more">Read more...</a></p></div>]]></description>
             <pubDate>Mon, 08 Apr 2013 00:00:00 -0700</pubDate>
        </item>
    
        <item>
            <link>http://movingthelamppost.com/blog/html/2013/04/04/hello__world_.html</link>
            <guid>http://movingthelamppost.com/blog/html/2013/04/04/hello__world_.html</guid>
            <title><![CDATA[Hello, World!]]></title>
            <description><![CDATA[<div class="section" id="hello-world">
<h1>Hello, World!</h1>
<p>I have recently joined the trend of taking my blog ‘static’ rather than using a
more complicated database driven solution like <a class="reference external" href="http://wordpress.com">Wordpress</a>.  One major reason for
me is that I like to write content in abstraction.  I love <a class="reference external" href="http://en.wikipedia.org/wiki/LaTeX">LaTeX</a> for this reason
and I love Sphinx for the same reason.  I chose <a class="reference external" href="http://www.tinkerer.me">Tinkerer</a> as my static engine
because it is actually built from <a class="reference external" href="http://sphinx-doc.org">Sphinx</a> and it is written in my favorite
language (<a class="reference external" href="http://www.python.org">Python</a>).  Sphinx is a code documentation framework originally built
for python.  I use it to generate my documentation using reStructuredText
markup. I figured since I am more-or-less used to the way Sphinx does its
markup, I would choose a blog engine that worked the same way.  So far, I
like it.  It is a bit less mature than some of the other static engines out
there (<a class="reference external" href="http://octopress.org">Octopress</a> being a VERY popular one) but I think it will grow nicely.</p>
<div class="section" id="the-name-moving-the-lamppost">
<h2>The name <em>“Moving The Lamppost”</em></h2>
<p><em>Moving The Lamppost</em> is a reference to <a class="reference external" href="http://en.wikipedia.org/wiki/Streetlight_effect">the streetlight
effect</a>.  I first heard the
parable from a professor in my Probabilistic Modeling of Biological Data
course.  It goes something like this:</p>
<blockquote>
<div></div></div><p><a class="readmore" href="http://movingthelamppost.com/blog/html/2013/04/04/hello__world_.html#more">Read more...</a></p></div>]]></description>
             <pubDate>Thu, 04 Apr 2013 00:00:00 -0700</pubDate>
        </item>
    
    </channel>
</rss>