<!DOCTYPE html><!--[if lt IE 7]>      <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]>         <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]>         <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]><!--> <html xmlns="http://www.w3.org/1999/xhtml"
    xmlns:og="http://ogp.me/ns#"
    xmlns:fb="https://www.facebook.com/2008/fbml" class="no-js"> <!--<![endif]-->
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
        <meta name="description" content="random musings of a molecular biologist turned code jockey in the era of big data and open science.">
        <meta name="viewport" content="width=device-width">
        <title>Exploring Various ways to combine transcription profile correlation (r), confidence in that correlation (p value associated with r), and phylogenetic distance (w(d)) between species into a useful PTC index (Phylogenetic Transcription Conservation Index) &mdash; moving the lamppost</title>
            <link rel="stylesheet" href="../../../_static/normalize.css" type="text/css">
            <link rel="stylesheet" href="../../../_static/main.css" type="text/css">
            <link rel="stylesheet" href="../../../_static/modern5.css" type="text/css">
            <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
            <link rel="stylesheet" href="../../../_static/webfont.css" type="text/css">
        <link rel="shortcut icon" href="../../../_static/tinkerer.ico" /><!-- Load modernizr and JQuery -->
        <script src="../../../_static/vendor/modernizr-2.6.2.min.js"></script>
        <script src="//ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js"></script>
        <script>window.jQuery || document.write('<script src="../../../_static/vendor/jquery-1.8.2.min.js"><\/script>')</script>
        <script src="../../../_static/plugins.js"></script>
        <script src="../../../_static/main.js"></script>
        <link rel="next" title="Hello, World!" href="../04/hello__world_.html" /><link rel="prev" title="Blogging my dissertation" href="../15/blogging_my_dissertation.html" /><link rel="alternate" type="application/rss+xml" title="RSS" href="http://feeds.feedburner.com/MovingTheLamppost" /><script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../../../',
        VERSION:     '1.1_wad',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script><script type="text/javascript" src="../../../_static/underscore.js"></script><script type="text/javascript" src="../../../_static/doctools.js"></script><script type="text/javascript" src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><script type="text/javascript" src="../../../_static/disqus.js"></script><script type="text/javascript" src="../../../_static/google_analytics.js"></script>

    <script type="text/javascript">
        $(document).ready(function () {
            // Adjusts document height if sidebar is taller
            var documentwrapper = document.getElementsByTagName("article")[0];
            var sidebar = document.getElementsByClassName("sidebar")[0];

            if (sidebar.offsetHeight > documentwrapper.offsetHeight)
            {
                var mainwrapper = document.getElementsByClassName("main")[0];
                documentwrapper.style.minHeight = mainwrapper.offsetHeight + "px";
            }

            // Scroll to content if on small screen
            if (screen.width < 480)
            {
                $(document).scrollTop(document.getElementsByTagName("article")[0].offsetTop - 44);
            }
        });
    </script>
    
    <link rel="stylesheet" type="text/css" href="http://fonts.googleapis.com/css?family=Droid+Serif:i|Noto+Sans:i|Titillium+Web|Titillium+Web:i"></head>
    <body>
        <!--[if lt IE 7]>
            <p class="chromeframe">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> or <a href="http://www.google.com/chromeframe/?redirect=true">activate Google Chrome Frame</a> to improve your experience.</p>
        <![endif]-->

      <div id="container"><header>
            <hgroup>
              <h1><a href="../../../index.html">moving the lamppost</a></h1><h2>random musings of a molecular biologist turned code jockey in the era of big data and open science.</h2></hgroup>
          </header>
      <nav>
            <ul><li class="quicklink"><div class="rss">
        <a href="http://feeds.feedburner.com/MovingTheLamppost" title="Subscribe via RSS"><span class="webfont">B</span></a>
    </div></li><li class="main-nav">
                  <a href="../../../index.html">Home</a>
                </li>
              <li class="main-nav">
                  <a href="../../../pages/about.html">About</a>
                </li>
              <li class="main-nav">
                  <a href="../../../pages/glossary.html">Glossary</a>
                </li>
              </ul>
          </nav><div class="main-container"><div class="main wrapper clearfix"><article><ul class="related clearfix">
            <li class="left"> &laquo; <a href="../15/blogging_my_dissertation.html">Blogging my dissertation</a></li>
            <li class="right"><a href="../04/hello__world_.html">Hello, World!</a> &raquo; </li>
        </ul><div class="timestamp postmeta">
            <span>April 08, 2013</span>
        </div>
    <div class="section" id="exploring-various-ways-to-combine-transcription-profile-correlation-r-confidence-in-that-correlation-p-value-associated-with-r-and-phylogenetic-distance-w-d-between-species-into-a-useful-ptc-index-phylogenetic-transcription-conservation-index">
<h1>Exploring Various ways to combine transcription profile correlation (<em>r</em>), confidence in that correlation (<em>p value</em> associated with <em>r</em>), and phylogenetic distance (<em>w(d)</em>) between species into a useful PTC index (Phylogenetic Transcription Conservation Index)</h1>
<div class="section" id="original-solution">
<h2>Original solution</h2>
<p><span class="math">\(PTCI = r \cdot (1-p) \cdot w(d)\)</span></p>
<p>Xiaohui objected that <em>1-p</em> is not optimal because the <em>1-p</em> will likely
have a distribution skewed toward <em>1</em>. He suggested loging the variable,
among other options (see below).</p>
<div class="section" id="original-results">
<h3>Original Results</h3>
<a class="reference internal image-reference" href="../../../_images/Ag_Aa_Cq_3way_PTCI.png"><img alt="../../../_images/Ag_Aa_Cq_3way_PTCI.png" src="../../../_images/Ag_Aa_Cq_3way_PTCI.png" style="width: 600px;" /></a>
<p>This is a histogram of the three way mean PTCIs using the original
solution. Blue=Real data. Grey=Randomizations that abolish the three way
orthology relationships. The &#8216;Null&#8217; distribution [grey] is significantly
different from the real data [blue] but has significant structure that
is not what one would predict (resemble a normal dist). Notably, there is
a large concentration around -0.5 that I can not explain.</p>
<div id="more"> </div></div>
<div class="section" id="suggested-alterations">
<h3>Suggested alterations</h3>
<ol class="arabic simple">
<li>log the 1-p value</li>
<li>convert to Z scores<ul>
<li>not sure exactly what he meant here</li>
<li>*<span class="math">\(z(r) \cdot (1-p) \cdot w(d)\)</span></li>
<li><span class="math">\(r \cdot z(1-p) \cdot w(d)\)</span></li>
<li><span class="math">\(z(r) \cdot z(1-p) \cdot w(d)\)</span></li>
<li><span class="math">\(z(r \cdot (1-p)) \cdot w(d)\)</span></li>
</ul>
</li>
<li>&#8216;fiddle&#8217; with it till the null distribution makes sense...</li>
</ol>
</div>
<div class="section" id="distributions-of-r-and-1-p">
<h3>Distributions of <em>r</em> and <em>1-p</em></h3>
<p>I will now plot the distributions of the separate variables in the PTCI
to determine if they help explain the structure seen above.</p>
<p>In[1]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="c"># run gfunc_build_n_way_one2one script and capture output in variable &#39;trap&#39;</span>
<span class="c"># NOTE: for syntax highlighting to work on this blog post, I had to comment out the ipython magic command below:</span>

<span class="c"># %run -i gfunc_build_n_way_one2one.py gfunc/Aa_Ag_Cq_sort_pci.conf</span>

<span class="n">graphHandler</span><span class="p">,</span><span class="n">graphBuilder</span><span class="p">,</span><span class="n">n_way_ortho_table</span><span class="p">,</span><span class="n">ortho_parser</span> <span class="o">=</span> <span class="n">trap</span>

<span class="c"># extract edge data from the graphs</span>
<span class="n">edges</span> <span class="o">=</span> <span class="n">graphHandler</span><span class="o">.</span><span class="n">edge_dict</span><span class="o">.</span><span class="n">values</span><span class="p">()</span>


<span class="c"># save original graphHandler in case we alter it and then want to use the original later</span>
<span class="kn">from</span> <span class="nn">copy</span> <span class="kn">import</span> <span class="n">deepcopy</span>
<span class="n">original_graphHandler</span> <span class="o">=</span>  <span class="n">deepcopy</span><span class="p">(</span><span class="n">graphHandler</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><pre>bestChoose is 'comb' from 'gmpy'.
main() completed.</pre>
</div>
<div class="section" id="here-i-define-a-function-edge-correlation-to-calculate-and-return-r-and-its-p">
<h4>Here I define a function &#8216;<span class="docutils literal"><span class="pre">edge_correlation()</span></span>&#8216; to calculate and return <em>r</em> and its <em>p</em></h4>
<p>In[2]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="kn">from</span> <span class="nn">scipy</span> <span class="kn">import</span> <span class="n">stats</span> <span class="k">as</span> <span class="n">sp_stats</span>

<span class="k">def</span> <span class="nf">edge_correlation</span><span class="p">(</span><span class="n">gFunc_edge</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Returns the pearson r value and corresponding p-value</span>
<span class="sd">    for a given edge&#39;s nodes.</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">node1</span><span class="p">,</span><span class="n">node2</span> <span class="o">=</span> <span class="n">gFunc_edge</span><span class="o">.</span><span class="n">nodes</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="n">r_val</span><span class="p">,</span><span class="n">p_val</span> <span class="o">=</span> <span class="n">sp_stats</span><span class="o">.</span><span class="n">pearsonr</span><span class="p">(</span><span class="n">node1</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">expression_vector</span><span class="p">,</span> <span class="n">node2</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">expression_vector</span><span class="p">)</span>
        <span class="k">if</span> <span class="n">np</span><span class="o">.</span><span class="n">isnan</span><span class="p">(</span><span class="n">r_val</span><span class="p">):</span>
            <span class="k">pass</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">return</span> <span class="n">r_val</span><span class="p">,</span><span class="n">p_val</span>

    <span class="k">except</span> <span class="ne">AttributeError</span> <span class="k">as</span> <span class="n">err</span><span class="p">:</span>
        <span class="k">if</span> <span class="s">&quot;&quot;&quot;&#39;Bunch&#39; object has no attribute&quot;&quot;&quot;</span> <span class="ow">in</span> <span class="n">err</span><span class="o">.</span><span class="n">message</span><span class="p">:</span>
            <span class="c"># if this executes then one of the nodes did not have an expression_vector which means no r is possible</span>
            <span class="c"># in this case return None</span>
            <span class="k">return</span> <span class="bp">None</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="c"># if this executes then something ELSE went wrong: thus I will fail.</span>
            <span class="k">raise</span> <span class="n">err</span>
</pre></div>
</div>
<p>In[3]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="c"># collect all the results using edge_correlation()</span>
<span class="n">edge_r_and_p_values</span> <span class="o">=</span> <span class="p">[</span><span class="n">edge_correlation</span><span class="p">(</span><span class="n">edge</span><span class="p">)</span> <span class="k">for</span> <span class="n">edge</span> <span class="ow">in</span> <span class="n">edges</span><span class="p">]</span>

<span class="c"># get rid of any results that equal None</span>
<span class="n">edge_r_and_p_values</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="n">edge_r_and_p_values</span> <span class="k">if</span> <span class="n">x</span> <span class="o">!=</span> <span class="bp">None</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="section" id="now-we-plot-a-histogram-of-the-r-values">
<h4>Now we plot a histogram of the r values</h4>
<p>In[4]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">r_values</span> <span class="o">=</span> <span class="p">[</span><span class="n">edge_r_and_p_values</span><span class="p">[</span><span class="n">i</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">edge_r_and_p_values</span><span class="p">))]</span>

<span class="n">hist</span><span class="p">(</span><span class="n">r_values</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">&#39;stepfilled&#39;</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">&#39;b&#39;</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">&#39;correlation values&#39;</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">&#39;number of edges in each bin&#39;</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">&#39;r values for pairwise edge comparisons&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Out[4]:</p>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x3861e950&gt;</pre>
</div>
<img alt="../../../_images/Exploring_PTCI_fig_00.png" src="../../../_images/Exploring_PTCI_fig_00.png" />
</div>
<div class="section" id="and-now-of-the-1-p-values">
<h4>And now of the (1 - p_values)</h4>
<p>In[5]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">p_values</span> <span class="o">=</span> <span class="p">[</span><span class="n">edge_r_and_p_values</span><span class="p">[</span><span class="n">i</span><span class="p">][</span><span class="mi">1</span><span class="p">]</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">edge_r_and_p_values</span><span class="p">))]</span>
<span class="n">one_minus_pvals</span> <span class="o">=</span> <span class="p">[(</span><span class="mi">1</span><span class="o">-</span><span class="n">p</span><span class="p">)</span> <span class="k">for</span> <span class="n">p</span> <span class="ow">in</span> <span class="n">p_values</span><span class="p">]</span>
<span class="n">hist</span><span class="p">(</span><span class="n">one_minus_pvals</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">&#39;stepfilled&#39;</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">&#39;g&#39;</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">&#39;1-p&#39;</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">&#39;number of edges in each bin&#39;</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">&#39;(1-p) for pairwise edge correlations&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Out[5]:</p>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x3ac69b50&gt;</pre>
</div>
<img alt="../../../_images/Exploring_PTCI_fig_01.png" src="../../../_images/Exploring_PTCI_fig_01.png" />
</div>
<div class="section" id="now-the-combined">
<h4>Now the combined <span class="math">\(r \cdot (1-p)\)</span></h4>
<p>In[6]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">r_by_1minusP</span> <span class="o">=</span> <span class="p">[</span><span class="n">r_values</span><span class="p">[</span><span class="n">i</span><span class="p">]</span> <span class="o">*</span> <span class="p">(</span><span class="mi">1</span> <span class="o">-</span> <span class="n">p_values</span><span class="p">[</span><span class="n">i</span><span class="p">])</span> <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">r_values</span><span class="p">))]</span>
<span class="n">hist</span><span class="p">(</span><span class="n">r_by_1minusP</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">&#39;stepfilled&#39;</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">&#39;c&#39;</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">&#39;r * (1-p)&#39;</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">&#39;number of edges in each bin&#39;</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">&#39;r * (1-p) for pairwise edge correlations&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Out[6]:</p>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x40ea4090&gt;</pre>
</div>
<img alt="../../../_images/Exploring_PTCI_fig_02.png" src="../../../_images/Exploring_PTCI_fig_02.png" />
<p>In[7]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">hist2d</span><span class="p">(</span><span class="n">r_values</span><span class="p">,</span><span class="n">p_values</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">25</span><span class="p">)</span>
<span class="n">colorbar</span><span class="p">()</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">&quot;r values&quot;</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">&quot;p-values&quot;</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">&#39;2D histogram of r and p-value pairs&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Out[7]:</p>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x3862f1d0&gt;</pre>
</div>
<img alt="../../../_images/Exploring_PTCI_fig_03.png" src="../../../_images/Exploring_PTCI_fig_03.png" />
<p>The 2D histogram above shows that as expected the p-values track mostly
but not completely with the r values. Generally, a more extreme r value
will also come with a &#8220;better&#8221; p-value, but this is not absolute which
warrants keeping the p-value variable in the PTCI in some capacity.
Also, this plot illustrates that the most populated bins are those with
positive r values close to +1.</p>
</div>
<div class="section" id="now-the-full-ptci">
<h4>Now the full PTCI</h4>
<p>In[8]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="c"># define function to calculate the PTCI</span>
<span class="kn">from</span> <span class="nn">gfunc.maths</span> <span class="kn">import</span> <span class="n">weight_d_for_ptci</span> <span class="k">as</span> <span class="n">scale_the_d</span>

<span class="k">def</span> <span class="nf">calc_ptci</span><span class="p">(</span><span class="n">gFunc_edge</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    calculate the PTCI</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="n">r_val</span><span class="p">,</span><span class="n">p_val</span> <span class="o">=</span> <span class="n">edge_correlation</span><span class="p">(</span><span class="n">gFunc_edge</span><span class="p">)</span>

        <span class="n">d_val</span><span class="p">,</span><span class="n">d_min</span><span class="p">,</span><span class="n">d_max</span> <span class="o">=</span> <span class="n">gFunc_edge</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">divergence</span>

        <span class="n">ptci</span> <span class="o">=</span> <span class="n">r_val</span> <span class="o">*</span> <span class="p">(</span><span class="mi">1</span><span class="o">-</span><span class="n">p_val</span><span class="p">)</span> <span class="o">*</span> <span class="n">scale_the_d</span><span class="p">(</span><span class="n">d_val</span><span class="p">,</span><span class="n">d_min</span><span class="p">,</span><span class="n">d_max</span><span class="p">)</span>

        <span class="k">if</span> <span class="ow">not</span> <span class="n">np</span><span class="o">.</span><span class="n">isnan</span><span class="p">(</span><span class="n">ptci</span><span class="p">):</span>
            <span class="c"># If we get a valid ptci store the value in the gFunc_edge object and also return it</span>
            <span class="n">gFunc_edge</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">PTCI</span> <span class="o">=</span> <span class="n">ptci</span>
            <span class="k">return</span> <span class="n">ptci</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="c"># If we get an invalid ptci, store it and return it as None</span>
            <span class="n">gFunc_edge</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">PTCI</span> <span class="o">=</span> <span class="bp">None</span>
            <span class="k">return</span> <span class="bp">None</span>
    <span class="k">except</span> <span class="ne">TypeError</span> <span class="k">as</span> <span class="n">err</span><span class="p">:</span>
        <span class="k">if</span> <span class="nb">str</span><span class="p">(</span><span class="n">err</span><span class="p">)</span> <span class="o">==</span> <span class="s">&quot;&#39;NoneType&#39; object is not iterable&quot;</span><span class="p">:</span>
            <span class="c"># If we get an invalid ptci, store it and return it as None</span>
            <span class="n">gFunc_edge</span><span class="o">.</span><span class="n">data</span><span class="o">.</span><span class="n">PTCI</span> <span class="o">=</span> <span class="bp">None</span>
            <span class="k">return</span> <span class="bp">None</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="k">raise</span>
</pre></div>
</div>
<p>In[9]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">pairwise_ptci_vals</span> <span class="o">=</span> <span class="p">[</span><span class="n">calc_ptci</span><span class="p">(</span><span class="n">edge</span><span class="p">)</span> <span class="k">for</span> <span class="n">edge</span> <span class="ow">in</span> <span class="n">edges</span><span class="p">]</span>

<span class="c"># remove any None values</span>
<span class="n">pairwise_ptci_vals</span> <span class="o">=</span> <span class="p">[</span><span class="n">ptci</span> <span class="k">for</span> <span class="n">ptci</span> <span class="ow">in</span> <span class="n">pairwise_ptci_vals</span> <span class="k">if</span> <span class="n">ptci</span> <span class="o">!=</span> <span class="bp">None</span><span class="p">]</span>
</pre></div>
</div>
<p>In[10]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">hist</span><span class="p">(</span><span class="n">pairwise_ptci_vals</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">&#39;stepfilled&#39;</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">&#39;c&#39;</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">&#39;ptci&#39;</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">&#39;number of edges in each bin&#39;</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">&#39;ptci for pairwise edge correlations&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Out[10]:</p>
<div class="highlight-python"><pre>&lt;matplotlib.text.Text at 0x476b5110&gt;</pre>
</div>
<img alt="../../../_images/Exploring_PTCI_fig_04.png" src="../../../_images/Exploring_PTCI_fig_04.png" />
</div>
</div>
</div>
<div class="section" id="explore-null-distributions-for-pairwise-edge-relationships-randomized-orthology-assignments">
<h2>Explore Null distributions for pairwise edge relationships (randomized orthology assignments)</h2>
<div class="section" id="generate-the-null-distributions">
<h3>Generate the Null distributions</h3>
<p>In[11]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="c"># import edge randomizer function</span>
<span class="kn">from</span> <span class="nn">gfunc.scripts.gfunc_build_n_way_one2one</span> <span class="kn">import</span> <span class="n">reset_random_edges</span>

<span class="c"># Scramble the edges many times and collect the pairwise ptci distributions</span>

<span class="n">reps</span> <span class="o">=</span> <span class="mi">50</span>

<span class="n">null_paired_ptci_distributions</span> <span class="o">=</span> <span class="p">[]</span>

<span class="k">for</span> <span class="n">rep</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="n">reps</span><span class="p">):</span>
    <span class="c"># scramble edges for this rep and set new r&amp;p vals</span>
    <span class="n">reset_random_edges</span><span class="p">(</span><span class="n">graphHandler</span><span class="p">,</span><span class="n">graphBuilder</span><span class="p">,</span><span class="n">n_way_ortho_table</span><span class="p">,</span><span class="n">ortho_parser</span><span class="p">)</span>
    <span class="n">graphHandler</span><span class="o">.</span><span class="n">measure_relations</span><span class="p">()</span>

    <span class="c"># calculate null ptci vals</span>
    <span class="n">null_edges</span> <span class="o">=</span> <span class="n">graphHandler</span><span class="o">.</span><span class="n">edge_dict</span><span class="o">.</span><span class="n">values</span><span class="p">()</span>
    <span class="n">null_pairwise_ptci_vals</span> <span class="o">=</span> <span class="p">[</span><span class="n">calc_ptci</span><span class="p">(</span><span class="n">edge</span><span class="p">)</span> <span class="k">for</span> <span class="n">edge</span> <span class="ow">in</span> <span class="n">null_edges</span><span class="p">]</span>

    <span class="c"># remove any None values</span>
    <span class="n">null_pairwise_ptci_vals</span> <span class="o">=</span> <span class="p">[</span><span class="n">ptci</span> <span class="k">for</span> <span class="n">ptci</span> <span class="ow">in</span> <span class="n">null_pairwise_ptci_vals</span> <span class="k">if</span> <span class="n">ptci</span> <span class="o">!=</span> <span class="bp">None</span><span class="p">]</span>

    <span class="c"># collect null ptci distribution</span>
    <span class="n">null_paired_ptci_distributions</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">null_pairwise_ptci_vals</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="graph-the-null-distributions">
<h3>Graph the Null distributions</h3>
<p>In[12]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="c"># Show what the actual data looks like for comparison</span>
<span class="n">hist</span><span class="p">(</span><span class="n">pairwise_ptci_vals</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">&#39;stepfilled&#39;</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">&#39;c&#39;</span><span class="p">,</span><span class="n">alpha</span><span class="o">=.</span><span class="mi">7</span><span class="p">,</span> <span class="n">label</span><span class="o">=</span><span class="s">&#39;Real Data&#39;</span><span class="p">)</span>

<span class="c"># Graph null distributions as grey slightly transparent histograms</span>
<span class="n">null_label</span> <span class="o">=</span> <span class="s">&#39;Null Data&#39;</span>

<span class="k">for</span> <span class="n">null_dist</span> <span class="ow">in</span> <span class="n">null_paired_ptci_distributions</span><span class="p">:</span>
    <span class="n">hist</span><span class="p">(</span><span class="n">null_dist</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">&#39;step&#39;</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">&#39;k&#39;</span><span class="p">,</span><span class="n">alpha</span><span class="o">=.</span><span class="mi">1</span><span class="p">,</span><span class="n">label</span><span class="o">=</span><span class="n">null_label</span><span class="p">)</span>
    <span class="n">null_label</span> <span class="o">=</span> <span class="bp">None</span>




<span class="n">xlabel</span><span class="p">(</span><span class="s">&#39;ptci&#39;</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">&#39;number of edges in each bin&#39;</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">&#39;null distributions for pairwise edge correlations (</span><span class="si">%s</span><span class="s"> reps)&#39;</span> <span class="o">%</span> <span class="p">(</span><span class="n">reps</span><span class="p">))</span>
<span class="n">legend</span><span class="p">()</span>
</pre></div>
</div>
<p>Out[12]:</p>
<div class="highlight-python"><pre>&lt;matplotlib.legend.Legend at 0x35fad690&gt;</pre>
</div>
<img alt="../../../_images/Exploring_PTCI_fig_05.png" src="../../../_images/Exploring_PTCI_fig_05.png" />
</div>
<div class="section" id="observations-and-discussions">
<h3>Observations and Discussions:</h3>
<ol class="arabic simple">
<li>The structure of the null distributions in the pairwise cases are
what I would have predicted<ul>
<li>most pairs show no correlation</li>
<li>there is no bias in either the strong negative or strong positive
correlation zones (near -1 or +1)</li>
</ul>
</li>
<li>[ <strong>QUESTION:</strong> ] It seems that the PTCI math itself is not the
ultimate cause of the strange structure found in the null
distributions of the 3-way comparisons at the start of this document.</li>
<li>[ <strong>QUESTION:</strong> ] This seems to negate the need to use z-score
conversions on any of the PTCI variables??<ul>
<li><strong>ACTUALLY:</strong> this is not totally true, although it will might
<strong>not</strong> fix the shape problem, as the z-score distributions will
all have the same &#8220;shape&#8221; as the un-converted distributions</li>
<li><strong>BUT:</strong> since the z-score distributions are x-shifted so that &#8220;0&#8221;
lies at the mean (<em>or alternately: median if I wanted to do it
that way</em>), z-converting the <strong>r values</strong> may still be a good way
to highlight those pairs that are most likely to be positively
correlated? (see figure below)</li>
</ul>
</li>
<li>[ <strong>QUESTION:</strong> ] Would this also suggest that the <strong>shape</strong> of the
p-value distribution is not a primary problem either (that taking the
log of the p-values would not help the shape of the 3-way null
distributions)</li>
<li>[ <strong>PROPOSITION:</strong> ] Perhaps I made a mistake in the way that I
calculated the mean 3-way PTCI values for the null distributions?</li>
<li>[ <strong>PROPOSITION:</strong> ] Perhaps I should exclude any mean PTCI info from
a 3-way ortholog triplicate if any pairwise PTCI is missing (usualy
due to one of the orthologs being undetected). Currently, if one
ortholog expression vector is missing (undetected) I simply take the
mean of the remaining two pairwise comparisons. Maybe this is
introducing strange structural elements in the distribution depending
on which species is missing data?<ul>
<li>I <strong>thought</strong> that the branch length data should correct this but
maybe not?</li>
</ul>
</li>
</ol>
<p>In[13]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="k">def</span> <span class="nf">convert_to_z_scores</span><span class="p">(</span><span class="n">data_vector</span><span class="p">):</span>
    <span class="n">vec_mean</span> <span class="o">=</span> <span class="n">mean</span><span class="p">(</span><span class="n">data_vector</span><span class="p">)</span>
    <span class="n">vec_median</span> <span class="o">=</span> <span class="n">median</span><span class="p">(</span><span class="n">data_vector</span><span class="p">)</span>
    <span class="n">vec_stdev</span> <span class="o">=</span> <span class="n">std</span><span class="p">(</span><span class="n">data_vector</span><span class="p">)</span>

    <span class="n">z_vector</span> <span class="o">=</span> <span class="p">(</span><span class="n">data_vector</span> <span class="o">-</span> <span class="n">vec_median</span><span class="p">)</span> <span class="o">/</span> <span class="p">(</span><span class="n">vec_stdev</span><span class="p">)</span>
    <span class="k">return</span> <span class="n">z_vector</span>
</pre></div>
</div>
<p>In[14]:</p>
<div class="code python highlight-python"><div class="highlight"><pre><span class="n">z_r_values</span> <span class="o">=</span> <span class="n">convert_to_z_scores</span><span class="p">(</span><span class="n">r_values</span><span class="p">)</span>


<span class="n">subplot</span><span class="p">(</span><span class="mi">211</span><span class="p">)</span>

<span class="n">vert_line_color</span> <span class="o">=</span> <span class="s">&#39;0.75&#39;</span>

<span class="n">hist</span><span class="p">(</span><span class="n">r_values</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">&#39;stepfilled&#39;</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">&#39;b&#39;</span><span class="p">)</span>
<span class="n">axvline</span><span class="p">(</span><span class="n">linewidth</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">vert_line_color</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">&#39;r values&#39;</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">&#39;edges per bin&#39;</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">&#39;r values for edges BEFORE z-convertion&#39;</span><span class="p">)</span>

<span class="n">subplot</span><span class="p">(</span><span class="mi">212</span><span class="p">)</span>
<span class="n">hist</span><span class="p">(</span><span class="n">z_r_values</span><span class="p">,</span><span class="n">bins</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span><span class="n">histtype</span><span class="o">=</span><span class="s">&#39;stepfilled&#39;</span><span class="p">,</span><span class="n">cumulative</span><span class="o">=</span><span class="bp">False</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="s">&#39;b&#39;</span><span class="p">)</span>
<span class="n">axvline</span><span class="p">(</span><span class="n">linewidth</span><span class="o">=</span><span class="mi">4</span><span class="p">,</span> <span class="n">color</span><span class="o">=</span><span class="n">vert_line_color</span><span class="p">)</span>
<span class="n">xlabel</span><span class="p">(</span><span class="s">&#39;r values&#39;</span><span class="p">)</span>
<span class="n">ylabel</span><span class="p">(</span><span class="s">&#39;edges per bin&#39;</span><span class="p">)</span>
<span class="n">title</span><span class="p">(</span><span class="s">&#39;r values for edges AFTER z-convertion&#39;</span><span class="p">)</span>

<span class="n">tight_layout</span><span class="p">()</span>
</pre></div>
</div>
<img alt="../../../_images/Exploring_PTCI_fig_06.png" src="../../../_images/Exploring_PTCI_fig_06.png" />
</div>
<div class="section" id="next-actions">
<h3>Next Actions:</h3>
<ol class="arabic simple">
<li>Re try calculating the 3-way null distributions: using new code or
code from this notebook rather than gFunc code to see if I simply
made a mistake.</li>
<li>Replot pairwise distribution using z-converted r values.</li>
</ol>
</div>
</div>
</div>

    <div class="postmeta">
        <div class="author">
            <span>Posted by W Augustine Dunn</span>
        </div>
        
        <div class="tags">
            <span>
                Tags:
                <a href="../../../tags/open_science.html">open science</a>, <a href="../../../tags/research.html">research</a>, <a href="../../../tags/ipython.html">ipython</a></span>
        </div>
        </div>
    <div id="disqus_thread"></div><script type="text/javascript">    var disqus_shortname = "movingthelamppost";    var disqus_identifier = "2013/04/08/exploring_ptci";    disqus_thread();</script><noscript>Please enable JavaScript to view the    <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript><ul class="related clearfix">
            <li class="left"> &laquo; <a href="../15/blogging_my_dissertation.html">Blogging my dissertation</a></li>
            <li class="right"><a href="../04/hello__world_.html">Hello, World!</a> &raquo; </li>
        </ul></article><aside class="sidebar"><section>
<div class="widget" id="sub_by_email">
    <h1>Don't use RSS aggregators?</h1>
    <a href="http://feedburner.google.com/fb/a/mailverify?uri=MovingTheLamppost&amp;loc=en_US">Subscribe via email</a>
</div>

</section><section><div class="widget" id="searchbox">
    <h1>Search</h1>
    <form action="../../../search.html" method="get">
        <input type="text" name="q" />
        <button type="submit"><span class="webfont">L</span></button>
    </form>
</div></section><section><div class="widget">
    <h1>Categories</h1>
    <ul><li><a href="../../../categories/background_science.html">Background Science</a> (1)</li><li><a href="../../../categories/journal_club.html">Journal Club</a> (1)</li><li><a href="../../../categories/journal_club__shorts_.html">Journal Club (shorts)</a> (1)</li><li><a href="../../../categories/my_code.html">My Code</a> (2)</li><li><a href="../../../categories/my_dissertation.html">My Dissertation</a> (3)</li><li><a href="../../../categories/my_presentations.html">My Presentations</a> (1)</li><li><a href="../../../categories/my_research.html">My Research</a> (3)</li></ul>
</div></section><section><div class="widget">
    <h1>Tags Cloud</h1>
      <a href="../../../tags/amino_acids.html" style="font-size: 8pt">amino acids</a>&nbsp;&nbsp;
      <a href="../../../tags/background.html" style="font-size: 10pt">background</a>&nbsp;&nbsp;
      <a href="../../../tags/bioinformatics.html" style="font-size: 8pt">bioinformatics</a>&nbsp;&nbsp;
      <a href="../../../tags/blacktie.html" style="font-size: 12pt">blacktie</a>&nbsp;&nbsp;
      <a href="../../../tags/blogging.html" style="font-size: 8pt">blogging</a>&nbsp;&nbsp;
      <a href="../../../tags/cell.html" style="font-size: 8pt">cell</a>&nbsp;&nbsp;
      <a href="../../../tags/conservation.html" style="font-size: 8pt">conservation</a>&nbsp;&nbsp;
      <a href="../../../tags/cuffdiff.html" style="font-size: 10pt">cuffdiff</a>&nbsp;&nbsp;
      <a href="../../../tags/cufflinks.html" style="font-size: 10pt">cufflinks</a>&nbsp;&nbsp;
      <a href="../../../tags/cuffmerge.html" style="font-size: 10pt">cuffmerge</a>&nbsp;&nbsp;
      <a href="../../../tags/cummerbund.html" style="font-size: 10pt">cummeRbund</a>&nbsp;&nbsp;
      <a href="../../../tags/dissertation__chapter_one__background_.html" style="font-size: 10pt">Dissertation: Chapter One (Background)</a>&nbsp;&nbsp;
      <a href="../../../tags/dna.html" style="font-size: 8pt">DNA</a>&nbsp;&nbsp;
      <a href="../../../tags/evolution.html" style="font-size: 8pt">evolution</a>&nbsp;&nbsp;
      <a href="../../../tags/gene_drive.html" style="font-size: 8pt">gene drive</a>&nbsp;&nbsp;
      <a href="../../../tags/gene_expression.html" style="font-size: 10pt">gene expression</a>&nbsp;&nbsp;
      <a href="../../../tags/gene_regulation.html" style="font-size: 8pt">gene regulation</a>&nbsp;&nbsp;
      <a href="../../../tags/genomics.html" style="font-size: 8pt">genomics</a>&nbsp;&nbsp;
      <a href="../../../tags/gmm.html" style="font-size: 10pt">GMM</a>&nbsp;&nbsp;
      <a href="../../../tags/gmo.html" style="font-size: 10pt">GMO</a>&nbsp;&nbsp;
      <a href="../../../tags/gripes.html" style="font-size: 8pt">gripes</a>&nbsp;&nbsp;
      <a href="../../../tags/ipython.html" style="font-size: 8pt">ipython</a>&nbsp;&nbsp;
      <a href="../../../tags/mosquitoes.html" style="font-size: 10pt">mosquitoes</a>&nbsp;&nbsp;
      <a href="../../../tags/my_dissertation.html" style="font-size: 8pt">My Dissertation</a>&nbsp;&nbsp;
      <a href="../../../tags/my_research.html" style="font-size: 8pt">My Research</a>&nbsp;&nbsp;
      <a href="../../../tags/my_research.html" style="font-size: 8pt">my research</a>&nbsp;&nbsp;
      <a href="../../../tags/namespace.html" style="font-size: 8pt">namespace</a>&nbsp;&nbsp;
      <a href="../../../tags/new_release.html" style="font-size: 10pt">new release</a>&nbsp;&nbsp;
      <a href="../../../tags/ngs.html" style="font-size: 12pt">NGS</a>&nbsp;&nbsp;
      <a href="../../../tags/nucleic_acids.html" style="font-size: 8pt">nucleic acids</a>&nbsp;&nbsp;
      <a href="../../../tags/nucleus.html" style="font-size: 8pt">nucleus</a>&nbsp;&nbsp;
      <a href="../../../tags/object_oriented_programming.html" style="font-size: 8pt">object oriented programming</a>&nbsp;&nbsp;
      <a href="../../../tags/open_science.html" style="font-size: 20pt">open science</a>&nbsp;&nbsp;
      <a href="../../../tags/open_source.html" style="font-size: 10pt">open source</a>&nbsp;&nbsp;
      <a href="../../../tags/programming.html" style="font-size: 8pt">programming</a>&nbsp;&nbsp;
      <a href="../../../tags/protein.html" style="font-size: 8pt">protein</a>&nbsp;&nbsp;
      <a href="../../../tags/python.html" style="font-size: 8pt">Python</a>&nbsp;&nbsp;
      <a href="../../../tags/python.html" style="font-size: 10pt">python</a>&nbsp;&nbsp;
      <a href="../../../tags/r.html" style="font-size: 10pt">R</a>&nbsp;&nbsp;
      <a href="../../../tags/research.html" style="font-size: 8pt">research</a>&nbsp;&nbsp;
      <a href="../../../tags/research_blogging.html" style="font-size: 8pt">Research Blogging</a>&nbsp;&nbsp;
      <a href="../../../tags/rna.html" style="font-size: 8pt">RNA</a>&nbsp;&nbsp;
      <a href="../../../tags/rna_seq.html" style="font-size: 15pt">RNA-seq</a>&nbsp;&nbsp;
      <a href="../../../tags/selection.html" style="font-size: 8pt">selection</a>&nbsp;&nbsp;
      <a href="../../../tags/sequence_analysis.html" style="font-size: 8pt">sequence analysis</a>&nbsp;&nbsp;
      <a href="../../../tags/the_central_dogma.html" style="font-size: 8pt">the central dogma</a>&nbsp;&nbsp;
      <a href="../../../tags/tinkerer.html" style="font-size: 8pt">tinkerer</a>&nbsp;&nbsp;
      <a href="../../../tags/tophat.html" style="font-size: 10pt">tophat</a>&nbsp;&nbsp;
      <a href="../../../tags/transcription.html" style="font-size: 8pt">transcription</a>&nbsp;&nbsp;
      <a href="../../../tags/transgenic_mosquitoes.html" style="font-size: 10pt">transgenic mosquitoes</a>&nbsp;&nbsp;
      <a href="../../../tags/translation.html" style="font-size: 8pt">translation</a>&nbsp;&nbsp;
      <a href="../../../tags/vector_control.html" style="font-size: 10pt">vector control</a>
</div></section></aside></div> <!-- #main --></div> <!-- #main-container -->

        <div class="footer-container"><footer class="wrapper">&copy; Copyright 2013, W Augustine Dunn. Powered by <a href="http://www.tinkerer.me/">Tinkerer</a> and <a href="http://sphinx.pocoo.org/">Sphinx</a>.</footer></div> <!-- footer-container -->

      </div> <!--! end of #container --><!--[if lt IE 7 ]>
          <script src="//ajax.googleapis.com/ajax/libs/chrome-frame/1.0.3/CFInstall.min.js"></script>
          <script>window.attachEvent('onload',function(){CFInstall.check({mode:'overlay'})})</script>
        <![endif]-->
    </body>
</html>